Loay Jabre - Last edited: 2023-03-28
This script performs differential expression analysis on the PS117 protein data. 
It first speratates bioassay1 and bioassay 2, does DE analysis and reports fold change. 


The DE analysis includes: 
Iron effect, temperature effect, iron x temperature effect, and a pairwise comparison of the different treatments.

```{r}
#https://www.bioconductor.org/packages/release/bioc/html/DEqMS.html
#https://academic.oup.com/bib/advance-article/doi/10.1093/bib/bbac138/6566001?login=true
#limma package should be better than edgeR
#https://seqqc.wordpress.com/2020/11/28/10-tips-tricks-for-complex-model-matrix-designs-in-dge-analysis/
```

Load necessary packages
```{r}
library(dplyr)
library(plyr)
library(ggplot2)
library(statmod)
library(tidyverse)
library(edgeR)
library(reshape2)
library(missForest)
library(plotly)
#https://www.nature.com/articles/s41598-021-81279-4#:~:text=Missing%20values%20in%20proteomic%20data%20are%20a%20mixture%20of%20MAR,missing%20values%20with%20estimated%20values.

#https://academic.oup.com/bioinformatics/article/28/1/112/219101?login=true  MissForestâ€”non-parametric missing value imputation for mixed-type data
setwd("D:/School/PhD/PS117/data/")
```


#Step 1 - Last edited: 2023-03-28
QE- yes
- Aggregate clusters 
- Massage the data

```{r}
prot_data <- read.csv("all_ps117_taxon-function_peptides_non-normalized_injections-means_20230324.csv", header = T)
prot_data$uniq_cluster <- sapply(lapply(strsplit(prot_data$uniq_cluster, split = ",\\s*"), sort), paste, collapse = ", ") #arrange the 'uniq_cluster' annotations alphabetically in each row to reduce redundancies. 

#since we're only using centric and pennate diatom groups, I changed the annotations of the two groups here. 
prot_data$F <- ifelse (prot_data$F  ==  "Polar-centric-Mediophyceae", "centric", 
                  ifelse (prot_data$F  ==  "Radial-centric-basal-Coscinodiscophyceae", "centric", 
                  ifelse (prot_data$F  ==  "Raphid-pennate", "pennate", 
                  ifelse (prot_data$F  ==  "Araphid-pennate", "pennate", 
                  prot_data$F))))

#work with these columns for now. I will combine all the annotations, number of peptides in each cluster etc. later
clusters <- prot_data [c(4, 37, 40:69)]

clusters2 <- melt(clusters, id.vars=c("uniq_cluster", "F")) %>%
            dplyr::rename (treatment = variable) %>%
            dplyr::rename (unnorm_abundance = value )


#aggregate all the peptides into their clusters (i.e. if two peptides belong to one cluster, aggregate their abundances )
clusters2_aggregated <- aggregate(clusters2$unnorm_abundance, by=list(Category=clusters2$uniq_cluster, clusters2$F, clusters2$treatment), FUN=sum) %>%
                 dplyr::rename (unnorm_abundance = x) %>%
                 dplyr::rename (cluster = Category) %>%
                 dplyr::rename (taxon = Group.2) %>%
                 dplyr::rename (treatment = Group.3)

clusters3 <- dcast(clusters2_aggregated, cluster+taxon~treatment)

clusters3$clustertaxon <- paste(clusters3$cluster, clusters3$taxon, sep = ';')

```


#Step 2A - Last edited: 2023-03-28
Work with BA1 first and run the whole DE analysis, then do the same for BA2 
-Remove  clusters with too many missing values in treatments (75%)
-Impute values for NA
```{r}
#BA1
DE_data <- clusters3 [c(33, 7:17)] 

names(DE_data) <- gsub (pattern = "BA1_*", replacement = "", x =  (names(DE_data))) #remove the BA identifier, so it's important to remember which BA is being DE analysed later on ! 

DE_data [DE_data == 0] <- NA  
DE_data$count_na <- rowSums(is.na(DE_data))

DE_data <- filter (DE_data, grepl ("^0$|^1$|^2$|^3$|^4$|^5$|^6$|^7$|^8$", count_na)) #here, if >75% of the columns are empty, we don't use that clusters

DE_data[is.na(DE_data)] <- 0 
DE_data[,2:12] <- t(apply(DE_data [,2:12], 1, function(x) 
                          replace(x, x ==0, 0.5*min(x[x > 0], na.rm = TRUE)))) #repalce NA in row with 0.5*minimum value 

#imputed <- missForest(DE_data) #using random forest model 
#DE_data <- data.frame (imputed$ximp)

prot_counts <- DE_data[c(2:12)] #select only the samples. 

#BA1
DE_data_groups <- factor(c('T8_HT_Fe', 'T8_HT_Fe','T8_HT_Fe',
                           'T8_HT_noFe', 'T8_HT_noFe', 'T8_HT_noFe',
                           'T8_LT_Fe', 'T8_LT_Fe','T8_LT_Fe',
                           'T8_LT_noFe', 'T8_LT_noFe'))
```

#Step 2B
```{r}
#BA2
DE_data <- clusters3 [c(33, 21:32)] 

names(DE_data) <- gsub (pattern = "BA2_*", replacement = "", x =  (names(DE_data))) #remove the BA identifier, so it's important to remember which BA is being DE analysed later on ! 

DE_data [DE_data == 0] <- NA  
DE_data$count_na <- rowSums(is.na(DE_data))

DE_data <- filter (DE_data, grepl ("^0$|^1$|^2$|^3$|^4$|^5$|^6$|^7$|^8$", count_na)) #here, if >75% of the columns are empty, we don't use that clusters

DE_data[is.na(DE_data)] <- 0 
DE_data[,2:13] <- t(apply(DE_data [,2:13], 1, function(x) 
                          replace(x, x ==0, 0.5*min(x[x > 0], na.rm = TRUE)))) #repalce NA in row with 0.5*minimum value 

#imputed <- missForest(DE_data) #using random forest model 
#DE_data <- data.frame (imputed$ximp)

prot_counts <- DE_data[c(2:13)] #select only the samples. 

#BA2
DE_data_groups <- factor(c('T8_HT_Fe', 'T8_HT_Fe','T8_HT_Fe',
                           'T8_HT_noFe', 'T8_HT_noFe', 'T8_HT_noFe',
                           'T8_LT_Fe', 'T8_LT_Fe','T8_LT_Fe',
                           'T8_LT_noFe','T8_LT_noFe', 'T8_LT_noFe'))
```


#Step 3 - Last edited: 2023-03-28
This is the same for BA1 and BA2, so be careful with which BA is being dealth with 
```{r}
# Create DGEList object with protein counts, groups, and accession
DGE_list<- DGEList(counts = prot_counts, 
                        group = DE_data_groups,
                        genes = DE_data$clustertaxon, #here I'm using clusters are the 'genes', but it could be peptides
                        remove.zeros = TRUE) #this shouldn't do anything, because we imputed values in the data. 

# Create design matrix 
design.mat <- model.matrix(~ 0 + DE_data_groups)

# Get tagwise dispersion of tags
#DGE_disp<- estimateDisp(DGE_list_L, design.mat_L, robust = TRUE)
DGE_disp <- estimateDisp(DGE_list, design.mat)

# Fit to GLM
#fit<- glmQLFit(DGE_disp_L, design.mat_L, robust = FALSE)
fit <- glmFit(DGE_disp, design.mat)
```


#Step 3 - Last edited: 2023-03-28
QC - yes
- Make the contrasts and run DE; this is the same for BA1 and BA2, so be careful with which BA is being run
```{r}
# Fe vs noFe
contrast_Fe <- makeContrasts(((DE_data_groupsT8_HT_Fe + DE_data_groupsT8_LT_Fe - DE_data_groupsT8_HT_noFe - DE_data_groupsT8_LT_noFe)/2), levels = design.mat)

# iron vs no iron LT (pairwise iron)
contrast_Fe_pairwise <- makeContrasts(DE_data_groupsT8_LT_Fe - DE_data_groupsT8_LT_noFe, levels = design.mat)


#HT vs lowTemp (pairwise temperature)
contrast_temp <- makeContrasts(((DE_data_groupsT8_HT_Fe - DE_data_groupsT8_LT_Fe + DE_data_groupsT8_HT_noFe - DE_data_groupsT8_LT_noFe)/2), levels = design.mat)

# LT_noFe vs HT_noFe
contrast_temp_pairwise <- makeContrasts(DE_data_groupsT8_HT_noFe - DE_data_groupsT8_LT_noFe, levels = design.mat)



#iron temp interaction
contrast_Fetemp_interaction = makeContrasts((DE_data_groupsT8_HT_Fe - DE_data_groupsT8_HT_noFe ) - (DE_data_groupsT8_LT_Fe - DE_data_groupsT8_LT_noFe), levels = design.mat)

# pairwise Fe-temp interaction
contrast_Fetempinteraction_pairwise <- makeContrasts(DE_data_groupsT8_LT_noFe - DE_data_groupsT8_HT_Fe, levels = design.mat)
  

# Fit tests
qlf_Fe <- glmLRT (fit, contrast = contrast_Fe)
qlf_Fe_pairwise <- glmLRT (fit, contrast = contrast_Fe_pairwise)

qlf_temp <- glmLRT (fit, contrast = contrast_temp)
qlf_temp_pairwise  <- glmLRT (fit, contrast = contrast_temp_pairwise)

qlf_Fetemp_interaction <- glmLRT (fit, contrast = contrast_Fetemp_interaction)
qlf_Fetempinteraction_pairwise <- glmLRT (fit, contrast = contrast_Fetempinteraction_pairwise)

# Export list of DE proteins per comparison
hits_Fe  <- topTags(qlf_Fe, n = Inf, adjust.method = 'BH', sort.by = 'none')
hits_Fe_pairwise <- topTags(qlf_Fe_pairwise, n = Inf, adjust.method = 'BH', sort.by = 'none')

hits_temp <- topTags(qlf_temp, n = Inf, adjust.method = 'BH', sort.by = 'none')
hits_temp_pairwise <- topTags(qlf_temp_pairwise, n = Inf, adjust.method = 'BH', sort.by = 'none')

hits_Fetemp_interaction <- topTags(qlf_Fetemp_interaction, n = Inf, adjust.method = 'BH', sort.by = 'none')
hits_Fetempinteraction_pairwise <- topTags(qlf_Fetempinteraction_pairwise, n = Inf, adjust.method = 'BH', sort.by = 'none')

# Extract dataframes
hits_Fe <- hits_Fe[[1]][,]
hits_Fe_pairwise <- hits_Fe_pairwise[[1]][,]

hits_temp <- hits_temp[[1]][,]
hits_temp_pairwise <- hits_temp_pairwise[[1]][,]

hits_Fetemp_interaction <- hits_Fetemp_interaction[[1]][,]
hits_Fetempinteraction_pairwise <- hits_Fetempinteraction_pairwise[[1]][,]

#combine all DE analyses into one dataframe
colnames(hits_Fe)[2:6] <- paste("Fe", colnames(hits_Fe[,c(2:6)]), sep = "_")
colnames(hits_Fe_pairwise)[2:6] <- paste("Fe_pairwise", colnames(hits_Fe_pairwise[,c(2:6)]), sep = "_")

colnames(hits_temp)[2:6] <- paste("temp", colnames(hits_temp[,c(2:6)]), sep = "_")
colnames(hits_temp_pairwise)[2:6] <- paste("temp_pairwise", colnames(hits_temp_pairwise[,c(2:6)]), sep = "_")

colnames(hits_Fetemp_interaction)[2:6] <- paste("Fetempinteraction", colnames(hits_Fetemp_interaction[,c(2:6)]), sep = "_")
colnames(hits_Fetempinteraction_pairwise)[2:6] <- paste("Fetempinteraction_pairwise", colnames(hits_Fetempinteraction_pairwise[,c(2:6)]), sep = "_")

combined_DE <- join_all (list(hits_Fe, hits_Fe_pairwise, hits_temp, hits_temp_pairwise, hits_Fetemp_interaction, hits_Fetempinteraction_pairwise ), by = "genes", type = "full")

combined_DE_BA1 <- combined_DE[c(1,2,6,7,11,12,16,17,21,22,26, 27, 31)] %>% separate (genes, into=c("cluster","taxon_F"), sep = ";")

write.csv(combined_DE_BA1, "ps117_BA2_clusterDE_20230328.csv", row.names = FALSE)
```



#Step 4 - Last edited: 2023-03-28
Join the DE analyses for BA1 and BA2 ? under construction


#number of missing values - 
- might use later (?)
```{r}

BA2 <- data.frame(table (DE_data$count_na)) #count occurances of certain values 

colnames(BA1)[2] <- paste("BA1", colnames(BA1[,c(2)]), sep = "_")

colnames(BA2)[2] <- paste("BA2", colnames(BA1[,c(2)]), sep = "_")

missing <- merge(BA1, BA2, by = "Var1",  all = T)

ggplot (missing, aes(x = Var1,
                     y = BA2_))+
  geom_col()+
  xlab (expression ("Number of samples where protein is missing")) +
  ylab (expression ("BA2 - Number of proteins ")) +

  theme_bw()+
  theme(axis.text.x=element_text(face = "bold", size = 15, color = "black"),
        axis.title.y=element_text(size=15, color = "black"), 
                axis.title.x=element_text(size=15, color = "black"), 

        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))

```




#DE peptides.. maybe later 
```{r}
prot_data_L <- read.csv("all_ps117_taxon_function_non-normalized_injection_means_20220421.csv", header = T)

#prot_data_L [prot_data_L == 0] <- NA


#norm_prot_data_L <- prot_data_L %>% mutate_at (24:53, funs ((. / sum(.)))) #this normalizes the data to total peptide abundance

DE_data_L <- prot_data_L [c(1, 28:38)]

DE_data_L = na.omit(DE_data_L)

prot_counts_L <- DE_data_L[,2:length(DE_data_L)]

# Make vector with treatment groups
DE_data_groups_L <- factor(c('BA1_T8_HT_Fe', 'BA1_T8_HT_Fe','BA1_T8_HT_Fe', 
                           'BA1_T8_HT_noFe', 'BA1_T8_HT_noFe', 'BA1_T8_HT_noFe', 
                           'BA1_T8_LT_Fe', 'BA1_T8_LT_Fe','BA1_T8_LT_Fe', 
                           'BA1_T8_LT_noFe', 'BA1_T8_LT_noFe'))

#groups <- paste('fe', samples_present$fe, 'temp', samples_present$temp, sep = '_')
```