Loay Jabre; Last updated: 2023-04-20
- These scripts analyze and plot proteomis and other data for the PS117 paper. 

To do - 
Figure out SD and SE

#I) Load packages - 2023-03-08
```{r}
#plotting 
library(ggplot2)
library(ggplotify)
library(cowplot)
library(ggridges)
library(ggpubr)
library(gridExtra)
library(grid)
library(gplots)
library(plotly)
library(ggthemes)
library(ggh4x) #mapping multiple aesthetics to colors 
library(patchwork) #allows for plot insets. e.g. map and the DFe and DMn in Figure 1
library (ggrepel) #makes point labels not touch each other in RA plots
#library(treemapify)
 
#other
library(data.table)
library(dplyr)
library(magrittr)
library(reshape2)
library(stringr)
library(scales)
library(tidyr)
library(tidyverse)
library(vegan) #for making NMDS analyses
library(goeveg) #for making NMDS analyses

#annotations <- fread("D:/School/PhD/PS117/data/datannotations/annotation_allTFG.grpnorm_mmetsp_fc_pn_reclassified.edgeR.csv") 
#colorblind friendly colors 
#https://rdrr.io/cran/ggthemes/man/colorblind.html

#list of colors 
#https://www.computerhope.com/htmcolor.htm #colors

#multi-pannel plots
#http://www.sthda.com/english/articles/24-ggpubr-publication-ready-plots/81-ggplot2-easy-way-to-mix-multiple-graphs-on-the-same-page/
```

#II) Import data of interest - 2023-03-30
```{r}
setwd("D:/School/PhD/PS117/data/") #this is the directory where I keep all the data

#import non-protein data
nonproteindata <- read.csv("LJ_ps117_allnonproteindata_20230308.csv", header = T)
nonproteindata_long <- melt (nonproteindata, id.vars = c(1:15, 17))

#import protein data
proteindata_unnorm <- read.csv("all_ps117_taxon-function_peptides_non-normalized_injections-means_20230324.csv", header = T)

#arrange the 'uniq_cluster' annotations alphabetically in each row to reduce redundancies. 
proteindata_unnorm$uniq_cluster <- sapply(lapply(strsplit(proteindata_unnorm$uniq_cluster, split = ",\\s*"), sort), paste, collapse = ", ") 

### a quick test to show how to normalize to all of proteome, vs to group
# https://stackoverflow.com/questions/50332613/normalize-by-group-for-all-columns
#group <- c("A", "A", "B", "B", "C", "C", "C", "A")
#abundance <- c(1,3,4,5,6,7,8,0)
#all_norm <- c(1,3,4,5,6,7,8,0)
#grp_norm <- c(1,3,4,5,6,7,8,0)
#x <- data.frame(group, abundance, all_norm, grp_norm) %>% mutate_at (3, funs ((. / sum(.)))) # %>%   group_by(group) %>% #mutate_at (4, funs ((. / sum(.))
```

#III) Plotting functions and other useful things- 2023-04-17
```{r}
#used in Figure 1, Figure 3
T0_T8_FenoFe <- function(yaxis, yaxislabel) {
  ggplot(filter (nonproteindata, grepl ("T0|T8", full_treatment_name)), 
                 aes(x = iron_treatment,
                 y = {{yaxis}},
                 color = temperature_treatment, 
                 shape = iron_treatment)) + 
  stat_summary(fun = mean, geom = "point", size = 4, stroke = 1.5, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, linewidth = 1, width = 0.1)+
  scale_color_manual(name = "Temperature",
                     breaks = c("LT","HT"),
                     labels = c("LT", "HT"),
                     values = c('black', 'darkorange2','black')) +
  scale_shape_manual(name = "Iron",
                       breaks = c("Fe","noFe", "T0"), 
                       labels = c("Fe", "noFe", "T0"),
                       values = c(19,1,5)) +
  scale_x_discrete(limits = c("T0","noFe","Fe"), 
                   #labels = c("T0", "T8(-Fe)", "T8(+Fe)"))+
                  #labels = expression(bold("T0"), bold("T8")[~-Fe], bold("T8")[~+Fe]))+
                  labels = expression(bold("T0"), bold("T8")~"(-Fe)", bold("T8")~"(+Fe)"))+


  xlab(expression ("")) +
  ylab (substitute(yaxislabel))+ 
  facet_wrap(~bioassay)+
  theme_bw()+
  theme(axis.text=element_text(face = "bold", size = 12, color = "black"),
        axis.title.x=element_text(size = 14,face = "bold", color = "black"), 
        axis.title.y=element_text(size=16, color = "black"), 
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 12, face = "bold"),
        legend.position = "none") 
}

#T0_T8_FenoFe (fvfm, F[v]/F[m]) + ylim(0.2, NA)


#used in Figure 1
x_vs_time <- function (yaxis, yaxislabel) {
  ggplot(nonproteindata, aes(x = daycount, 
                             y= {{yaxis}},
                             color = temperature_treatment, 
                             shape = iron_treatment)) + 
  stat_summary(fun = mean, geom = "point", show.legend = F, size = 4, stroke = 1.5, alpha = 0.5)+
  stat_summary(fun = mean, geom = "line", linewidth = 0.9, aes(linetype = iron_treatment))+
  stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1, width = 0.1) +
  ylab (substitute(yaxislabel)) + 
  xlab (expression ("")) +
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("LT", "HT"),
                     values = c('black', "darkorange", "blue")) +
  scale_linetype_manual(name = "Iron",
                        breaks = c("Fe","noFe"),   #this removes the T0 from the legend
                        labels = c("Fe", "noFe"),
                        values = c("solid", "dotted", "solid")) +
  scale_shape_manual (name = "Iron",
                       breaks = c("noFe","Fe", "T0"), 
                       labels = c("noFe", "Fe", "T0"),
                       values = c(1, 19, 5))+
  scale_x_continuous( labels = c("T0","T2","T4", "T6", "T8"))+
    facet_wrap(~bioassay)+
    theme_bw()+
        theme(axis.text=element_text(face = "bold", size = 12, color = "black"),
        axis.title.x=element_text(size = 14,face = "bold", color = "black"), 
        axis.title.y=element_text(size=16, color = "black"), 
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 12, face = "bold"),
        legend.position = "none") 
}


#this legend is used in most of the plots, so it's good to have 
vertical_legend <- ggplot(filter (nonproteindata, grepl ("T0|T8", full_treatment_name) ), 
       aes(x=iron_treatment, 
       y=fvfm,
       color = tempfe_treatment, 
       shape = tempfe_treatment))+
  geom_point(size = 4, alpha =0.65, stroke =1.5)+
  scale_color_manual(breaks = c("LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("LT -Fe", "LT +Fe", "HT -Fe", "HT +Fe"),
                      values = c('black', 'black', 'darkorange2', "darkorange2", "black")) +
  scale_shape_manual(breaks = c("LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("LT -Fe", "LT +Fe", "HT -Fe", "HT +Fe"),
                      values = c(1, 19,1, 19, 5)) +
  theme_bw()+
  theme(legend.title = element_blank(),
        legend.text = element_text(size = 12, margin = margin(unit = 'cm')), 
        legend.margin = margin(c(0,0,0,0)), 
        legend.spacing.x = unit(0, "mm"),
        legend.spacing.y = unit(0, "mm"))

vertical_legend <- get_legend(vertical_legend)
vertical_legend <- as_ggplot(vertical_legend)
vertical_legend
```

#Figure 1 - 2023-04-17
- Map of Antarctica showing bioassay sites - 2023-04-20
```{r}
library(ggOceanMaps)
#the numbers in the limits represent the range of the longs and lats 
#the first two numbers represent the longitude range to be shown, the secnd two numbers are the latitude range to be shown. 
bioassay_latlon <- data.frame(lon = c(-11.044,-0.011), lat =c(-70.25, -65), bioassay= c("BA2", "BA1")) 

#basemap(limits = c(-130, 100, -90,-55),
map <- basemap(limits = c(-55),
        bathymetry = TRUE, #this adds the bathymetry (depth info) 
        glaciers = FALSE, 
        grid.col = "NA",
        grid.size = 0.2,
        lat.interval = 5)+ 
        #xlab("Longitude")+
        labs(x = "", y = "") +
  
        theme(axis.text = element_blank(), 
              axis.ticks.y =element_blank(),
              axis.ticks.x =element_blank(),
              legend.position = "none")+
        #annotation_scale(location = "br", text_cex = 1, text_face = "bold") + #this is map scale
        geom_spatial_point(data = bioassay_latlon, aes(x = lon, y = lat), color = "red", alpha = 0.7, size = 3) +
        geom_spatial_text(size = 5, fontface = "bold", aes(label=bioassay_latlon$bioassay, x = bioassay_latlon$lon, 
                                                            y = bioassay_latlon$lat), hjust =1.2, vjust = 0, color = "black") +
        theme(plot.margin=grid::unit(c(0,0,0,0), "mm")) 
 

maplegend <- basemap(limits = c(-55),
        bathymetry = TRUE, #this adds the bathymetry (depth info) 
        glaciers = FALSE)+ 
        labs(x = "", y = "") +
        theme(legend.title = element_text(size = 10, face = "bold"),
              legend.text = element_text(face = "bold", size = 8), 
              legend.position = "top",
              legend.direction = "horizontal")+
        theme(plot.margin=grid::unit(c(0,0,0,0), "mm"), 
              legend.margin = margin(c(t = -20)), 
              legend.spacing.x = unit(0, "mm"),
              legend.spacing.y = unit(0, "mm"))
  
 
maplegend <- get_legend(maplegend)
maplegend <- as_ggplot(maplegend)
maplegend


##Additional information about maps: 
#https://mran.microsoft.com/snapshot/2021-03-14/web/packages/ggOceanMaps/vignettes/ggOceanMaps.html
#https://cran.r-project.org/web/packages/ggOceanMaps/ggOceanMaps.pdf
#bioassays <- data.frame(lon = c(165.4133), lat =c(-77.61895), bioassay= c("")) # mcmurdo ross sea TFG incubations
```

Dissolved Fe and Mn 2023-04-20
```{r}
inset <- ggplot(filter (nonproteindata_long, !grepl ("0.35", value) & grepl("DFe56|DMn", variable) & grepl ("T0", full_treatment_name)), 
       aes (x= variable,
       y = value))+
       stat_summary(fun = mean, geom = "bar", size = 4, fill = "grey60")+
       stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1) +
       labs(x = expression (""), y = expression ("In situ (nM)"))+
       facet_wrap(~bioassay)+
       theme_bw()+
       scale_x_discrete(labels=c('DFe', 'DMn'))+
    theme(axis.text=element_text(face = "bold", size = 12, color = "black"),
        axis.title.x=element_text(size = 14,face = "bold", color = "black"), 
        axis.title.y=element_text(size=16, color = "black"), 
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 12, face = "bold"))+
     ylim(0,0.2) 
     
# ggplot(filter (nonproteindata_long, !grepl ("0.35", value) & grepl("DCu|DZn", variable) & grepl ("T0", full_treatment_name)), 
#        aes (x= variable,
#        y = value))+
#        stat_summary(fun = mean, geom = "bar", size = 4)+
#        stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1) +
#        labs(x = expression (""), y = expression ("nM"))+
#        facet_wrap(~bioassay)+
#        theme_bw()+
#        scale_x_discrete(labels=c('DCu', 'DZn'))+
#        theme(axis.text = element_text(size = 9, color = "black"),
#              axis.title.y = element_text(size=12, color = "black" ), 
#              strip.background = element_rect(fill = "white"),
#              strip.text.x = element_text(size = 10, face = "bold"), 
#              panel.grid.major = element_blank(), panel.grid.minor = element_blank())
```

- Fv/Fm, Chla, POC, Nitrate, phosphate, silicate
```{r}
fvfm <- T0_T8_FenoFe (fvfm, F[v]/F[m]) + ylim(0.2, NA)
poc <- T0_T8_FenoFe (POC_uM, POC~~(mu*M))
chla <- T0_T8_FenoFe (total_chla_ug_L, Total~chl-a~~(mu*g~L^-1))
no3<- x_vs_time(NO3_uM, NO[3]~~(mu*M)) + ylim(5, 30)
po4 <- x_vs_time(PO4_uM, PO[4]~~(mu*M))  + ylim(0, 2) 
#si <- T0_T8_FenoFe (Si_uM, Si~mu*M) + ylim(0,80)

si_no3_drawdown <- read.csv("LJ_ps117_si_consumption.csv", header = T)
sino3drawdown <- ggplot(filter (si_no3_drawdown, grepl ("8|0", daycount) & grepl("BA1|", bioassay)), 
       aes(x=iron_treatment, 
       y=SiNO3ratio,
       color = temperature_treatment, 
       shape = iron_treatment))+ 
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 4, stroke = 1.5, alpha = 0.65)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1.3, width = 0.1) +
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("LT", "HT"),
                     values = c('black', 'darkorange2','black')) +
  scale_shape_manual(name = "Iron",
                       breaks = c("Fe","noFe", "T0"), #this removes the T0 from the legend
                       labels = c("+Fe", "noFe", "T0"),
                       values = c(19,1,5)) +
  scale_x_discrete(limits=c("noFe","Fe"), 
                  labels = expression(bold("-Fe"), bold("+Fe")))+ 
  ylab (expression (Si:NO[3]~drawdown~~(M:M))) + 
 # ylab (expression (Delta*Si:Delta*NO[3]~(M:M))) + 
  xlab (expression ("")) +
  theme_bw()+
  theme(axis.text=element_text(face = "bold", size = 12, color = "black"),
        axis.title.x=element_text(size = 14,face = "bold", color = "black"), 
        axis.title.y=element_text(size=14, color = "black"), 
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 12, face = "bold"),
        legend.position = "none") +
  ylim(0,2.5)+
  facet_wrap(~bioassay)

sino3drawdown
```

Figure 1 plot 2023-04-20
```{r}
x <- plot_grid(NULL, NULL, fvfm, NULL, no3,
               NULL, NULL, chla, NULL,  po4, 
               inset, NULL,  poc, NULL, sino3drawdown,
       labels = c("A","", "C", "", "F",
                  "", "",  "D", "", "G",
                  "B","",  "E", "", "H"), 
       ncol = 5, 
       rel_widths = c(0.7,0.05,1,0.05,1),
       align = "vh")

y <- ggdraw (x) + draw_plot (map, x = -0.04, y = 0.49, width = 0.35, height = 0.5)  +
          draw_plot (maplegend , x = 0.09, y= 0.37, width = 0.1, height = 0.1)

#png("ps117_figure1_highres_20230421.png", width=19.2, height=9.4, units="in", res=900) 
plot_grid(y, vertical_legend, 
               ncol = 2, 
               rel_widths = c(1, 0.05), 
               rel_heights = c(1, 0.5))
#dev.off()
```



#Figure 2 - 2023-03-09
- Pigments heatmap
```{r}
pigment_fract <- nonproteindata [c(15,62:67)] #use the 'fract' columns of all groups 
pigment_fract <- na.omit(pigment_fract)
pigment_fract2 <- melt(pigment_fract, id.vars=c("full_treatment_name")) %>%
            rename(taxon_group = variable) %>%
            rename(fract = value )

pigment_fract2 <- pigment_fract2 %>% separate(full_treatment_name, c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate")) 

pigment_fract2$fulltreatment <- paste(pigment_fract2$temperature_treatment,pigment_fract2$iron_treatment, sep = "_")

pigment_fract2 <- pigment_fract2 %>% separate(taxon_group, c("bla", "taxon_group")) 

pigment_fract2$taxon_group <- reorder(pigment_fract2$taxon_group, pigment_fract2$fract) #this reorders the data so when it's plotted, the most abundant fraction is on the bottom #only wroks when there are no NA's

#dev.off()
#pigmentheatmap
pigment_means <- aggregate(pigment_fract2$fract, by=list( pigment_fract2$fulltreatment, pigment_fract2$bioassay, pigment_fract2$taxon_group), FUN=mean) %>%
  rename (fulltreatment = Group.1, 
          bioassay = Group.2, 
          taxon_group = Group.3, 
          fract = x)
pigment_means$taxon_group2 <-   ifelse (pigment_means$taxon_group ==  "diatoms", "Bacillariophyta", 
                   ifelse (pigment_means$taxon_group ==  "haptophytes", "Haptophyta", 
                   ifelse (pigment_means$taxon_group ==  "pelagophytes", "Pelagophyceae", 
                   ifelse (pigment_means$taxon_group ==  "cryptophytes", "Cryptophyta",
                   ifelse (pigment_means$taxon_group ==  "chlorophytes", "Chlorophyta", 
                   ifelse (pigment_means$taxon_group ==  "dinoflagellates", "Dinophyta",
                           "else" 
                    ))))))

pigment_means$taxon_group2 <- reorder(pigment_means$taxon_group2, pigment_means$fract) #this reorders the data so when it's plotted, the most abundant fraction is on the bottom #only wroks when there are no NA's

pigment_means$treatment2 <- ifelse (pigment_means$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (pigment_means$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (pigment_means$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (pigment_means$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))

#x = factor(treatment2, level = tempFe_order)

fig2_pigmentmeans <-ggplot(pigment_means, aes(x = treatment2,
                                              y = taxon_group2, 
                                              fill = fract)) +  
    geom_tile(color = "black", lwd = 0.7)+
    #coord_fixed()+ #this makes it all squares 
    scale_fill_gradientn(colours = c("deepskyblue2","white","darkred"),
                         values = rescale(c(0.0000001,0.35,.9)),
                         guide = "colorbar", limits=c(0.0000001,0.9),
                       name = "Fraction")+
    labs(x = "", y =  "")+
  ggtitle("Pigment") +
  theme (plot.title = element_text(size = 12, vjust = -8, hjust = 0.01))+
    scale_x_discrete(limits=c("T0", "LT_noFe","LT_Fe", "HT_noFe", "HT_Fe"))+
    facet_grid(.~ bioassay, scales = "free_x") + 
    theme(axis.text.x= element_blank(),
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank(), 
        axis.ticks  = element_blank()) +
    theme(strip.background =element_rect(fill="white"), 
        strip.text = element_text(size = 11, face = "bold"),
        legend.box.margin=margin(l=-10))+
         theme(plot.margin=grid::unit(c(-2.5,0,0,0), "mm")) 

fig2_pigmentmeans



#stacked bar chart
# ggplot (pigment_fract2, aes(
#   x = fulltreatment,
#   y = fract, 
#   fill = taxon_group)) +  
#   geom_bar(position = "fill", stat = "identity") + #position = "fill" does percentage, stack does raw values
#   facet_grid(.~ bioassay)+
# theme_bw()+
#    theme (axis.text.x = element_text(angle = 90)) +
#    ylab ("Contribution to total pigment (fraction)")+
#     xlab ("")+
#   scale_x_discrete(limits = c('T0_T0', 'LT_noFe', 'LT_Fe', 'HT_noFe', 'HT_Fe'))+
#    scale_fill_manual(name="",
#                      values = c("diatoms" = "#117733",
#                                 "dinoflagellates" = "#0072B2",
#                                 "haptophytes" = "#CC79A7",
#                                 "prasinophytes" = "#88CCEE",
#                                 "cryptophytes" = "grey", #000000",
#                                 "pelagophytes" = "#E69F00"))+
#   theme(axis.text.x=element_text(face = "bold", size = 18, color = "black", vjust = 1, hjust = 1, angle = 45),
#         axis.title.y=element_text(size=20,face = "bold", color = "black"), 
#         axis.text.y=element_text(face = "bold", size = 15, color = "black"),
#         strip.background =element_rect(fill = "white"),
#         legend.text = element_text(size = 15), 
#         strip.text.x = element_text(size = 18, face = "bold"))
```

- Protein heatmap - 2023-03-09 
```{r}
allgroups <- proteindata_unnorm [c(32,34,35, 40:69)]

allgroups <- melt(allgroups, id.vars=c("domain", "C","class")) %>%
             rename(treatment = variable) %>%
             rename(unnorm_abundance = value )
sum(allgroups$unnorm_abundance) #this should equal 1.893E13 (total abundance of all peptides for all samples)

#normalize the peptide abundances for within each sample, irrespective of taxonomy
allgroups <- allgroups %>% #this creates a new column of normalized protein abundances grouped by treatment   
  group_by(treatment) %>% 
  mutate(norm_abundance = unnorm_abundance / sum(unnorm_abundance)) %>% 
  ungroup()
sum(allgroups$norm_abundance) #this should = 30 after normalization (number of replicates)

allgroups$class_a <- ifelse (allgroups$class ==  "Bacillariophyta", "Bacillariophyta", 
           
                    ifelse (allgroups$C ==  "Dinophyta", "Dinophyta",
                                    
                    ifelse (allgroups$class ==  "Haptophyta", "Haptophyta",
                    ifelse (allgroups$class ==  "Prymnesiophyceae", "Haptophyta",
                    ifelse (allgroups$class ==  "Pavlovophyceae", "Haptophyta",
                                    
                    ifelse (allgroups$class ==  "Cryptophyceae", "Cryptophyta",
                    ifelse (allgroups$class ==  "Cryptophyta", "Cryptophyta",
                                    
                    ifelse (allgroups$class ==  "Pelagophyceae", "Pelagophyceae",
                                    
                    ifelse (allgroups$class ==  "Chlorophyceae", "Chlorophyta",
                    ifelse (allgroups$class ==  "Chlorophyta", "Chlorophyta",
                    ifelse (allgroups$class ==  "Pyramimonadales", "Chlorophyta",
                    ifelse (allgroups$class ==  "Prasinophyceae", "Chlorophyta",
                            
                    ifelse (allgroups$class ==  "Spirotrichea", "Ciliophora",
                    ifelse (allgroups$class ==  "Prostomatea", "Ciliophora",  
                    ifelse (allgroups$class ==  "Oligohymenophorea", "Ciliophora",
                    ifelse (allgroups$class ==  "Litostomatea", "Ciliophora",   
                    ifelse (allgroups$class ==  "Heterotrichea", "Ciliophora", 
                    ifelse (allgroups$class ==  "Colpodea", "Ciliophora", 
                    ifelse (allgroups$class ==  "Ciliophora", "Ciliophora", 
                            
                    ifelse (allgroups$class == "Eukaryota", "Other Eukaryota",
                    ifelse (allgroups$domain == "Bacteria", "Bacteria",
                    ifelse (allgroups$domain == "Viruses", "Viruses",
                    "Other")))))))))))))))))))))) #this takes a second

allgroups_aggregated <- aggregate(allgroups$norm_abundance, by=list( allgroups$treatment, allgroups$class_a), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.1) %>%
                  rename(class = Group.2) %>% 
                  separate(treatment, into =  c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"))  

allgroups_aggregated$class <- reorder (allgroups_aggregated$class , allgroups_aggregated$norm_abundance)

allgroups_aggregated$tempfe_treatment<- paste (allgroups_aggregated$temperature_treatment, allgroups_aggregated$iron_treatment, sep = "_")

allgroups_aggregated_means <- aggregate(allgroups_aggregated$norm_abundance, by=list (allgroups_aggregated$tempfe_treatment, allgroups_aggregated$bioassay, allgroups_aggregated$class), FUN=mean) %>%
  rename (treatment = Group.1, 
          bioassay = Group.2, 
          class = Group.3, 
          norm_abundance = x)

allgroups_aggregated_means$treatment2 <- ifelse (allgroups_aggregated_means$treatment ==  "T0_T0", "T0",
                                                 allgroups_aggregated_means$treatment)

fig2_proteintopmeans <- ggplot(filter(allgroups_aggregated_means, grepl ("Bacillar|Hapto|Bacteria", class)), 
                               aes (x = treatment2,  y = class, fill= norm_abundance))+ 
  geom_tile(color = "black", lwd = 0.7) +
 # coord_fixed(ratio=2.5)+ #this makes it all squares 
  scale_fill_gradientn(colours = c("deepskyblue2","white","darkred"),
                       values = rescale(c(0.0000001,0.22,0.5)),
                       guide = "colorbar", limits=c(0.0000001,0.5),
                       name = "Fraction")+
   labs(x = "", y = "")+
   scale_x_discrete(limits=c("T0", "LT_noFe","LT_Fe", "HT_noFe", "HT_Fe"))+
   theme(axis.text.x=element_blank(),
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank(), 
        axis.ticks  = element_blank()) +
    ggtitle("Protein") +
  theme (plot.title = element_text(size = 12, vjust = -8, hjust = 0.01))+
     #guides(fill = guide_colourbar(barwidth = 1, barheight = 8)) +
  facet_grid(.~ bioassay)+
  theme(strip.background =element_rect(fill="white"), 
        strip.text = element_text(size = 11, face = "bold"), 
        plot.margin=margin(b=0, unit="cm"), 
        legend.box.margin=margin(l=-10)) + 
  theme(plot.margin=grid::unit(c(-5,0,0,0), "mm")) 

fig2_proteintopmeans

fig2_proteinbottommeans <- ggplot(filter(allgroups_aggregated_means, grepl ("Dinophy|Ciliop|Cryptophy|Pelagophy|Chlorophy", class)), 
       aes (x = treatment2,  y = class, fill= norm_abundance))+ 
  geom_tile(color = "black", lwd = 0.7) +
  scale_fill_gradientn(colours = c("deepskyblue2","white","darkred"),
                       values = rescale(c(0.0000001,0.002,.034)),
                       guide = "colorbar", limits=c(0.0000001,0.034),
                       name = "Fraction")+
   labs(x = "", y = "")+
   scale_x_discrete(limits=c("T0", "LT_noFe","LT_Fe", "HT_noFe", "HT_Fe"))+
   theme(axis.text.x=element_text(face = "bold", size = 11, color = "black", vjust = 1.2, hjust = 1,  angle = 45),
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank(), 
        axis.ticks  = element_blank()) +
  facet_grid(.~ bioassay)+
  theme(strip.background =element_rect(fill="white"), 
        strip.text = element_blank(), 
        plot.margin=margin(b=0, unit="cm"), 
        legend.box.margin=margin(l=-10)) + 
  theme(plot.margin=grid::unit(c(0,0,0,0), "mm")) 
fig2_proteinbottommeans
```

The vegan package displays 'species' and 'sites'. The sites are the treatments, the 'species' are the variables (pigments, FC-groups, proteomics species etc) 

First, make NMDS analyses from peptide, FC and pigment data. No need to re-run this! 
Peptide NMDS -  2023-03-026 
```{r}
protein_nmds <- proteindata_unnorm [c(1, 40:69)]

protein_nmds <- melt(protein_nmds, id.vars=c("peptide")) %>%
             rename(treatment = variable) %>%
             rename(unnorm_abundance = value )
sum(protein_nmds$unnorm_abundance) #this should equal 1.893E13 (total abundance of all peptides for all samples)

#normalize the peptide abundances for within each sample, irrespective of taxonomy
protein_nmds <- protein_nmds %>% #this creates a new column of normalized protein abundances grouped by treatment   
  group_by(treatment) %>% 
  mutate(norm_abundance = unnorm_abundance / sum(unnorm_abundance)) %>% 
  ungroup()

sum(protein_nmds$norm_abundance) #this should = 30 after normalization (number of replicates)

protein_nmds2 <- dcast(protein_nmds, treatment~peptide, fun.aggregate = sum,  value.var = "norm_abundance") #this creates MANY columns

protein_nmds2 <- protein_nmds2 %>% separate(treatment, into=c("bioassay", "timepoint" ,"temperature_treatment", "iron_treatment", "replicate"), sep = "_")

protein_nmds3 = protein_nmds2 [c(6:37477)] 

protein_nmds3 = as.matrix(protein_nmds3)

dimcheckMDS(protein_nmds3,
  distance = "bray",
  k = 4,
  trymax = 1000,
  autotransform = TRUE)

set.seed(123)

nmds_protein = metaMDS(protein_nmds3, try = 1000, trymax = 1000, k = 2, distance = "bray") # try = 500, trymax = 500
nmds_protein
scores(nmds_protein, display = "sites")


stressplot(nmds_protein, p.col = "black", l.col = "blue", title("Peptides")) #for the suplement

data.scores_protein <- as.data.frame(scores(nmds_protein, "sites")) #can select sites (treatments) or species (variable)

data.scores_protein$temperature_treatment <- protein_nmds2$temperature_treatment
data.scores_protein$iron_treatment <- protein_nmds2$iron_treatment
data.scores_protein$bioassay <- protein_nmds2$bioassay

data.scores_protein$bioassaytempfe <- paste(data.scores_protein$bioassay, data.scores_protein$temperature_treatment,data.scores_protein$iron_treatment, sep = "_")
```

Flow cytometry NMDS - 2023-03-26
```{r}
FC <- nonproteindata [c(15,69,71,73,75,77,79,81,83,85,87,89,91,93,95,97,99,101,103,105, 107,109,111,113)] %>% na.omit() %>% separate(full_treatment_name, c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate")) 

FC_nmds <- FC [c(6:28)] 

FC_nmds = as.matrix(FC_nmds)

set.seed(123)
nmds_FC = metaMDS(FC_nmds, try = 10000, distance = "bray") # try = 500, trymax = 500
nmds_FC

data.scores_FC <- as.data.frame(scores(nmds_FC, "sites"))
stressplot(nmds_FC, p.col = "black", l.col = "blue", title("Flow Cytometry")) #for the suplement

data.scores_FC$temperature_treatment <- FC$temperature_treatment
data.scores_FC$iron_treatment <- FC$iron_treatment
data.scores_FC$bioassay <- FC$bioassay

data.scores_FC$bioassaytempfe <- paste(data.scores_FC$bioassay, data.scores_FC$temperature_treatment,data.scores_FC$iron_treatment, sep = "_")
```

Pigment NMDS - 2023-03-26 
```{r}
pigment_nmds <- nonproteindata [c(15,49:54)] %>% na.omit() %>% separate(full_treatment_name, c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate")) 

pigment_nmds2 <- pigment_nmds [c(6:11)] 

pigment_nmds2 = as.matrix(pigment_nmds2)

set.seed(123)
nmds_pigment = metaMDS(pigment_nmds2, try = 1000, distance = "bray") # try = 500, trymax = 500
nmds_pigment

data.scores_pigment <- as.data.frame(scores(nmds_pigment, "sites"))
stressplot(nmds_pigment, p.col = "black", l.col = "blue", title("Pigments")) #for the suplement

data.scores_pigment$temperature_treatment <- pigment_nmds$temperature_treatment
data.scores_pigment$iron_treatment <- pigment_nmds$iron_treatment
data.scores_pigment$bioassay <- pigment_nmds$bioassay

data.scores_pigment$bioassaytempfe <- paste(data.scores_pigment$bioassay, data.scores_pigment$temperature_treatment,data.scores_pigment$iron_treatment, sep = "_")
```

Then combine all three analyses outputs into one file for plotting - 20220421
```{r}
#data.scores_FC$analysis <- "FC"
#data.scores_pigment$analysis <- "pigment"
#data.scores_protein$analysis <- "peptide"

#ps117_nmds_20230421 <- rbind (data.scores_FC, data.scores_pigment, data.scores_protein)
#write.csv(ps117_nmds_20230421, "ps117_nmds_20230421.csv", row.names = F)
```

After making all the NMDS analysis and combining the results, make nmds plots 2023-04-21
```{r}
nmdsplotdata <- read.csv ("ps117_nmds_20230421.csv")

nmdsplots <- function (nmdsdata, title) {
          ggplot((filter (nmdsplotdata, grepl (nmdsdata, analysis))), aes(x = NMDS1, y = NMDS2))+
          geom_point(size = 3, alpha = 0.75, stroke = 1.2, 
                    aes(fill = temperature_treatment,
                    shape = bioassaytempfe,
                    color = temperature_treatment))+
          scale_shape_manual(values = c("BA1_T0_T0" = 1, 
                                        "BA1_LT_noFe" = 1,
                                        "BA1_LT_Fe" = 19,
                                        "BA1_HT_Fe" = 19,
                                        "BA1_HT_noFe" = 1, 
                                        "BA2_T0_T0" = 2, 
                                        "BA2_LT_noFe" = 2,
                                        "BA2_LT_Fe" = 17,
                                        "BA2_HT_Fe" = 17,
                                        "BA2_HT_noFe" =2 ))+
          scale_color_manual(values = c("HT" = "darkorange", "LT" = "black", "T0" = "purple"))+
          scale_fill_manual(values = c("HT" = "darkorange", "LT" = "black", "T0" = "purple"), guide = FALSE) +
          ggtitle({title})+
          theme(axis.text = element_text(colour = "black", size = 10, face = "bold"), 
                axis.title = element_text(face = "bold", size = 12), 
                legend.position = "none", 
                panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
                legend.key=element_blank(), 
                plot.title = element_text(vjust = -7, hjust = 0.01))
}



library(ggalt)

peptide_nmds <- nmdsplots ("peptide", "Protein") + geom_encircle(aes (fill = bioassay), 
                                                 s_shape = 0.5,alpha = 0.2) 
                                
fc_nmds <- nmdsplots ("FC", "Flow Cytometry") + geom_encircle(aes (fill = bioassay), 
                                                 s_shape = 0.5,alpha = 0.2) 

pigment_nmds <- nmdsplots ("pigm", "Pigment") + geom_encircle(aes (fill = bioassay), 
                                                  s_shape = 0.5,alpha = 0.2) 

```

Normalized diatom and Phaeocystis protein abundance - 2023-03-10 
```{r}
allgroups <- proteindata_unnorm [c(32,34,35, 37, 40:69)]

allgroups <- melt(allgroups, id.vars=c("domain", "C","class", "F")) %>%
             rename(treatment = variable) %>%
             rename(unnorm_abundance = value )
sum(allgroups$unnorm_abundance) #this should equal 1.893E13 (total abundance of all peptides for all samples)

#normalize the peptide abundances for within each sample, irrespective of taxonomy
allgroups <- allgroups %>% #this creates a new column of normalized protein abundances grouped by treatment   
  group_by_at(c("treatment")) %>% #this can be normalized to treatment and/or class
  mutate(norm_abundance = unnorm_abundance / sum(unnorm_abundance)) %>% 
  ungroup()
sum(allgroups$norm_abundance) #this should = 30 after normalization (number of replicates)

allgroups$taxon <- ifelse (grepl("pennate", allgroups$F),  "Pennate diatom", 
                    ifelse (grepl("centric", allgroups$F), "Centric diatom", 
                    ifelse (grepl("Phaeocystaceae", allgroups$F), "Phaeocystis",
                    allgroups$F))) 

allgroups_aggregated <- aggregate(allgroups$norm_abundance, by=list( allgroups$treatment, allgroups$taxon), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.1) %>%
                  rename(taxon = Group.2) %>% 
                  separate(treatment, into =  c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"))  

allgroups_aggregated <- allgroups_aggregated %>%
  mutate(tempfe_treatment = paste(temperature_treatment, iron_treatment, sep = "_")) %>%
  mutate(tempfe_treatment = str_replace(tempfe_treatment, "T0_T0", "T0"))

centricgroup <- ggplot ((filter(allgroups_aggregated, grepl("Centric", taxon))), aes(
  x = tempfe_treatment,
  y = norm_abundance)) +  
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 4, alpha =0.7,  position= position_dodge(width=0.4))+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1, position =     
               position_dodge(width=0.4))+   facet_grid(.~ bioassay )+
  theme_bw()+
  labs (x = "", y =  (expression(paste(""))))+
  ggtitle("Centric Diatoms") +
  scale_x_discrete(limits=c("T0", "LT_noFe","LT_Fe", "HT_noFe", "HT_Fe"))+
  theme(axis.text.x= element_blank(),
        axis.title.y=element_text(size=12,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 10, face = "bold"), 
        strip.text.x = element_text(size = 11, face = "bold"), 
        legend.position = "none", 
        plot.title = element_text(size = 12))+
   guides(shape = guide_legend(override.aes = list(size=5))) + 
  ylim (0,NA) #+ #this locks the legend in position so it doesnt move around as you zoom in and out of the plot)
  #theme(plot.margin = unit(c(0, 0, 0, 0), "cm"))
centricgroup


pennategroup <- ggplot ((filter(allgroups_aggregated, grepl("Pennate", taxon))), aes(
  x = tempfe_treatment,
  y = norm_abundance)) +  
  ggtitle("Pennate Diatoms")+

  stat_summary(fun = mean, geom = "point", show.legend = T, size = 4, alpha = 0.7 ,position= position_dodge(width=0.4))+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1, position =     
               position_dodge(width=0.4))+   facet_grid(.~ bioassay )+
  theme_bw()+
  labs (x = "", y =  (expression(paste(""))))+
  scale_x_discrete(limits=c("T0", "LT_noFe","LT_Fe", "HT_noFe", "HT_Fe"))+
  theme(axis.text.x=element_blank(),
        axis.title.y=element_text(size=14,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 11, face = "bold"), 
        plot.title = element_text(size = 12),
        legend.position = "none") + 
  ylim(0,NA) #+ 
  #theme(plot.margin = unit(c(0, 0, 0, 0), "cm"))
pennategroup

phaeogroup <- ggplot ((filter(allgroups_aggregated, grepl("Phaeocystis", taxon))), aes(
  x = tempfe_treatment,
  y = norm_abundance)) +  
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 4,  position= position_dodge(width=0.4), alpha = 0.7)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1, position =     
               position_dodge(width=0.4))+   facet_grid(.~ bioassay )+
  theme_bw()+
  ggtitle("Phaeocystis") +
  labs (x = "", y =  (expression(paste(""))))+
  scale_x_discrete(limits=c("T0", "LT_noFe","LT_Fe", "HT_noFe", "HT_Fe"))+
  theme(axis.text.x=element_text(face = "bold", size = 11, color = "black", vjust = 1, hjust = 1,  angle = 45),
        axis.title.y=element_text(size=14,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 11, face = "bold"), 
        plot.title = element_text(size = 12),
        legend.position = "none") + 
  ylim (0, NA) #+ theme(plot.margin = unit(c(0, 0, 0, 0), "cm"))
phaeogroup
```

Looking at photosynthesis proteins instead of all proteins - 2023-03-18. 
Just an exploration, but something to consider. I do this a bit differently later. 
```{r}
#First make sure that all the ribosomal peptides are annotated as such, and that the groups are properly annotated 
proteindata_unnorm$compartment <- ifelse (grepl("Chloroph|chloroph|Photosy|Phycobilisome", proteindata_unnorm$uniq_cluster_annotation), "photosynthesis", 
                                  ifelse (grepl("photosynth|Photosynth|Porphyrin and chlorophyll metabolism", proteindata_unnorm$uniq_KO_pathway), "photosynthesis",                                                   proteindata_unnorm$uniq_cluster_annotation))


# proteindata_unnorm$compartment <- ifelse (proteindata_unnorm$uniq_grpnorm_compartment == "chloro", "photosynthesis", 
#                                          proteindata_unnorm$uniq_cluster_annotation)

#test <- proteindata_unnorm [c(5, 21, 70, 21, 5,7, 13 )]      

compartment <- proteindata_unnorm [c(70, 37, 40:69  )]                           

compartment$taxon <- ifelse (compartment$F ==  "Polar-centric-Mediophyceae", "Centric diatom", 
                 ifelse (compartment$F ==  "Radial-centric-basal-Coscinodiscophyceae", "Centric diatom", 
                 ifelse (compartment$F ==  "Raphid-pennate", "Pennate diatom", 
                 ifelse (compartment$F ==  "Araphid-pennate", "Pennate diatom", 
                 ifelse (compartment$F ==  "Phaeocystaceae", "Phaeocystis",
                                          compartment$F)))))

compartment2 <- compartment [c(1,33,3:32)]

compartment2 <- melt(compartment2, id.vars=c("compartment", "taxon")) %>%
            rename(treatment = variable) %>%
            rename(unnnorm_abundance = value)
            #separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")


#now, I calculated the contribution of diatom and haptophyte photosynthesis proteins to total photosynthesis proteins
compartment3 <- filter (compartment2, grepl ("photosynthesis", compartment))

compartment_aggregated_group  <- aggregate(compartment3$unnnorm_abundance, 
                                  by=list(Category=compartment3$compartment, compartment3$treatment, compartment3$taxon), FUN=sum) %>%
                        rename(unnorm_abundance = x) %>%
                        rename(compartment = Category) %>%
                        rename(treatment = Group.2) %>% 
                        rename(taxon = Group.3) 

compartment_aggregated_group  <- compartment_aggregated_group  %>% 
                          mutate(groupnorm_abundance = unnorm_abundance / sum(unnorm_abundance)) 

sum (compartment_aggregated_group$groupnorm_abundance)

compartment_aggregated_group <- compartment_aggregated_group %>% 
  separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")


compartment_aggregated_group <- compartment_aggregated_group %>%
  mutate(tempfe_treatment = paste(temperature_treatment, iron_treatment, sep = "_")) %>%
  mutate(tempfe_treatment = str_replace(tempfe_treatment, "T0_T0", "T0"))

ggplot ((filter(compartment_aggregated_group, grepl("Phaeocystis", taxon))), aes(
  x = tempfe_treatment,
  y = groupnorm_abundance)) +  
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 4, alpha =0.7,  position= position_dodge(width=0.4))+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1, position =     
               position_dodge(width=0.4))+   facet_grid(.~ bioassay )+
  theme_bw()+
  labs (x = "", y =  (expression(paste(""))))+
  ggtitle("Diatoms") +
  scale_x_discrete(limits=c("T0", "LT_noFe","LT_Fe", "HT_noFe", "HT_Fe"))+
  theme(axis.text.x=element_text(face = "bold", size = 11, color = "black", vjust = 1, hjust = 1,  angle = 45),
        axis.title.y=element_text(size=14,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 10, face = "bold"), 
        strip.text.x = element_text(size = 11, face = "bold"), 
        legend.position = "none", 
        plot.title = element_text(size = 14))+
       guides(shape = guide_legend(override.aes = list(size=5))) + 
  ylim (0,NA) #this locks the legend in position so it doesnt move around as you zoom in and out of the plot)


grpnorm_compartment <- function (compartments, taxon, yaxislabel) { 
ggplot(filter (compartment_aggregated_group, grepl (compartments, compartment) & grepl (taxon, taxon_F)),
       aes(x= iron_treatment,
       y= groupnorm_abundance,
       color = temperature_treatment, 
       shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  stat_summary(fun = mean, geom = "point", size = 4, stroke = 1.5, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, linewidth = 1, width = 0.1)+
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("LT", "HT"),
                     values = c('black', 'darkorange2','black')) +
  scale_shape_manual(name = "Iron",
                       breaks = c("Fe","noFe", "T0"), 
                       labels = c("Fe", "noFe", "T0"),
                       values = c(19,1,5)) +
  scale_x_discrete(limits=c("T0","noFe","Fe"), 
                   labels = c("T0", "T8 (-Fe)", "T8 (+Fe)"))+
  xlab(expression ("")) +
  ylab (substitute(yaxislabel))+ 

  theme_bw()+
  theme(axis.text=element_text(face = "bold", size = 12, color = "black"),
        axis.title.x=element_text(size = 14,face = "bold", color = "black"), 
        axis.title.y=element_text(size=16, color = "black"), 
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 12, face = "bold"),
        legend.position = "none") 
}

D <- grpnorm_compartment ("^chloro$", "Pennate di", atop ("Pennate diatom", paste ("ribosomal protein fraction"))) 
E <- grpnorm_compartment ("^chloro$", "Centric di", atop ("Centric diatom", paste ("ribosomal protein fraction"))) 
F <- grpnorm_compartment ("^chloro$", "Phaeocystis", atop ("Phaeocystis", paste ("ribosomal protein fraction"))) 
```

Fig 2 legend  2023-03-13.
```{r}
nmdsplotdata$tempfe_treatment <- paste (nmdsplotdata$temperature_treatment, nmdsplotdata$iron_treatment, sep = "_")

legendA <- ggplot(nmdsplotdata, 
       aes(x=NMDS1, 
       y=NMDS2,
       color = tempfe_treatment, 
       shape = tempfe_treatment))+
  geom_point(size = 4, alpha =0.65, stroke =1.5)+
  scale_color_manual(breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT -Fe", "LT +Fe", "HT -Fe", "HT +Fe"),
                      values = c('purple', 'black', 'black','darkorange2', "darkorange2")) +
  scale_shape_manual(breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT -Fe", "LT +Fe", "HT -Fe", "HT +Fe"),
                      values = c(1, 1, 19,1, 19)) +
  theme_bw()+
  theme(legend.title = element_blank(),
      #  legend.position = "top",
        #legend.justification = "left",
        legend.text = element_text(size = 10), 
        legend.margin  = unit(c(0, 0, 0, 0), "cm"))
legendA

legendA <- get_legend(legendA)
fig2_legendA <- as_ggplot(legendA)
fig2_legendA

###

legendB <- ggplot(nmdsplotdata, 
       aes(x=NMDS1, 
       y=NMDS2,
       shape = bioassay))+
  geom_point(size = 4, alpha =0.65, stroke =1.5)+
  scale_shape_manual(breaks = c("BA1", "BA2"),
                      labels = c("BA1", "BA2"),
                      values = c(1, 2)) +
  theme_bw()+
  theme(legend.title = element_blank(),
        legend.text = element_text(size = 10), 
        #legend.position = "top",
        #legend.justification = "left",
        legend.margin  = unit(c(0, 0, 0, 0), "cm"))
 
legendB <- get_legend(legendB)
fig2_legendB <- as_ggplot(legendB)
fig2_legendB

###
#fig2legend <- ggdraw (legendB) + draw_plot (legendA, x = 0.32 , y = 0) 


fig2legend <- ggdraw (legendA) + draw_plot (legendB, x = -0.1, y = 0.13) 
fig2legend
```

Figure 2 Plot : 2023-03-13
```{r}
# nmds <- plot_grid (peptide_nmds, NULL, fc_nmds, NULL, pigment_nmds, fig2legend,
#                   ncol=6, 
#                   labels = c("A", "", "B", "","C", ""), 
#                   rel_widths = c(1, 0.03, 1, 0.03, 1, 0.3), 
#                   rel_heights = c(1,1,1,1,1, 0.3))
# 
# nmds <- plot_grid (nmds, NULL, 
#            ncol = 1, 
#           rel_heights = c(0.5,1))

nmds <- plot_grid (pigment_nmds, fc_nmds, peptide_nmds,   NULL,
                  ncol=1, 
                  rel_heights = c(1,1,1,0.12),
                  labels = c("A", "B","C", ""))



# plot_grid(nmds, fig2legend, 
#           ncol = 2, 
#           rel_widths = c(1,0.07), 
#           rel_heights = c(1,0.001))
 
heatmaps <- plot_grid(fig2_pigmentmeans, fig2_proteintopmeans, fig2_proteinbottommeans, 
          ncol=1,
          labels = c("D","E", ""), 
          align = "v", 
          rel_heights = c(1.25,0.65,1.1))


groups <-  plot_grid(NULL, pennategroup, centricgroup, phaeogroup,
            align='v',
            ncol =1, 
            rel_heights = c(0.05, 1,1,1.25))#                    labels = c("F", "G", "H"))

y.grob <- textGrob("Fraction of total protein abundance", 
                   gp=gpar( fontsize=13, fontface="bold"), rot=90) #fontface="bold",
allgroups <- grid.arrange(arrangeGrob(groups, left = y.grob))
labelz <- plot_grid (NULL, NULL, NULL, ncol=1, 
                    labels = c("F", "G", "H"))

allgroup_labeled <- plot_grid (labelz, allgroups,
           ncol = 2, rel_widths =c(0.05, 1))
allgroup_labeled

# #png("ps117_figure2_highres_20230420b.png", width=12, height=13, units="in", res=900) 
# ggdraw (nmds) + draw_plot (heatmaps, x = 0, y = -0.01, width = 0.55, height = 0.66) + 
#                 draw_plot (allgroup_labeled, x = 0.56, y = 0.0, width = 0.43, height = 0.65)  
#dev.off()  
  

#png("ps117_figure2_highres_20230421.png", width=20, height=10, units="in", res=900) 
fig2 <- plot_grid (nmds, NULL, heatmaps, NULL, allgroup_labeled, 
            ncol = 5,
            rel_widths = c(0.7, 0.05, 1, 0.05, 0.7),
            rel_heights =  c(1,1,1,1,1),
            axis = "bt",
            align = "vh")
#dev.off()

png("ps117_figure2_highres_20230423.png", width=20, height=10, units="in", res=900) 
ggdraw(fig2) + draw_plot (fig2_legendB, x = 0.295, y = 0.45, width = 0.0001, height = 0.001) +
               draw_plot (fig2_legendA, x = 0.3, y = 0.35, width = 0.0001, height = 0.001) 

              #draw_plot (fig2_legendB, x = 0.05, y = 0.97, width = 0.0001, height = 0.001) +
              # draw_plot (fig2_legendA, x = 0.05, y = 0.6, width = 0.0001, height = 0.001) 
                
dev.off()
```

Figure 2 - Individual replicates - supplement ? 
```{r}
allgroupshm$treatmentrep <- paste (allgroupshm$temperature_treatment, allgroupshm$iron_treatment, allgroupshm$replicate, sep = "_")


treatmentrep_order <- c('T0_T0_A', 'T0_T0_B', 'T0_T0_C', 'T0_T0_D',
                     'LT_noFe_A', 'LT_noFe_B','LT_noFe_C',
                     'LT_Fe_A', 'LT_Fe_B', 'LT_Fe_C', 
                     'HT_noFe_A', 'HT_noFe_B','HT_noFe_C',
                     'HT_Fe_A', 'HT_Fe_B', 'HT_Fe_C')

  allgroupstop <- filter (allgroupshm, grepl("Bacillar|Other|Hapto|Bacteria", class))

ggplot(allgroupstop, aes( x = factor(treatmentrep, level = treatmentrep_order), y = class, fill= abundance))+ 
  geom_tile(aes(fill = norm_abundance),color = "black", lwd = 0.7 ) + 
  #coord_fixed()+ #this makes it all squares 
  scale_fill_gradient2(low = "deepskyblue2", 
                       high = "darkred",
                       mid = "white", 
                       midpoint = 0.25,
                       name = "normalized abundance") +
   ylab ("")+
   xlab ("")+
   theme(axis.text.x=element_text(face = "bold", size = 8, color = "black", vjust = 1,            hjust = 1, angle = 45),
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank(), 
        axis.ticks.y  = element_blank()) +
   guides(fill = guide_colourbar(barwidth = 1, barheight = 15)) +
  facet_grid(.~ bioassay, scales = "free_x") #free_x removes replicates that aren't there


allgroupsbottom <- filter (allgroupshm, !grepl("Bacillar|Other|Hapto|Bacteria", class))
ggplot(allgroupsbottom, aes( x = factor(treatmentrep, level = treatmentrep_order), y = class, fill= abundance))+   geom_tile(aes(fill = norm_abundance),color = "black", lwd = 0.7 ) + 
  #coord_fixed()+ #this makes it all squares 
  scale_fill_gradient2(low = "deepskyblue2", 
                       high = "darkred",
                       mid = "white", 
                       midpoint = 0.003,
                       name = "normalized abundance") +
   ylab ("")+
   xlab ("")+
   theme(axis.text.x=element_text(face = "bold", size = 8, color = "black", vjust = 1,            hjust = 1, angle = 45),
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank(), 
        axis.ticks.y  = element_blank()) +
   guides(fill = guide_colourbar(barwidth = 1, barheight = 15)) +
  facet_grid(.~ bioassay, scales = "free_x") #free_x removes replicates that aren't there

```



#Figure 3 -  2023-03-29
Metal Plots - use only PON for Figure 3, and the rest in supplement
```{r}
#nonproteindata$PFe57_nM <-  ifelse (nonproteindata$PFe57_nM ==  "0.454", NA, 
                            #nonproteindata$PFe57_nM)

#Fe57POC <- T0_T8_FenoFe (PFe57_nM/(POC_uM),PFe[57]:POC~~(M:M)) + scale_x_discrete(limits=c("T0","noFe","Fe"),labels = c("T0", "T8 (-Fe)", "T8 (+Fe)")) + xlab("") + theme (axis.title.y=element_text(size=14, color = "black", face = "bold"))
#Fe57PON <- T0_T8_FenoFe (PFe57_nM/(PON_uM),PFe[57]:PON~~(mM:M)) + scale_x_discrete(limits=c("T0","noFe","Fe"),labels = c("T0", "T8 (-Fe)", "T8 (+Fe)")) + xlab("") + theme (axis.title.y=element_text(size=14, color = "black", face = "bold"))

MnPOC <- T0_T8_FenoFe (PMn_nM/(POC_uM),PMn:POC~~(mM:M)) #+ scale_x_discrete(limits=c("T0","noFe","Fe"),labels = c("T0", "T8 (-Fe)", "T8 (+Fe)")) + xlab("") #+ theme (axis.title.y=element_text(size=14, color = "black", face = "bold"))
MnPON <- T0_T8_FenoFe (PMn_nM/(PON_uM),PMn:PON~~(mM:M)) #+ scale_x_discrete(limits=c("T0","noFe","Fe"),labels = c("T0", "T8 (-Fe)", "T8 (+Fe)")) #+ xlab("") #+ theme (axis.title.y=element_text(size=14, color = "black", face = "bold"))

CuPOC <- T0_T8_FenoFe (PCu_nM/(POC_uM),PCu:POC~~(mM:M)) #+ scale_x_discrete(limits=c("T0","noFe","Fe"),labels = c("T0", "T8 (-Fe)", "T8 (+Fe)")) + xlab("")+ theme (axis.title.y=element_text(size=14, color = "black", face = "bold"))
CuPON <- T0_T8_FenoFe (PCu_nM/(PON_uM),PCu:PON~~(mM:M)) #+ scale_x_discrete(limits=c("T0","noFe","Fe"),labels = c("T0", "T8 (-Fe)", "T8 (+Fe)")) + xlab("")+ theme (axis.title.y=element_text(size=14, color = "black", face = "bold"))

ZnPOC <- T0_T8_FenoFe (PZn_nM/(POC_uM),PZn:POC~~(mM:M)) #+ scale_x_discrete(limits=c("T0","noFe","Fe"),labels = c("T0", "T8 (-Fe)", "T8 (+Fe)")) + xlab("")+ theme (axis.title.y=element_text(size=14, color = "black", face = "bold"))
ZnPON <- T0_T8_FenoFe (PZn_nM/(PON_uM),PZn:PON~~(mM:M)) #+ scale_x_discrete(limits=c("T0","noFe","Fe"),labels = c("T0", "T8 (-Fe)", "T8 (+Fe)")) + xlab("")+ theme (axis.title.y=element_text(size=14, color = "black", face = "bold"))
#metals_Co <- fig3_metals (PCo_nM, PCo:PON~(M:M))
#metals_Ni <- fig3_metals (PNi_nM, PNi:PON)
```


Massage DE and fold change analysis outputs - 2023-03-28
```{r}
#Read in the DE analyses for BA1 and B2, and combine
BA1 <- read.csv("ps117_BA1_clusterDE_20230328.csv", header = T)
colnames(BA1)[3:14] <- paste("BA1", colnames(BA1[,c(3:14)]), sep = "_")

BA2 <- read.csv("ps117_BA2_clusterDE_20230328.csv", header = T)
colnames(BA2)[3:14] <- paste("BA2", colnames(BA2[,c(3:14)]), sep = "_")

BA_DE <- merge (BA1, BA2, by = c("cluster", "taxon_F"), all = TRUE)
   
#pick the three groups I'm interested in, and change the names of clusters to include the names I'm interested in.                     
BA_DE <- filter (BA_DE, grepl ("centric|pennate|Phaeocyst", taxon_F))


#this will be the end all be all cluster annotations. 
#This is based on consensus annotations as well as manual blasting, checking and double checking
BA_DE$cluster_annotation <- ifelse (BA_DE$cluster ==  "clust_1814", "ISIP-1", 
                            ifelse (BA_DE$cluster ==  "clust_343", "ISIP-2A", 
                            ifelse (BA_DE$cluster ==  "clust_1154", "ISIP-3", 
                                    
                            ifelse (BA_DE$cluster ==  "clust_1820", "Plastocyanin", 
                            ifelse (BA_DE$cluster ==  "clust_534", "Flavodoxin", 
                            ifelse (BA_DE$cluster ==  "clust_231", "Rhodopsin",
                                                     
                            ifelse (BA_DE$cluster ==  "clust_332", "LHC", 
                            ifelse (BA_DE$cluster ==  "clust_3986", "LHC", #double check
                            ifelse (BA_DE$cluster ==  "clust_5925", "LHC", 
                            ifelse (BA_DE$cluster ==  "clust_1044", "LHC",
                            ifelse (BA_DE$cluster ==  "clust_195", "LHC",
                            ifelse (BA_DE$cluster ==  "clust_1236", "LHC",
                            ifelse (BA_DE$cluster ==  "clust_3440", "LHC",
                            ifelse (BA_DE$cluster ==  "clust_3658", "LHC",
                            ifelse (BA_DE$cluster ==  "clust_3703", "LHC",
                                    
                            ifelse (BA_DE$cluster ==  "clust_55", "PSI PsaA/PsaB", 
                            ifelse (BA_DE$cluster ==  "clust_2815", "PSI RC",
                                    
                            ifelse (BA_DE$cluster ==  "clust_1906", "Mn-cluster",
                            ifelse (BA_DE$cluster ==  "clust_2763", "Mn-cluster",
                            ifelse (BA_DE$cluster ==  "clust_613", "Cyt-b6",
                            ifelse (BA_DE$cluster ==  "clust_1497", "Cyt-f",
                            ifelse (BA_DE$cluster ==  "clust_42", "PSII RC",
                            ifelse (BA_DE$cluster ==  "clust_1978", "PSII PsbU",
                            ifelse (BA_DE$cluster ==  "clust_498", "PSII CP47",
                            
                            ifelse (BA_DE$cluster ==  "clust_2490", "Rubisco",
                            ifelse (BA_DE$cluster ==  "clust_655", "CA_clust_655",
                            ifelse (BA_DE$cluster ==  "clust_6573", "CA_clust_6573", 
                            ifelse (BA_DE$cluster ==  "clust_1525", "SLC-4",
                            ifelse (BA_DE$cluster ==  "clust_1886", "SLC",

                            ifelse (BA_DE$cluster ==  "clust_411", "Nitrate reductase",        
                            ifelse (BA_DE$cluster ==  "clust_417", "AMT", 
                            ifelse (BA_DE$cluster ==  "clust_891", "Glutamate Synthase", 
                            ifelse (BA_DE$cluster ==  "clust_1372", "Urea transporter",
                            ifelse (BA_DE$cluster ==  "clust_320", "Urea carboxylase",
                            ifelse (BA_DE$cluster ==  "clust_6019", "Ornithine cyclodeaminase",   
                                    
                            ifelse (BA_DE$cluster ==  "clust_5562", "Cu-Zn SOD", 
                            ifelse (BA_DE$cluster ==  "clust_2501", "Fe-Mn SOD", 
                                    
                            #ifelse (BA_DE$cluster ==  "clust_35", "Ribosome recycling",

                           # ifelse (BA_DE$cluster ==  "clust_812", "ZIP",
                            #ifelse (BA_DE$cluster ==  "clust_16168", "TonB",
                            #ifelse (BA_DE$cluster ==  "clust_10646", "TonB_2",
                            ifelse (BA_DE$cluster ==  "clust_2244", "PEPC",
                           # ifelse (BA_DE$cluster ==  "clust_1830", "Acireductone dioxygenase Fe containing",
                            ifelse (BA_DE$cluster ==  "clust_2088", "Fasciclin",
                  
                            ifelse (BA_DE$cluster ==  "clust_808", "Ferredoxin", 
                            ifelse (BA_DE$cluster ==  "clust_2934", "Multicopper Oxidase", 
                                    
                            #highly DE clusters
                            ifelse (BA_DE$cluster ==  "clust_1289", "DoxX",
                           # ifelse (BA_DE$cluster ==  "clust_3468", "Ukn",
                            ifelse (BA_DE$cluster ==  "clust_753", "Fructose bisphosphatase",
                            ifelse (BA_DE$cluster ==  "clust_3563", "Nitrite/Sulphite reductase",
                            ifelse (BA_DE$cluster ==  "clust_997", "APRT",
                            ifelse (BA_DE$cluster ==  "clust_10365", "PET8",
                            ifelse (BA_DE$cluster ==  "clust_1550", "Cyt-b559",
                            ifelse (BA_DE$cluster ==  "clust_1322", "Porphobilinogen deaminase",  
                                          BA_DE$cluster 
                    ))))))))))))))))))))))))))))))))))))))))))))))))

#pfam_PF00006_ATP synthase alpha/beta family, nucleotide-binding domain ?
#proteasome 
# porins - clustclust_3629

protein <- filter (proteindata_unnorm, grepl ("clust_110|clust_11709", uniq_cluster))


clusters <- proteindata_unnorm [c(4, 37, 40:69)]

clusters2 <- melt(clusters, id.vars=c("uniq_cluster", "F")) %>%
            rename (treatment = variable) %>%
            rename (unnorm_abundance = value ) %>% 
            separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

clusters2$F <- ifelse (clusters2$F  ==  "Polar-centric-Mediophyceae", "Centric diatom", 
                  ifelse (clusters2$F  ==  "Radial-centric-basal-Coscinodiscophyceae", "Centric diatom", 
                  ifelse( clusters2$F  ==  "Raphid-pennate", "Pennate diatom", 
                  ifelse(clusters2$F  ==  "Araphid-pennate", "Pennate diatom", 
                  ifelse(clusters2$F  ==  "Phaeocystaceae", "Phaeocystis",
                  clusters2$F)))))

clusters2_aggregated <- aggregate(clusters2$unnorm_abundance, 
                                  by=list(Category=clusters2$uniq_cluster, clusters2$F, clusters2$bioassay), FUN=sum) %>%
                        rename(unnorm_abundance = x) %>%
                        rename(cluster = Category) %>%
                        rename(taxon_F = Group.2) %>%
                        rename(treatment = Group.3)

clusters3 <- dcast(clusters2_aggregated, taxon_F+cluster~treatment) #%>% mutate_at (3, funs ((. / sum(.))))

clusters3 <- clusters3 %>% #this creates a new column of normalized protein abundances, either normalized to total protein or group normalized. 
  group_by(taxon_F) %>% 
  mutate(BA1_grpnorm = BA1 / sum(BA1)) %>% 
  ungroup()

clusters3 <- clusters3 %>% #this creates a new column of normalized protein abundances, either normalized to total protein or group normalized. 
  group_by(taxon_F) %>% 
  mutate(BA2_grpnorm = BA2 / sum(BA2)) %>% 
  ungroup()
#because some clusters are not found in any taxa, we have NA values in the grpnorm column. (dividing by 0 )

#Step 2: get the DE and fold change for taxon-normalized clusters
BA_DE$taxon_F <- ifelse (BA_DE$taxon_F  ==  "centric", "Centric diatom", 
                  ifelse (BA_DE$taxon_F  ==  "pennate", "Pennate diatom", 
                  ifelse(BA_DE$taxon_F  ==  "Phaeocystaceae", "Phaeocystis",
                  BA_DE$taxon_F)))

alldata <- merge(BA_DE, clusters3, by = c("cluster", "taxon_F"))
alldata <- alldata [c(1,27,2, 30, 31, 3:26)]
```

DE sliding plot
```{r}
cluster_order <- c('LHC', #split into PSI and PSII LHCs
                   
                   "PSII PsbU", 
                   'PSII CP47',
                   'PSII RC',
                   'Cyt-b559',

                   'Mn-cluster', 
                   'Plastocyanin',  
                   'Rhodopsin',
                   'Flavodoxin', 
                   'Ferredoxin', 
                   "Cyt-b6", 
                   'Cyt-f',
                 
                   'PSI RC',
                   'PSI PsaA/PsaB',
                 
                   'ISIP-1',
                   'ISIP-2A', 
                   'ISIP-3',
                   
                   'Fasciclin',
                   
                   'Cu-Zn SOD',
                   'Fe-Mn SOD',
                   'Nitrate reductase',
                   "Nitrite/Sulphite reductase", 
                 
                   'Porphobilinogen deaminase')

BA_DE3 <- filter (alldata, grepl (paste(cluster_order, collapse = "|"), cluster_annotation)) 
BA_DE3$BA1_ironDE <-  ifelse (BA_DE3$BA1_Fe_FDR < 0.05, "*", "")
BA_DE3$BA2_ironDE <-  ifelse (BA_DE3$BA2_Fe_FDR < 0.05, "*", "")

# y= factor(cluster_annotation, level = cluster_order),
DE_slide <- function (effect, significance, title) {
  ggplot (BA_DE3, aes(x= {{effect}}, 
                   #y = reorder (cluster_annotation, {{effect}}), #this is for reordering from positive to negative DE
                   y =  factor(cluster_annotation, level = rev(cluster_order)), #this is for ordering based on the order of things in 'cluster_order'
                   color = taxon_F, 
                   #size = (BA1_grpnorm),
                   shape = {{significance}}))+ 
  geom_point (alpha = 0.7, size = 4) + 
    #scale_y_reverse()+
  # geom_text(aes (label = {{significance}}), show.legend = FALSE, size = 8, nudge_y = 0.2)+
    scale_color_manual(name="",
                       breaks = c("Centric diatom", "Pennate diatom","Phaeocystis"),
                       labels = c("Centric diatoms", "Pennate diatoms", "Phaeocystis"),
                       values = c("black", '#661100', '#117733'))+
    
    scale_shape_manual(name="",
                       breaks = c("*", ""),
                       values = c(19,1), 
                       guide = "none") +
    ylab("")+ 
    xlab(expression (paste("Log"[2]~fold~change)))+
    ggtitle(title)+
    theme_bw()+
    theme(plot.title = element_text(hjust = 0, size = 10, color = "black", face = "bold"))+
    theme(axis.text.x = element_text(size = 10, face = "bold", color = "black"),
          axis.title.x = element_text(size = 10, face = "bold", color = "black"), 
          axis.text.y = element_text(size = 10,face = "bold", color="black"), 
          legend.text = element_text(size = 9, color = "black", face = "bold")) +
          geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )}

 
FeDE_BA1 <- DE_slide (BA1_Fe_logFC, BA1_ironDE, "BA1 Fe Effect" ) + 
  theme (legend.position = c(0.01, 0.97), legend.justification = c(0,0.95), legend.key.size = unit (0.4, "cm")) +
  guides(colour = guide_legend(override.aes = list(size=3))) + 
  theme(legend.margin=margin(t=-0.7,  r=0, b=0, l=0, unit="cm")) +
  xlim(-8,8)

 
FeDE_BA2 <- DE_slide (BA2_Fe_logFC, BA2_ironDE, "BA2 Fe Effect" ) + 
  theme (legend.position = "none", axis.text.y = element_blank()) +
    xlim(-8,8)



# cluster abundance
# ggplot (BA_DE3, aes(x = taxon_F, 
#                    y=  reorder (cluster_annotation, BA1_Fe_logFC),
#                    color = taxon_F, 
#                    size = BA1_grpnorm))+
#    geom_point(shape = "square", alpha = 0.7)+  
#    scale_size_continuous(name = "")+
#      scale_color_manual(name="",
#                        breaks = c("Centric diatom", "Pennate diatom","Phaeocystis"),
#                        labels = c("Centric diatoms", "Pennate diatoms", "Phaeocystis"),
#                        values = c("black", '#661100', '#117733'),
#                       guide = "none") + 
#     theme_bw ()+ 
#      theme(axis.text = element_blank(),
#          axis.title = element_blank(), 
#          axis.ticks.x = element_blank()) +
#    theme (legend.position = "none", 
#           legend.margin=margin(t=-4,  r=0, b=-10, l=0))
 
 
#cluster abundance  
# ggplot (BA_DE3, aes(x = taxon_F, 
#                    y= reorder (cluster_annotation, BA2_Fe_logFC),
#                    color = taxon_F, 
#                    size = BA2_grpnorm))+
#    geom_point(shape = "square", alpha = 0.7)+  
#        scale_color_manual(name="",
#                        breaks = c("Centric diatom", "Pennate diatom","Phaeocystis"),
#                        labels = c("Centric diatoms", "Pennate diatoms", "Phaeocystis"),
#                        values = c("black", '#661100', '#117733'),
#                       guide = "none") + 
#     scale_size_continuous(name = "")+
# 
#    theme_bw ()+ 
#    theme(axis.text = element_blank(),
#          axis.title = element_blank(), 
#          axis.ticks.x = element_blank()) +
#   
#   theme (legend.position = "top", 
#          legend.justification = "left",
#          legend.margin=margin(t=-4,  r=0, b=-10, l=-10))
#           #legend.title = element_text(angle = -90)) 
#  


``` 

Figure 3
```{r}
x <- plot_grid(MnPON, 
          CuPON,
          ZnPON,
          ncol = 1, 
       
          align = "hv")

#FeDE_BA1

y <- plot_grid(FeDE_BA1, FeDE_BA2,
          ncol = 2, 
          rel_widths = c(1,0.8))

plot_grid(x,y, 
          ncol = 2, 
          rel_widths = c(0.8,1.5))

#png("ps117_figure4_highres_20230330.png", width=8.3, height=10, units="in", res=900) 

#dev.off()


plot
```{r}
x <- plot_grid(MnPOC, NA,  MnPON, 
               CuPOC, NA,  CuPON,
               ZnPOC, NA,  ZnPON,
          ncol = 3, 
          labels = c("A", "", "B", 
                     "C" , "", "D", 
                     "E", "", "F"), 
          rel_widths = c(1,0.1,1),
          axis = 'tb',
          align = 'hv')

#png("ps117_figure3_highres_20230329.png", width=14, height=10, units="in", res=900) 
plot_grid (x, vertical_legend, 
           ncol = 2, 
            rel_widths = c(1, 0.1), 
            rel_heights = c(1, 0.5))
#dev.off()
```






```



RA plots - I don't think these will make it into the paper 
```{r}
# make list of labels of interest for RA plot  
label_list <- c("ISIP-1", "ISIP-2A", "ISIP-3", "Fld", "Pcyn", "LHC", "Ukn", "DoxX", "Rhodopsin", "Fe-Mn SOD", "Cu-Zn SOD",  "PSI PsaA/PsaB", "PSII RC, D1", "PSII CP47", "Cyt f", "Cyt b6", "Fasciclin", "Rhodopsin", "OEC_PSBO", "OEC_PSBQ", "CA_clust_655", "CA_clust_6573", "Fructose bisphosphatase", "NO2/SO3 reductase" )


#Step 3: combine FC and abundance 
taxon_F <- filter(alldata, grepl ("Phaeocys|diatom", taxon_F))
taxon_F$newabund <- abs(log2(taxon_F$BA1_grpnorm))

foldchangevsabundance <- function(title, xaxis, yaxis, FDR) {
       ggplot(taxon_F, aes (x = {{xaxis}},y = {{yaxis}}, color = (taxon_F), 
       label = ifelse({{FDR}} <0.05 & cluster_annotation %in% label_list, cluster_annotation, "")))+
    geom_point(size = 2, aes(alpha = {{FDR}} < 0.05)) +
    geom_label_repel (size = 3, max.overlaps = Inf, fontface = "bold", show.legend = F, box.padding   = 1, 
                 point.padding = 0.01, force = 10) +
    scale_alpha_manual(values = c(0.1,0.7), guide = 'none') +
    scale_colour_manual(values = c('black','orange4', "seagreen"), name = "") +
   # scale_x_log10()+
    xlab (expression ("Group-normalized protein abundance")) +
    ylab (expression(Fe-Log[2]~fold~change))+ 
    labs(title = title)+
    theme_bw()+
    theme(axis.title=element_text(size = 14,face = "bold", color = "black"), #axes formatting
          axis.text=element_text(face = "bold", size = 12, color = "black", vjust = 0.5, hjust = 1)) + 
    theme(legend.text = element_text(size = 11, face = "bold"))  
}

foldchangevsabundance ("BA1", BA1_grpnorm, BA1_Fe_logFC, BA1_Fe_FDR) + theme (legend.position = "none",   legend.box.margin=margin(-10,-10,-10,-10), legend.background = element_blank()) + xlim(0, 0.03)



foldchangevsabundance ("BA1", BA1_grpnorm, BA1_Fe_logFC, BA1_Fe_FDR) + theme (legend.position = c(0.01,1), legend.justification = c(0.01,1),   legend.box.margin=margin(-10,-10,-10,-10), legend.background = element_blank())


BA2 <- foldchangevsabundance ("BA2", BA2_grpnorm, BA2_Fe_logFC, BA2_Fe_FDR ) + theme (legend.position = "none")



#png("ps117_figure3_highres_20230328.png", width=10, height=11, units="in", res=900) 

plot_grid(BA1, BA2,
          ncol = 1)

#dev.off()

```


#Figure 5 - 2023-03-30
This includes an exploration of ribosomes and chloroplast proteins
N:P
```{r}
A <- T0_T8_FenoFe( ((1000*PON_uM)/POP_nM), PON:POP~~(M:M)) + scale_x_discrete(limits=c("T0","noFe","Fe"),labels = c("T0", "T8 (-Fe)", "T8 (+Fe)")) + xlab("") + expand_limits(y = 0) + theme(axis.title.y=element_text(size=16, color = "black"))

npdrawdown <- read.csv("LJ_ps117_NO3_PO4_consumption.csv")
npdrawdown$bioassayFe <- paste (npdrawdown$bioassay, npdrawdown$iron_treatment, sep = "_")

npdrawdown$temperature <- ifelse (npdrawdown$bioassay ==  "BA1" & npdrawdown$temperature_treatment ==  "T0",  -0.3, 
                         ifelse (npdrawdown$bioassay ==  "BA1" & npdrawdown$temperature_treatment ==  "HT",  1.7, 
                         ifelse (npdrawdown$bioassay ==  "BA1" & npdrawdown$temperature_treatment ==  "LT",  -0.3, 
                         ifelse (npdrawdown$bioassay ==  "BA2" & npdrawdown$temperature_treatment ==  "T0",  -1.4, 
                         ifelse (npdrawdown$bioassay ==  "BA2" & npdrawdown$temperature_treatment ==  "HT",  1, 
                         ifelse (npdrawdown$bioassay ==  "BA2" & npdrawdown$temperature_treatment ==  "LT",  -1, 
                         "Other"))))))

B <- ggplot((filter(npdrawdown, grepl("8", daycount))) , aes(
    x = iron_treatment,
    y = (total_Ndrawdown/total_Pdrawdown),
    shape = iron_treatment, 
    color = temperature_treatment))+
   stat_summary(fun = mean, geom = "point", show.legend = T, size = 4, stroke = 1.5, alpha = 0.65)+
   stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1, width = 0.1)+
   ylab (expression (NO[3]:PO[4]~drawdown~~(M:M))) + 
  xlab("") +
   facet_wrap (~bioassay)+
   scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("LT", "HT"),
                     values = c('black', 'darkorange2','black')) +
  scale_shape_manual(name = "Iron",
                      breaks = c("Fe","noFe", "T0"), 
                      labels = c("Fe", "noFe", "T0"),
                      values = c(19,1,5)) +
  scale_x_discrete(limits=c("noFe","Fe"))+
  theme_bw()+
  theme(axis.text=element_text(face = "bold", size = 12, color = "black"),
        axis.title.x=element_text(size = 14,face = "bold", color = "black"), 
        axis.title.y=element_text(size=16, color = "black"), 
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 12, face = "bold"),
        legend.position = "none") +
   expand_limits(y = 0)
```

Ribosomes normalized to total protein -  2023-03-30
```{r}
#First make sure that all the ribosomal peptides are annotated as such, and that the groups are properly annotated 
proteindata_unnorm$compartment <- ifelse (proteindata_unnorm$uniq_cluster  ==  "unassigned_cluster", "ribosome", 
                               ifelse (proteindata_unnorm$uniq_grpnorm_compartment == "ribosomal", "ribosome",
                               ifelse (grepl("ribosom|Ribosom", proteindata_unnorm$uniq_cluster_annotation) ,  "ribosome",
                                       proteindata_unnorm$uniq_grpnorm_compartment))) # proteindata_unnorm$uniq_cluster))) #

test <- proteindata_unnorm [c(4, 70, 21, 5,7, 13 )]                           



compartment <- proteindata_unnorm [c(70, 37, 40:69  )]                           

compartment <- melt(compartment, id.vars=c("compartment", "F")) %>%
            rename(treatment = variable) %>%
            rename(unnnorm_abundance = value) 
            #separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

compartment$F <- ifelse (compartment$F ==  "Polar-centric-Mediophyceae", "Centric diatom", 
                 ifelse (compartment$F ==  "Radial-centric-basal-Coscinodiscophyceae", "Centric diatom", 
                 ifelse (compartment$F ==  "Raphid-pennate", "Pennate diatom", 
                 ifelse (compartment$F ==  "Araphid-pennate", "Pennate diatom", 
                 ifelse (compartment$F ==  "Phaeocystaceae", "Phaeocystis",
                                          compartment$F)))))
#notmalize to total protein
compartment_aggregated_total <- aggregate(compartment$unnnorm_abundance, 
                                  by=list(Category=compartment$compartment, compartment$treatment), FUN=sum) %>%
                        rename(unnorm_abundance = x) %>%
                        rename(compartment = Category) %>%
                        rename(treatment = Group.2) 
                       
compartment_aggregated_total <- compartment_aggregated_total %>% 
                          group_by(treatment) %>% 
                          mutate(totalproteinnorm_abundance = unnorm_abundance / sum(unnorm_abundance)) %>% 
                          ungroup()

compartment_aggregated_total <- compartment_aggregated_total %>% 
  separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")





ggplot(filter (compartment_aggregated_total, grepl ("^ribosome$", compartment)),
       aes(x= iron_treatment,
       y= totalproteinnorm_abundance,
       color = temperature_treatment, 
       shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  stat_summary(fun = mean, geom = "point", size = 4, stroke = 1.5, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, linewidth = 1, width = 0.1)+
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("LT", "HT"),
                     values = c('black', 'darkorange2','black')) +
  scale_shape_manual(name = "Iron",
                       breaks = c("Fe","noFe", "T0"), 
                       labels = c("Fe", "noFe", "T0"),
                       values = c(19,1,5)) +
  scale_x_discrete(limits=c("T0","noFe","Fe"), 
                   labels = c("T0", "T8 (-Fe)", "T8 (+Fe)"))+
  xlab(expression ("")) +
  ylab (expression(atop ("Ribosome fraction of", paste ("total protein")))) + 
  theme_bw()+
  theme(axis.text=element_text(face = "bold", size = 12, color = "black"),
        axis.title.x=element_text(size = 14,face = "bold", color = "black"), 
        axis.title.y=element_text(size=16, color = "black"), 
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 12, face = "bold"),
        legend.position = "none") 
```

Ribosomes normalized to taxonomic groups -  2023-03-30
```{r}
#normalize to group 
compartment_aggregated_group  <- aggregate(compartment$unnnorm_abundance, 
                                  by=list(Category=compartment$compartment, compartment$treatment, compartment$F), FUN=sum) %>%
                        rename(unnorm_abundance = x) %>%
                        rename(compartment = Category) %>%
                        rename(treatment = Group.2) %>% 
                        rename(taxon_F = Group.3) 

compartment_aggregated_group  <- compartment_aggregated_group  %>% 
                          group_by(treatment,taxon_F) %>% 
                          mutate(groupnorm_abundance = unnorm_abundance / sum(unnorm_abundance)) %>% 
                          ungroup()

compartment_aggregated_group <- compartment_aggregated_group %>% 
  separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")



grpnorm_compartment <- function (compartments, taxon, yaxislabel) { 
ggplot(filter (compartment_aggregated_group, grepl (compartments, compartment) & grepl (taxon, taxon_F)),
       aes(x= iron_treatment,
       y= groupnorm_abundance,
       color = temperature_treatment, 
       shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  stat_summary(fun = mean, geom = "point", size = 4, stroke = 1.5, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, linewidth = 1, width = 0.1)+
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("LT", "HT"),
                     values = c('black', 'darkorange2','black')) +
  scale_shape_manual(name = "Iron",
                       breaks = c("Fe","noFe", "T0"), 
                       labels = c("Fe", "noFe", "T0"),
                       values = c(19,1,5)) +
  scale_x_discrete(limits=c("T0","noFe","Fe"), 
                   labels = c("T0", "T8 (-Fe)", "T8 (+Fe)"))+
  xlab(expression ("")) +
  ylab (substitute(yaxislabel))+ 

  theme_bw()+
  theme(axis.text=element_text(face = "bold", size = 12, color = "black"),
        axis.title.x=element_text(size = 14,face = "bold", color = "black"), 
        axis.title.y=element_text(size=16, color = "black"), 
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 12, face = "bold"),
        legend.position = "none") 
}

D <- grpnorm_compartment ("^ribosome$", "Pennate di", atop ("Pennate diatom", paste ("ribosomal protein fraction"))) 
E <- grpnorm_compartment ("^ribosome$", "Centric di", atop ("Centric diatom", paste ("ribosomal protein fraction"))) 
F <- grpnorm_compartment ("^ribosome$", "Phaeocystis", atop ("Phaeocystis", paste ("ribosomal protein fraction"))) 


G <- grpnorm_compartment ("^chloro$", "Pennate di", atop ("Pennate diatom", paste ("photosynthetic protein fraction"))) 
H <- grpnorm_compartment ("^chloro$", "Centric di", atop ("Centric diatom", paste ("photosynthetic protein fraction"))) 
I <- grpnorm_compartment ("^chloro$", "Phaeocystis", atop ("Phaeocystis", paste ("photosynthetic protein fraction"))) 


grpnorm_compartment ("^chloro$", "di", atop ("Phaeocystis", paste ("photosynthetic protein fraction"))) 


plot_grid (G,H,I,
                ncol = 1, 
      
           align = "hv")

```
```

Plot
```{r}
fig5_legend <-  ggplot(filter (nonproteindata, grepl ("T0|T8", full_treatment_name) ), 
       aes(x=iron_treatment, 
       y=fvfm,
       color = tempfe_treatment, 
       shape = tempfe_treatment))+
  geom_point(size = 4, alpha =0.65, stroke =1.5)+
  scale_color_manual(breaks = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c('black', 'black', 'black','darkorange2', "darkorange2")) +
  scale_shape_manual(breaks = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c(5, 1, 19,1, 19)) +
  theme_bw()+
  theme(legend.title = element_blank(),
        legend.text = element_text(size = 12, margin = margin(unit = 'cm')), 
        legend.margin = margin(c(0,0,0,0)), 
        legend.spacing.x = unit(0, "mm"),
        legend.spacing.y = unit(0, "mm"))

fig5_legend <- get_legend(fig5_legend)
fig5_legend <- as_ggplot(fig5_legend)
fig5_legend

x <- plot_grid (A, NULL, D,
           B, NULL, E,
           C, NULL, F,
           ncol = 3, 
           labels = c("A", "", "D", 
                      "B", "", "E", 
                      "C", "", "F"),
           rel_widths = c(1,0.08,1), 
           align = "hv")

png("ps117_figure5_highres_20230330.png", width=14, height=11, units="in", res=900) 

plot_grid(x, fig5_legend, 
          rel_widths = c(1,0.08), 
          rel_heights = c(1, 0.5))
dev.off()
```


#Figure 6
```{r}
#Consider not splitting the data into bioassays, the Cd and CA trends become strong. 
T0_T8_FenoFe (PCd_nM/(PON_uM),PCd:POC~~(mM:M))

metals_Cd <- fig3_metals (PCd_nM, PCd:PON)

CA_centric <- protein_clusters ("centric|Bacill", "CA_clust_6573", protein = c("CA_clust_6573"), shape = c(19,15)) + ggtitle ("Centric diatoms - clust_6573") + theme(legend.position = "none") 

CA_pennate <- protein_clusters ("pennate", "CA_clust_655", protein = c("CA_clust_655"), shape = c(19,15)) + ggtitle ("Pennate diatoms - clust_655") + theme(legend.position = "none") 

CA_phaeo <- protein_clusters ("Phaeo", "CA_clust_655", protein = c("CA_clust_655"), shape = c(19,15)) + ggtitle ("Phaeocystis - clust_655") + theme(legend.position = "none")

#protein_clusters ("pennate", "Cation Efflux protein", protein = c("Cation Efflux protein"), shape = c(19,15)) look into efflux proteins

#protein_clusters_hm ("centric|pennate", "Carbonic|CA", "BA1")

#protein_clusters_hm ("centric|pennate", "Carbonic|CA", "BA2")

#Maybe look atr SLC4 ? 
```


plot
```{r}
plot_grid( metals_Cd,  CA_centric,  
           CA_pennate, CA_phaeo, 
          nrow = 4, 
        
          align = 'hv')
         
#and heatmap
```

#metal contribution to proteome
Protein per L 
```{r}
alldata$fulltreatment <- paste (alldata$temperature_treatment, alldata$iron_treatment, sep = "_")

ggplot(filter (alldata, grepl ("T0|T8", full_treatment_name) & grepl ("BA2|", bioassay)), 
       aes(#x=iron_treatment, 
       #x=PCd_nM,
       #x = protein_ug_per_L,
       x = PON_ug_L*4.78,

       y = (PCd_nM*68.76),
          #y = (PCd_nM),
       #x= PCd_nM/PON_uM,
       
       color = tempFe, 
       shape = tempFe))+
       #color = temperature_treatment, 
       #shape = iron_treatment)) + 
  geom_point(size = 4, alpha =0.65, stroke =1.5)+
  
    scale_color_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c('black', 'black', 'black','darkorange2', "darkorange2")) +
  
     scale_shape_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c(5, 1, 19,1, 19)) +
  labs (x = "Total protein (ug/L)", y = "Theortical maximum Cd-CA (ug/L)")+
    theme_bw()+
  
     facet_wrap(~bioassay) +
  
  theme(axis.title = element_text (size = 14, face = "bold"), 
        axis.text = element_text(size = 12, color = 'black'), 
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 10, margin = margin(r = 0.5, unit = 'cm')), 
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 14, face = "bold"))
     
 ggplot(filter (alldata, grepl ("T0|T8", full_treatment_name) & grepl ("BA2|", bioassay)), 
       aes(#x=iron_treatment, 
       #x=PCd_nM,
       #x = protein_ug_per_L,
       x = factor(tempFe, level = tempFe_order),

       y = (PCd_nM*68.76)/(PON_ug_L*4.78),
          #y = (PCd_nM),
       #x= PCd_nM/PON_uM,
       
       color = tempFe, 
       shape = tempFe))+
       #color = temperature_treatment, 
       #shape = iron_treatment)) + 
  geom_point(size = 4, alpha =0.65, stroke =1.5)+
  
    scale_color_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c('black', 'black', 'black','darkorange2', "darkorange2")) +
  
     scale_shape_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c(5, 1, 19,1, 19)) +
  labs (x = "", y = "Theortical maximum Cd-CA abundance:protein (ug:ug)")+
    theme_bw()+
  
     facet_wrap(~bioassay) +
  scale_y_log10()+
  
  theme(axis.title = element_text (size = 14, face = "bold"), 
        axis.text = element_text(size = 12, color = 'black'), 
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 10, margin = margin(r = 0.5, unit = 'cm')), 
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 14, face = "bold"))

cdca <- filter(norm_proteomicsdata, grepl("clust_6573", cluster))
cdca2 <- cdca [c(3, 24:53)]

cdca3 <- melt(cdca2, id.vars=c("cluster")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value ) %>% 
            separate(treatment, into =  c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"))  
 
cdca3$tempFe <- paste (cdca3$temperature_treatment, cdca3$iron_treatment, sep = "_")
  
cdca4 <- aggregate(cdca3$norm_abundance, by=list(cdca3$bioassay, cdca3$tempFe, cdca3$replicate), FUN = sum) %>%
          rename (bioassay = Group.1, 
          tempFe = Group.2, 
          replicate = Group.3, 
          cdca_norm_abundance = x)

alldata2 <- merge (alldata, cdca4, by = c("bioassay", "tempFe", "replicate"), all=TRUE)

#write.csv(alldata2, "testtest.csv")

ggplot(filter (alldata2, grepl ("T0|T8", full_treatment_name) & grepl ("BA2|", bioassay)), 
       aes(#x=PCd_nM,
       #x = protein_ug_per_L,
       #x = factor(tempFe, level = tempFe_order),
        x = cdca_norm_abundance, 
       y = (PCd_nM*68.76)/(PON_ug_L*4.78),
          #y = (PCd_nM),
       color = tempFe, 
       shape = tempFe))+
       #color = temperature_treatment, 
       #shape = iron_treatment)) + 
  geom_point(size = 4, alpha =0.65, stroke =1.5)+
  
    scale_color_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c('black', 'black', 'black','darkorange2', "darkorange2")) +
  
     scale_shape_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c(5, 1, 19,1, 19)) +
  labs (x = "Cd-CA:total peptide abuncance", y = "theoretical maximum Cd-CA : total protein " )+
    theme_bw()+
  
     facet_wrap(~bioassay) +
 # scale_y_log10()+
  
  theme(axis.title = element_text (size = 14, face = "bold"), 
        axis.text = element_text(size = 12, color = 'black'), 
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 10, margin = margin(r = 0.5, unit = 'cm')), 
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 14, face = "bold"))



#I was thinking just to calculate the proportion of the metal that can be accounted for by protein.

#step 1: Calculate moles of pCd from mass of Cd-CA
#each kg of Cd-CA = 0.0148 moles


#1 ug of cdca = 1.454 e-11 moles

#calculate ug/L of cdca: cdca_norm_abundance*PON_ug_L*4.78)
#calculate moles/L of cdca from ug/L : 1.454e-11 *(cdca_norm_abundance*PON_ug_L*4.78)

#calculate nM pCd from moles/L cd-ca 1e9 *(1.454e-11 *(cdca_norm_abundance*PON_ug_L*4.78))

ggplot(filter (alldata2, grepl ("T0|T8", full_treatment_name) & grepl ("BA2|", bioassay)), 
       aes(#x=PCd_nM,
       #x = PCd_nM,
      # x = factor(tempFe, level = tempFe_order),
       x = bioassay, 
       #y = (PCd_nM*68.76)/(PON_ug_L*4.78),
         y = (1e9 *(1.454e-11 *(cdca_norm_abundance*PON_ug_L*4.78)))/PCd_nM*100))+
       #color = tempFe, 
       #shape = tempFe))+
  geom_point(size = 4, alpha =0.1)+
   stat_summary(fun = mean, geom = "point", size = 4)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1)+
  
  
    scale_color_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c('black', 'black', 'black','darkorange2', "darkorange2")) +
  
     scale_shape_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c(5, 1, 19,1, 19)) +
  labs (x = "", y = "Percent of pCd accounted for by Cd-CA" )+
    theme_bw()+
       #facet_wrap(~bioassay) +
  theme(axis.title = element_text (size = 14, face = "bold"), 
        axis.text = element_text(size = 12, color = 'black'), 
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 10, margin = margin(r = 0.5, unit = 'cm')), 
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 14, face = "bold"))
```

Mn cluster
```{r}
#clust_5856
#clust_1906 - psbO
#clust_2763 - psbQ
oec <- filter(norm_proteomicsdata, grepl("clust_1906", cluster))
oec2 <- oec [c(3, 24:53)]

oec3 <- melt(oec2, id.vars=c("cluster")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value ) %>% 
            separate(treatment, into =  c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"))  
 
oec3$tempFe <- paste (oec3$temperature_treatment, oec3$iron_treatment, sep = "_")
  
oec4 <- aggregate(oec3$norm_abundance, by=list(oec3$bioassay, oec3$tempFe, oec3$replicate, oec3$cluster), FUN = sum) %>%
          rename (bioassay = Group.1, 
          tempFe = Group.2, 
          replicate = Group.3, 
          oec_norm_abundance = x)

oec4$normabunancadjusted <- ifelse (oec4$Group.4 ==  "clust_1906", oec4$oec_norm_abundance*3.0303e-11, 
                                    ifelse(oec4$Group.4 == "clust_2763", oec4$oec_norm_abundance*6.6667e-11, 
                                           ifelse(oec4$Group.4 == "clust_5856", oec4$oec_norm_abundance*4.347e-11, 
                                         "else")))
                                    
alldata2 <- merge (alldata, oec4, by = c("bioassay", "tempFe", "replicate"), all=TRUE)


#I was thinking just to calculate the proportion of the metal that can be accounted for by protein.
#Calculate moles of pMn from mass of Mn-cluster

#each kg of OEC-cluster = 0.0303moles

#1 ug of oec = 3.0303 e-11 moles

#calculate ug/L of oec: oec_norm_abundance*PON_ug_L*4.78)
#calculate moles/L of oec from ug/L : 3.0303e-11 *(oec_norm_abundance*PON_ug_L*4.78)

#calculate nM pMn from moles/L cd-ca 1e9 *(3.0303e-11 *(oec_norm_abundance*PON_ug_L*4.78))

ggplot(filter (alldata2, grepl ("T0|T8", full_treatment_name) & grepl ("BA2|", bioassay)), 
       aes(#x=PCd_nM,
        x = factor(tempFe, level = tempFe_order),
       #x = bioassay, 
       #y = (PCd_nM*68.76)/(PON_ug_L*4.78),
         y = (4*1e9* as.numeric(normabunancadjusted)*PON_ug_L*4.78 )/PMn_nM*100,
       # y = (4*1e9 *((as.numeric(normabunancadjusted)*protein_ug_per_L)))/PMn_nM*100,
       color = tempFe, 
       shape = tempFe))+
  #geom_point(size = 4, alpha =0.2)+
   stat_summary(fun = mean, geom = "point", size = 4)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1)+
    scale_color_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c('black', 'black', 'black','darkorange2', "darkorange2")) +
  
     scale_shape_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c(5, 1, 19,1, 19)) +
  labs (x = "", y = "Percent of pMn accounted for by Mn-cluster" )+
    theme_bw()+
  #scale_y_log10()+
       facet_wrap(~bioassay) +
  theme(axis.title = element_text (size = 14, face = "bold"), 
        axis.text = element_text(size = 12, color = 'black'), 
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 10, margin = margin(r = 0.5, unit = 'cm')), 
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 14, face = "bold"))
```

Mn cluster
```{r}

#clust_1906 - psbO
#clust_2763 - psbQ
oec <- filter(norm_proteomicsdata, grepl("clust_1906", cluster))
oec2 <- oec [c(3, 24:53)]

oec3 <- melt(oec2, id.vars=c("cluster")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value ) %>% 
            separate(treatment, into =  c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"))  
 
oec3$tempFe <- paste (oec3$temperature_treatment, oec3$iron_treatment, sep = "_")
  

oec4 <- aggregate(oec3$norm_abundance, by=list(oec3$bioassay, oec3$tempFe, oec3$replicate), FUN = sum) %>%
          rename (bioassay = Group.1, 
          tempFe = Group.2, 
          replicate = Group.3, 
          oec_norm_abundance = x)

alldata2 <- merge (alldata, oec4, by = c("bioassay", "tempFe", "replicate"), all=TRUE)

#I was thinking just to calculate the proportion of the metal that can be accounted for by protein.
#Calculate moles of pMn from mass of Mn-cluster

#each kg of OEC-cluster = 0.0303moles

#1 ug of oec = 3.0303 e-11 moles

#calculate ug/L of oec: oec_norm_abundance*PON_ug_L*4.78)
#calculate moles/L of oec from ug/L : 3.0303e-11 *(oec_norm_abundance*PON_ug_L*4.78)

#calculate nM pMn from moles/L cd-ca 1e9 *(3.0303e-11 *(oec_norm_abundance*PON_ug_L*4.78))

ggplot(filter (alldata2, grepl ("T0|T8", full_treatment_name) & grepl ("BA2|", bioassay)), 
       aes(#x=PCd_nM,
        x = factor(tempFe, level = tempFe_order),
       #x = bioassay, 
       #y = (PCd_nM*68.76)/(PON_ug_L*4.78),
         y = (4 * 1e9 * 3.0303e-11 * oec_norm_abundance * PON_ug_L * 4.78)/PMn_nM*100,
       color = tempFe, 
       shape = tempFe))+
  #geom_point(size = 4, alpha =0.2)+
   stat_summary(fun = mean, geom = "point", size = 4)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1)+
    scale_color_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c('black', 'black', 'black','darkorange2', "darkorange2")) +
  
     scale_shape_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c(5, 1, 19,1, 19)) +
  labs (x = "", y = "Percent of pMn accounted for by Mn-cluster" )+
    theme_bw()+
       facet_wrap(~bioassay) +
  theme(axis.title = element_text (size = 14, face = "bold"), 
        axis.text = element_text(size = 12, color = 'black'), 
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 10, margin = margin(r = 0.5, unit = 'cm')), 
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 14, face = "bold"))
```

SOD cluster
```{r}
sod <- filter(norm_proteomicsdata, grepl("^clust_2501$", cluster))
sod2 <- sod [c(3, 24:53)]

sod3 <- melt(sod2, id.vars=c("cluster")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value ) %>% 
            separate(treatment, into =  c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"))  
 
sod3$tempFe <- paste (sod3$temperature_treatment, sod3$iron_treatment, sep = "_")

sod4 <- aggregate(sod3$norm_abundance, by=list(sod3$bioassay, sod3$tempFe, sod3$replicate), FUN = sum) %>%
          rename (bioassay = Group.1, 
          tempFe = Group.2, 
          replicate = Group.3, 
          sod_norm_abundance = x)

alldata2 <- merge (alldata, sod4, by = c("bioassay", "tempFe", "replicate"), all=TRUE)


#I was thinking just to calculate the proportion of the metal that can be accounted for by protein.
#Calculate moles of pMn from mass of Mn-cluster

#each kg of sod-cluster = 0.0303moles

#1 ug of sod = 3.0303 e-11 moles

#calculate ug/L of sod: sod_norm_abundance*PON_ug_L*4.78)
#calculate moles/L of sod from ug/L : 3.0303e-11 *(sod_norm_abundance*PON_ug_L*4.78)

#calculate nM pMn from moles/L cd-ca 1e9 *(3.0303e-11 *(sod_norm_abundance*PON_ug_L*4.78))

ggplot(filter (alldata2, grepl ("T0|T8", full_treatment_name) & grepl ("BA2|", bioassay)), 
       aes(#x=PCd_nM,
       #x = PCd_nM,
       x = factor(tempFe, level = tempFe_order),
       #x = bioassay, 
       #y = (PCd_nM*68.76)/(PON_ug_L*4.78),
         y = (1e9 *(4e-11 *(sod_norm_abundance*PON_ug_L*4.78)))/PMn_nM*100,
       color = tempFe, 
       shape = tempFe))+
  geom_point(size = 4, alpha =0.2)+
   stat_summary(fun = mean, geom = "point", size = 4)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1)+
    scale_color_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c('black', 'black', 'black','darkorange2', "darkorange2")) +
  
     scale_shape_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c(5, 1, 19,1, 19)) +
  labs (x = "", y = "Percent of pMn accounted for by Fe-Mn SOD" )+
    theme_bw()+
       facet_wrap(~bioassay) +
  theme(axis.title = element_text (size = 14, face = "bold"), 
        axis.text = element_text(size = 12, color = 'black'), 
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 10, margin = margin(r = 0.5, unit = 'cm')), 
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 14, face = "bold"))
```








#Other things to include
fasciclin (supplement ? )
```{r}
fas <- read.table ("fasciclin_environmental_parameters.csv")

```

Z-score for things ? 
```{r}
silicon <- filter(norm_proteomicsdata, grepl('clust_202 |clust_19688',cluster)) clust_202 |clust_13423|clust_15873

#proteases 2022-04-28
#https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1326381/
#https://pubmed.ncbi.nlm.nih.gov/10849347/
#https://academic.oup.com/plcell/article/12/3/419/6008767
#some proteases function to stabalize the Mn cluster or keep the Mn cluster good and going. So if we have an increase in Mn cluster proteins, we'll need more proteases to maintain these proteins. 
```




#Misc 
proteins (see chucnk with same name below without a function) 2022-08-30
```{r}

protein_clusters ("centric|pennate|Bacillariop", "Plastocyanin", protein = c("Plastocyanin", "Fe-Mn SOD"), shape = c(1,4))

protein_clusters ("centric|pennate|Bacillariop", "", "Fe-Mn|Mn-clus" )

protein_clusters ("centric|pennate|Bacillariop", "", "ISIP" ) +geom_point(size =2, position= position_dodge(width=0.3) )
Mnproteins_phaeo <- protein_clusters ("Phaeo|Prymn", "", "Fe-MnSOD|Mn-clus")

Cuproteins_diatoms <- protein_clusters ("centric|pennate|Bacillariop", "", "Plastocy|Cu-Zn" )
Cuproteins_phaeo <- protein_clusters ("Phaeo|Prymn", "", "Plastocy|Cu-Zn")

Znproteins_diatoms <- protein_clusters ("centric|pennate|Bacillariop", "", "ZIP|RNA" )
Znproteins_phaeo <- protein_clusters ("Phaeo|Prymn", "", "ZIP|RNA")

Cdproteins_diatoms <- protein_clusters ("centric|pennate|Bacillariop", "", "CA" )
Cdproteins_phaeo <- protein_clusters ("Phaeo|Prymn", "", "CA")

```

legend
```{r}
x<- c(1,2,3)
y <- c("a", "b", "c")
colr <- c (-1.2,0, 1.7)
z <- data.frame(x,y, colr)

legend3b <- ggplot(z, aes (x,y, fill = colr))+
   geom_point()+
    scale_fill_gradientn(colors = c("deepskyblue2","white","darkred"),
         values = rescale(c(-1.2,0,1.7)),
         guide = "colorbar", limits=c(-1.2,1.7),
         name = "Row Z-score")+
   theme(legend.text = element_text(face = "bold", size=13),
        legend.title = element_text (face = "bold", size = 13, angle = 90), 
        legend.title.align=0.5)+
   guides(fill = guide_colourbar(title.position = "left", barwidth = 1.5, barheight = 15))
  

legend3b <- get_legend(legend3b)
legend3b <- as_ggplot(legend3b)
```

proteins heatmap
```{r}
protein_clusters_hm ("pennate", "Fe-Mn|Mn-clus|Anhyd", "BA" )
protein_clusters ("Phaeo|Prymn", "", "Fe-Mn|Mn-clus|Anhyd", "BA2")

Cuproteins_centric <- protein_clusters ("centric", "Plastocy|Cu-Zn", "BA" )
Cuproteins_pennate <- protein_clusters ("pennate", "Plastocy|Cu-Zn", "BA" )
Cuproteins_phaeo <- protein_clusters ("Phaeo|Prymn", "Plastocy|Cu-Zn", "BA")

Znproteins_diatoms <- protein_clusters ("centric|pennate|Bacillariop", "", "ZIP|RNA" )
Znproteins_phaeo <- protein_clusters ("Phaeo|Prymn", "", "ZIP|RNA")

Cdproteins_diatoms <- protein_clusters ("centric|pennate|Bacillariop", "", "CA" )
Cdproteins_phaeo <- protein_clusters ("Phaeo|Prymn", "", "CA")

```

plot
```{r}
row1 <- plot_grid(NULL, metals_Mn, NULL, Mnproteins_diatoms, NULL, Mnproteins_phaeo, 
          nrow = 1,
          rel_widths = c(0.05, 1, 0.1, 1.3, 0.05, 1.3),
          labels = c("", "A", "", "B", "", "C"), 
          align = 'h')
row2 <- plot_grid(NULL, metals_Cu, NULL, Cuproteins_diatoms, NULL, Cuproteins_phaeo, 
          nrow = 1,
          rel_widths = c(0.05, 1, 0.1, 1.3, 0.05, 1.3),
          labels = c("", "D", "", "E", "", "F"), 
          align = 'h')
row3 <- plot_grid(NULL, metals_Zn, NULL, Znproteins_diatoms, NULL, Znproteins_phaeo, 
          nrow = 1,
          rel_widths = c(0.05, 1, 0.1, 1.3, 0.05, 1.3),
          labels = c("", "G", "", "H", "", "I"), 
          align = 'h')

plot_grid(row1, row2, row3, 
          nrow = 3)
```

plot_v2
```{r}
row1 <- plot_grid(NULL, metals_Mn, NULL, Mnproteins_diatoms, NULL, Mnproteins_phaeo, 
          nrow = 1,
          rel_widths = c(0.05, 1, 0.1, 1.3, 0.05, 1.3),
          labels = c("", "A", "", "B", "", "C"), 
          align = 'h')

plot_grid(metals_Cu, Cuproteins_diatoms, Cuproteins_phaeo, 
          nrow = 3, align = 'hv', 
          rel_heights = c(1,0.5,0.5), 
          axis = 'l')


           
row3 <- plot_grid(NULL, metals_Zn, NULL, Znproteins_diatoms, NULL, Znproteins_phaeo, 
          nrow = 1,
          rel_widths = c(0.05, 1, 0.1, 1.3, 0.05, 1.3),
          labels = c("", "G", "", "H", "", "I"), 
          align = 'h')


plot_grid(row1, row2, row3, 
          nrow = 3)
```








#Supplemental plots 
macronutrients 
```{r}
x_vs_time (alldata, daycount, (Si_uM), Si~mu*M)

x_vs_time (alldata, daycount, (PON_uM), PON~mu*M)

x_vs_time (alldata, daycount, (POC_uM), POC~mu*M)

x_vs_time (alldata, daycount, (POP_nM), POP~nM)

x_vs_time (alldata, daycount, (total_cells_mL), POP~nM)
```

metals
```{r}
x_vs_time (alldata, daycount, (DMn_nM), DFe57~n*M) +
  expand_limits(y = 0:0.5) 

x_vs_time (alldata, daycount, (PFe57_nM), PFe57~n*M) +
  expand_limits(y = 0:0.5) 

x_vs_treatment (alldata, temperature_treatment, temptreatment_order,  ((POC_uM)/((PAl_nM))), PFe[total]:POC~(mol:mol))+
  geom_point(size = 3)+
   #expand_limits(y = 0:0.00002) +
  scale_y_continuous(labels = function(x) format(x, scientific = FALSE, drop0trailing = TRUE)) #this ensures that the decimal points are exact every time. 
 

#png("Fe56_20220406.png", width=40*0.75, height=30*0.75, units="cm", res=600) 
ggplot(alldata, aes(
                x = temperature_C, #factor (iron_treatment, level = Fetreatment_order),
                y = ((PON_uM)/((POP_nM))),
                color = bioassay,
                shape = iron_treatment)) +
   geom_point(size = 5, alpha=0.5, stroke = 1.8)
 # stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.8, alpha = 0.5)+
 # stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1.3, width = 0.1)+
  
  scale_color_manual(name="Temperature",
                     breaks = c("T0", "LT","HT"),
                     labels = c("T0", "ambient", "warm"),
                     values = c('blue', 'black', 'orange')) +
    scale_shape_manual(name = "Iron",
                     breaks = c("T0", "noFe","Fe"), #this removes the T0 from the legend
                     labels = c("T0", "noFe", "addedFe"),
                     values = c(1, 1, 19)) + #this keeps the T0 formatting that I want
  ylab (expression ("PFe56")) + 
 xlab (expression ("DFe56")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))+
     expand_limits(y = 0) 

#dev.off()
```

cellcounts
```{r}
#png("total_cellcount_vs_time_20220329.png", width=40*0.75, height=30*0.75, units="cm", res=600) 
x_vs_time (total_cells_mL,  Total~cells~(mL^-1)) + scale_y_log10()
#dev.off()

day6 <- filter (nonproteindata, grepl("6", daycount))

Fe_treatment_order <- c('T0', 'noFe', 'Fe')

temp_treatment_order <- c('T0', 'LT', 'HT')

ggplot(day6, aes(
                x = (factor(temperature_treatment, level = temptreatment_order)),
                y = total_cells_mL,
                color = temperature_treatment,
                shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.8, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1.3, width = 0.1)+
  
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('black','darkorange2', 'black')) +
    scale_shape_manual(name = "Iron",
                     breaks = c("noFe","Fe"), #this removes the T0 from the legend
                     labels = c("noFe", "Fe"),
                     values = c(1,19, 1)) + #this keeps the T0 formatting that I want
  #ylab (expression (PFe[total]~~nM)) + 
  xlab (expression ("")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))
         
#dev.off()
```

cellsize
```{r}
ggplot(filter(nonproteindata, !grepl("9", daycount)), aes(x=daycount, 
              # y=cellnumber_normalized_volume,
              #y=  (total_cell_volume_um3_mL/total_cells_mL),
              y=  total_cell_volume_um3_mL,
                color = temperature_treatment, 
                shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  #geom_point(size =3, alpha =0.5)+
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.8, alpha = 0.5)+
  #stat_summary(fun = mean, geom = "line", size = 1.1, aes(linetype = iron_treatment))+
  stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1.3, width = 0.1)+
  
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('black', 'darkorange2', 'black')) +
  
  scale_shape_manual(name = "Iron",
                     breaks = c("noFe","Fe", "T0"), #this removes the T0 from the legend
                     labels = c("noFe", "Fe", "T0"),
                     values = c(1,19, 1)) + #this keeps the T0 formatting that I want
  #scale_y_continuous(expand = c(0,5))+
  ylab (expression ("biovolume normalized to cell count")) + 
  xlab (expression ("Day Count")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))

####
cellsize <- alldata [c(14,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95,97,99,101,103,105)]

meltedsize <- melt(cellsize, id.vars=c("full_treatment_name"))

cellsizemap <- c ("Pico2_cells_mL" = "1.18",
                  "Pico3_cells_mL" = "1.99",
                  "Pico4_cells_mL" = "2.03",
                  "Pico5_cells_mL" = "2.53",
                  "Pico6_cells_mL" = "3",
                  "Pico7_cells_mL" = "3",
                  "Nano1_cells_mL" = "3.23",
                  "Nano2_cells_mL" = "3.71",
                  "Nano3_cells_mL" = "4.31",
                  "Nano4_cells_mL" = "4.93",
                  "Nano7_cells_mL" = "5.17",
                  "Nano8_cells_mL" = "5.52",
                  "Nano9_cells_mL" = "6",
                  "Nano11_cells_mL" = "6.7",
                  "Nano12_cells_mL" = "7.06",
                  "Nano13_cells_mL" = "7.32",
                  "Nano14_cells_mL" = "7.75",
                  "Nano17_cells_mL" = "8.81",
                  "Nano19_cells_mL" = "10.36",
                  "Nano20_cells_mL" = "13.27",
                  "Nano22_cells_mL" = "15.39",
                  "Nano23_cells_mL" = "19",
                  "Nano24_cells_mL" = "19.78")

meltedsize$estimated_size <- cellsizemap [meltedsize$variable]

meltedsize <- meltedsize %>% separate(full_treatment_name, c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate")) 

meltedsize_day6 <- filter(meltedsize, grepl("T6", timepoint))

meltedsize_day6 <- filter(meltedsize_day6, grepl("BA1", bioassay))

ggplot(meltedsize_day6, aes(
  #x = variable,
  x = estimated_size,
  #x=as.numeric(estimated_size), 
                y=value,
              #y=  total_cell_volume_um3_mL,
               # color = temperature_treatment))+ 
                shape = iron_treatment)) + 
  facet_wrap(~temperature_treatment, ncol=1)+
  geom_point(size =3, alpha =0.5)+
  #stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.8, alpha = 0.5)+
  stat_summary(fun = mean, geom = "line", size = 1.1, aes(linetype = iron_treatment))+
  #stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1.3, width = 0.1)+
  
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('darkorange2', 'black', 'black')) +
  scale_shape_manual(name = "Iron",
                     breaks = c("noFe","Fe"), #this removes the T0 from the legend
                     labels = c("noFe", "addedFe"),
                     values = c(1,19, 1)) +
   #this keeps the T0 formatting that I want
  #scale_y_continuous(expand = c(0,5))+
 # ylab (expression ("norm relative size")) + 
  #xlab (expression ("Day Count")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 10, color = "black" , angle = 90),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))


#meltedsize$rounded <- round (meltedsize$value, -2)

meltedsize2 <- subset (meltedsize, rounded <200000 )

#https://aslopubs.onlinelibrary.wiley.com/doi/10.1002/lno.12023 - anderson et al. 

# https://adrienne-marshall.github.io/ggplot2_workshop/
#no big effect of temp or iron on community cell size ? 
```


POC-chla
```{r}
alldata$PONchla <- alldata$PON_ug_L/alldata$total_chla_ug_L


ggplot(filter(alldata, grepl("9|", daycount)), aes(x=daycount, 
                y=  PONchla,
                color = temperature_treatment, 
                shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  #geom_point(size =3, alpha =0.5)+
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.8, alpha = 0.5)+
  #stat_summary(fun = mean, geom = "line", size = 1.1, aes(linetype = iron_treatment))+
  stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1.3, width = 0.1)+
  
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('black', 'darkorange2', 'black')) +
  
  scale_shape_manual(name = "Iron",
                     breaks = c("noFe","Fe", "T0"), #this removes the T0 from the legend
                     labels = c("noFe", "Fe", "T0"),
                     values = c(1,19, 1)) + #this keeps the T0 formatting that I want
  ylab (expression ("PON:Chla (ug:ug)")) + 
  xlab (expression ("Day Count")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))


```





#photophysiology
raw PAM curve fitting
```{r}
setwd("D:/School/PhD/PS117/data/photophysiology/")


rawpam <- read.csv("ps117_rawPAM.csv", header = T)
rawpam <- filter (rawpam, !grepl ("blank", iron_treatment) )

rawpam$fvfm <-  (rowMeans(rawpam[,c(47:72)]) - rowMeans(rawpam[,c(17:32)]))/ rowMeans(rawpam[,c(47:72)])

rawpam2 <- rawpam [c(143,14)]


ggplot(rawpam, aes(x=fvfm, 
                y=YII)) + 
  geom_point(size =3, alpha =0.4, aes(color = temperature_treatment, 
                shape = iron_treatment))+
  geom_abline(intercept = 0, slope = 1, color = "Blue", size = 1, linetype = 2)+
  scale_x_continuous(expand = c(0, 0), limits = c(0, 0.7)) + 
  scale_y_continuous(expand = c(0, 0), limits = c(0, 0.7))+
  #geom_smooth(method='lm', formula= y~x, color = "green")+

    scale_shape_manual(name = "Iron",
                     breaks = c("T0", "noFe","Fe"), #this removes the T0 from the legend
                     labels = c("T0","noFe", "addedFe"),
                     values = c(1, 1, 19)) + #this keeps the T0 formatting that I want
  scale_color_manual(name="Temperature",
                     breaks = c("T0","LT","HT"),
                     labels = c("T0","ambient", "warm"),
                     values = c('black', 'black', "red")) +
  ylab (expression (PAM~~F[v]/F[m])) + 
  xlab (expression (Loay~~F[v]/F[m])) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"), 
        #legend.position = "top", # c(0.04,0.88), 
        legend.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 12))+ 
        
   guides(color = guide_legend(override.aes = list(size= 4)))+
   guides(shape = guide_legend(override.aes = list(size= 4)))


ggplot(rawpam, aes(x=daycount, 
                y=YII,
                color = temperature_treatment, 
                shape = iron_treatment)) + 
  #geom_point(size =5, alpha =0.5)+
  stat_summary(fun = mean, geom = "point", size = 2, stroke = 2, alpha = 0.6)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", size = 0.8, width = 0.2)+
  #geom_boxplot()+
  scale_shape_manual(name = "Iron",
                     breaks = c("T0", "noFe","Fe"), #this removes the T0 from the legend
                     labels = c("T0","noFe", "addedFe"),
                     values = c(1, 1, 19)) + #this keeps the T0 formatting that I want
  scale_color_manual(name="Temperature",
                     breaks = c("T0","LT","HT"),
                     labels = c("T0","ambient", "warm"),
                     values = c('black', 'black', 'red')) +
  #ylab (expression (F[v]/F[m])) + 
  xlab (expression ("Day Count")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"), 
        #legend.position = "top", # c(0.04,0.88), 
        legend.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 12)) +
   guides(color = guide_legend(override.aes = list(size= 4)))+
   guides(shape = guide_legend(override.aes = list(size= 4)))
```

plots
```{r}
#photophysiology
#png("fvfm_timeseries.png", width=40*0.75, height=30*0.75, units="cm", res=600) 
ggplot(alldata, aes(x=daycount, 
                y=fvfm,
                color = temperature_treatment, 
                shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  geom_point(size = 5, alpha =0.5, stroke =1)+
  #stat_summary(fun = mean, geom = "point", size = 5, stroke = 1.8, alpha = 0.5)+
   #stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1.3, width = 0.1)+
    scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('darkorange2', 'black', 'black')) +
   
    scale_shape_manual(name = "Iron",
                     breaks = c("Fe","noFe"), #this removes the T0 from the legend
                     labels = c("Fe", "noFe"),
                     values = c(19, 1, 1)) + #this keeps the T0 formatting that I want
    #scale_y_continuous(expand = c(0,5))+
  ylab (expression (F[v]/F[m])) + 
  xlab (expression ("Day Count")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))

#dev.off()

#############

photophysi <- filter (alldata, grepl ("T0", full_treatment_name) )

#(DFe56_nM+DFe57_nM)

#png("fvfm_vs_iron.png", width=40*0.75, height=30*0.75, units="cm", res=600) 
ggplot(alldata, aes(x=(PFe57_nM), 
                y=fvfm,
                color = temperature_treatment, 
                shape = iron_treatment)) + 
   #stat_summary(fun = mean, geom = "point", size = 2, stroke = 2, alpha = 0.6)+
#  stat_summary(fun.data =  mean_se, geom = "errorbar", size = 0.8, width = 0.2)+
  facet_wrap(~bioassay)+
  geom_point(alpha =0.5, stroke =1, size = 6)+
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('darkorange2', 'black', 'black')) +
       scale_shape_manual(name = "PFe57",
                     breaks = c("Fe","noFe"), #this removes the T0 from the legend
                     labels = c("Fe", "noFe"),
                     values = c(19, 1, 1)) + #this keeps the T0 formatting that I want
  #scale_y_continuous(expand = c(0,5))+
  ylab (expression (F[v]/F[m])) + 
  xlab (expression ("Iron (nM)")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14)) +
       expand_limits(y = 0:0.6) 
```
