Load necessary packages
```{r}
library(cowplot)
library(data.table)
library(dplyr)
library(ggplot2)
library(ggplotify)
library(ggridges)
library(ggpubr)
library(gridExtra)
library(grid)
library(gplots)
library(magrittr)
library(plotly)
library(reshape2)
library(stringr)
library(scales)
library(tidyr)
library(tidyverse)
#library(treemapify)
library(vegan)
library(goeveg)
library(ggh4x)
setwd("D:/School/PhD/PS117/data/")

alldata <- read.csv("LJ_ps117_allnonproteindata_20220505.csv", header = T)

norm_proteomicsdata <- read.csv("all_ps117_taxon_function_non-normalized_injection_means_20220421.csv", header = T) %>% mutate_at (24:53, funs ((. / sum(.)))) #this normalizes the data to total peptide abundance

source("D:/School/PhD/PS117/PS117_functions.R")
#annotations <- fread("D:/School/PhD/PS117/data/datannotations/annotation_allTFG.grpnorm_mmetsp_fc_pn_reclassified.edgeR.csv") 
#colorblind friendly colors 
#https://rdrr.io/cran/ggthemes/man/colorblind.html

#multi-pannel plots
#http://www.sthda.com/english/articles/24-ggpubr-publication-ready-plots/81-ggplot2-easy-way-to-mix-multiple-graphs-on-the-same-page/

```

plotting functions for all figures 
```{r}
taxon_compartment_temp <- function (taxon, proteincompartment, yaxislabel) {
ribosometaxon <- filter (norm_proteomicsdata, grepl(taxon, F))

ribosometaxon <- ribosometaxon [c(12, 23 ,24:53)]

ribosometaxon2 <- ribosometaxon %>% mutate_at (3:32, funs (((. / sum(.)))))

ribosometaxon3 <- melt(ribosometaxon2, id.vars=c("grpnorm_compartment", "F")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

ribosometaxon_aggregated <- aggregate(ribosometaxon3$norm_abundance, by=list(Category=ribosometaxon3$grpnorm_compartment, ribosometaxon3$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(compartment = Category) %>%
          separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

ribosometaxon_aggregated$fulltreatment <- paste (ribosometaxon_aggregated$temperature, ribosometaxon_aggregated$iron , sep ='_')
ribosometaxon_aggregated$fulltreatment <- ifelse (ribosometaxon_aggregated$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (ribosometaxon_aggregated$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (ribosometaxon_aggregated$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (ribosometaxon_aggregated$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))

#ribosometaxon_aggregated <- filter (ribosometaxon_aggregated, !grepl("T0", fulltreatment))


ribosometaxon_aggregated$temperature <- ifelse (ribosometaxon_aggregated$bioassay ==  "BA1" & ribosometaxon_aggregated$temperature_treatment ==  "T0",  -0.3, 
                         ifelse (ribosometaxon_aggregated$bioassay ==  "BA1" & ribosometaxon_aggregated$temperature_treatment ==  "HT",  1.7, 
                         ifelse (ribosometaxon_aggregated$bioassay ==  "BA1" & ribosometaxon_aggregated$temperature_treatment ==  "LT",  -0.3, 
                         ifelse (ribosometaxon_aggregated$bioassay ==  "BA2" & ribosometaxon_aggregated$temperature_treatment ==  "T0",  -1.4, 
                         ifelse (ribosometaxon_aggregated$bioassay ==  "BA2" & ribosometaxon_aggregated$temperature_treatment ==  "HT",  1, 
                         ifelse (ribosometaxon_aggregated$bioassay ==  "BA2" & ribosometaxon_aggregated$temperature_treatment ==  "LT",  -1, 
                         "Other"))))))

ribosometaxon_aggregated$fulltreatments <- paste (ribosometaxon_aggregated$bioassay, ribosometaxon_aggregated$iron_treatment, sep = "_")


ggplot(filter(ribosometaxon_aggregated, grepl(proteincompartment,compartment)) , aes(
    x = as.numeric(temperature),
    y = norm_abundance,
       shape = fulltreatments, 
    color = fulltreatments))+
  
   #geom_point(size = 4, alpha=0.65, stroke = 1.5)+
  scale_color_manual(name= "Bioassay",
                breaks = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                labels = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
               values = c('darkgreen', "black","black", "darkgreen", "black","black")) +
   stat_summary(fun = mean, geom = "point", show.legend = T, size = 4, stroke = 1.5, alpha = 0.65)+
   stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1, width = 0.1)+
  scale_shape_manual(name = "Bioassay",
                       breaks = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                       labels = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                       values = c(1, 19, 1, 2, 17, 2)) +
    ylab (substitute(yaxislabel)) +
    xlab (expression("Temperature Â°C")) +
    theme_bw()+
    theme(axis.text.x=element_text(face = "bold", size = 12, color = "black", vjust = 0.5, hjust = 0.5),
          axis.title.x=element_text(size=16, color = "black"),
          axis.title.y=element_text(size=16, color = "black"),
          axis.text.y=element_text(face = "bold", size = 12, color = "black"),
          legend.position = "none",
          legend.title = element_text(size = 11, face = "bold"),
          legend.text = element_text(size = 10), 
          strip.text = element_text(size = 11, face = "bold"))
 
}

fig3_metals <- function(yaxis, yaxislabel) {
  ggplot((filter(alldata, grepl("8", daycount))) , aes(
  y = ({{yaxis}}/(1000*PON_uM)),
    #y = ({{yaxis}}/(POP_nM)),
  x = factor(iron_treatment, level = Fe_order),
  color = temperature_treatment, 
  shape = iron_treatment))+
  facet_wrap(~bioassay)+
   scale_color_manual(name="Temperature",
                      breaks = c("LT","HT"),
                      values = c( 'black','darkorange2', "black")) +
    scale_shape_manual(name = "Iron",
                       breaks = c("Fe","noFe", "T0"), #this removes the T0 from the legend
                       labels = c("Fe", "noFe", "T0"),
                       values = c(19,1,5)) +
  stat_summary(fun = mean, geom = "point", size = 4, stroke = 1.5, alpha = 0.65)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F,  size = 1, width = 0.1)+
  theme_bw() + 
  ylab (substitute(bold(yaxislabel))) + 
  xlab (expression ("")) +
  theme(axis.text.x=element_text(face = "bold", size = 10, color = "black"),
        axis.title.y=element_text(face = "bold", size=12, color = "black"), 
        axis.text.y=element_text(size = 10, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 11, face = "bold"),
        #plot.margin = margin(0.3, 0, 0, -0.5, "cm"), #trbl

        #legend.position = c(0.9,0.88),
        legend.position = "none",
        legend.text = element_text(size = 10))
}

protein_clusters <- function (taxonomic_groups, clusternames, protein, shape) {
clusters <- filter (norm_proteomicsdata, grepl(taxonomic_groups, F)) 
clusters <- clusters [c(12, 3, 24:53)]
clusters2 <- clusters %>% mutate_at (3:32, funs (((. / sum(.)))))
clusters3 <- filter(clusters2, grepl("clust_2934|^clust_891$|^clust_332$|^clust_808$|^clust_55$|^clust_4660|^clust_1820$|^clust_1154$|^clust_343$|^clust_1814$|^clust_534$|^clust_5562$|^clust_417$|^clust_411$|clust_2501|clust_6573|^clust_1525$|clust_1552|^clust_231$|^clust_1906$|^clust_812$|^clust_16168$|^clust_10646$|^clust_1372$|^clust_320$|^clust_6019$|^clust_35$|^clust_1830$|^clust_655$|^clust_2490$|^clust_1156$|^clust_2244$", cluster)) 


clusters3 <- melt(clusters3, id.vars=c("cluster", "F")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


clusters3_aggregated <- aggregate(clusters3$norm_abundance, by=list(Category=clusters3$cluster, clusters3$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(cluster = Category) %>% 
                    separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

clusters3_aggregated$fulltreatment <- paste (clusters3_aggregated$temperature, clusters3_aggregated$iron , sep ='_')
clusters3_aggregated$fulltreatment <- ifelse (clusters3_aggregated$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))

clusters4_aggregated <- filter (clusters3_aggregated, !grepl("T0", iron_treatment)) 
clusters4_aggregated$zscore_BA <- ave(clusters4_aggregated$norm_abundance, list(clusters4_aggregated$cluster, clusters4_aggregated$bioassay), FUN=scale) #z-score for each bioassay separately

clusters4_aggregated$clust_annotation <-   ifelse (clusters4_aggregated$cluster ==  "clust_343", "ISIP-2A", 
                   ifelse (clusters4_aggregated$cluster ==  "clust_411", "Nitrate reductase", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_534", "Flavodoxin", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1154", "ISIP-3", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1814", "ISIP-1", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1820", "Plastocyanin", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_4660", "Flavodoxin-1", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_5562", "Cu-Zn SOD", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_55", "PSI", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_808", "Ferredoxin", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_2934", "Multicopper Oxidase", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_891", "Glutamate Synthase", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_332", "Chl A-B binding protein", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_417", "AMT", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_2501", "Fe-Mn SOD", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1525", "SLC-4",
                            ifelse (clusters4_aggregated$cluster ==  "clust_231", "Rhodopsin",
                            ifelse (clusters4_aggregated$cluster ==  "clust_1906", "Mn-cluster",
                            ifelse (clusters4_aggregated$cluster ==  "clust_812", "ZIP",
                            ifelse (clusters4_aggregated$cluster ==  "clust_16168", "TonB",
                            ifelse (clusters4_aggregated$cluster ==  "clust_10646", "TonB_2",
                            ifelse (clusters4_aggregated$cluster ==  "clust_1372", "Urea transporter",
                            ifelse (clusters4_aggregated$cluster ==  "clust_320", "Urea carboxylase",
                            ifelse (clusters4_aggregated$cluster ==  "clust_6019", "Ornithine cyclodeaminase",    
                            ifelse (clusters4_aggregated$cluster ==  "clust_655", "CA_clust_655",
                            ifelse (clusters4_aggregated$cluster ==  "clust_6573", "CA_clust_6573",
                            ifelse (clusters4_aggregated$cluster ==  "clust_35", "Ribosome recycling",
                            ifelse (clusters4_aggregated$cluster ==  "clust_2490", "Rubisco",
                            ifelse (clusters4_aggregated$cluster ==  "clust_2244", "PEPC",
                            ifelse (clusters4_aggregated$cluster ==  "clust_1830", "Acireductone dioxygenase Fe containing",
                            ifelse (clusters4_aggregated$cluster ==  "clust_15664", "Cation Efflux protein",
                                    clusters4_aggregated$cluster 
                    )))))))))))))))))))))))))))))))


clusters4_aggregated$zscore_BAnew <- ifelse (clusters4_aggregated$norm_abundance ==  "0", NA, 
                                             clusters4_aggregated$zscore_BA) #this is for z-score for each bioassay separately

proteinclusterorder <- c("Plastocyanin", 
                         "Multi-Cu Oxidase",  
                         "Cyt Oxidase",
                         "Cu-Zn SOD - centric",  
                         "ZIP", 
                         "RNA polymerase",
                         "Fe-Mn SOD", 
                         "Mn-cluster",
                         "CDCA - centric", 
                         "CDCA - pennate", 
                         "Urea transporter",
                         "Urea Carboxylase")

ggplot((filter(clusters4_aggregated, grepl(clusternames, clust_annotation))), aes(x = factor(iron_treatment, level = Fe_order),
  y = zscore_BAnew, 
  #shape   = clust_annotation, 
  shape = iron_treatment,
  color = temperature_treatment))+ 
  stat_summary(fun = mean, geom = "point", size = 4, stroke = 1.5, alpha = 0.65, position= position_dodge(width=0.3))+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F,  size = 1, width = 0.1, position= position_dodge(width=0.3))+
  theme_bw() +
  scale_color_manual(name="",  

                     breaks = c("LT","HT"),
                     values = c( 'black','darkorange2', "black")) +
   scale_shape_manual(name = "",
                       breaks = c("Fe","noFe", "T0"), #this removes the T0 from the legend
                       labels = c("Fe", "noFe", "T0"),
                       values = c(19,1,5)) +
  
      # breaks = protein,
                    #values =shape) +

  #scale_shape_discrete(name = "")+
    facet_wrap(~bioassay)+
 labs (x = "", y = "Z-Score")+# title = {{plot_title}})+
    theme(axis.text.x=element_text(face = "bold", size = 10, color = "black"),
        axis.title.y=element_text(face = "bold", size=12, color = "black"), 
        axis.text.y=element_text(size = 10, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 11, face = "bold"),
        legend.text = element_text(size = 9, face = "bold"),
        legend.margin = margin(0,0,0,0, "cm"),
        title = element_text(size = 8, face = "bold"),
        legend.position = "top", legend.justification='left') + 
  guides(shape = guide_legend(override.aes = list(size=4)))
        #plot.margin = margin(0.3, 0, 0, -0.5, "cm")) #trbl
}

#need to work on the order of how things are presented
protein_clusters_hm <- function (taxonomic_groups, clusternames, bioassay_name) {
clusters <- filter (norm_proteomicsdata, grepl(taxonomic_groups, F)) 
clusters <- clusters [c(12, 3, 24:53)]
clusters2 <- clusters %>% mutate_at (3:32, funs (((. / sum(.)))))
clusters3 <- filter(clusters2, grepl("clust_2934|^clust_891$|^clust_332$|^clust_808$|^clust_55$|^clust_4660|^clust_1820$|^clust_1154$|^clust_343$|^clust_1814$|^clust_534$|^clust_5562$|^clust_417$|^clust_411$|clust_2501|clust_6573|^clust_1525$|clust_1552|^clust_231$|^clust_1906$|^clust_812$|^clust_16168$|^clust_10646$|^clust_1372$|^clust_320$|^clust_6019$|^clust_35$|^clust_1830$|^clust_655$|^clust_2490$|^clust_1156$|^clust_2244$", cluster)) 

clusters3 <- melt(clusters3, id.vars=c("cluster", "F")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


clusters3_aggregated <- aggregate(clusters3$norm_abundance, by=list(Category=clusters3$cluster, clusters3$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(cluster = Category) %>% 
                    separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

clusters3_aggregated$tempFerep <- paste (clusters3_aggregated$temperature_treatment, clusters3_aggregated$iron_treatment,clusters3_aggregated$replicate,  sep ='_') #add clusters3_aggregated$replicate if reps needed for heatmap

clusters4_aggregated <- filter (clusters3_aggregated, !grepl("T0", iron_treatment)) 
clusters4_aggregated$zscore_BA <- ave(clusters4_aggregated$norm_abundance, list(clusters4_aggregated$cluster, clusters4_aggregated$bioassay), FUN=scale) #z-score for each bioassay separately

clusters4_aggregated$clust_annotation <-   ifelse (clusters4_aggregated$cluster ==  "clust_343", "ISIP-2A", 
                   ifelse (clusters4_aggregated$cluster ==  "clust_411", "Nitrate reductase", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_534", "Flavodoxin", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1154", "ISIP-3", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1814", "ISIP-1", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1820", "Plastocyanin", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_4660", "Flavodoxin-1", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_5562", "Cu-Zn SOD", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_55", "PSI", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_808", "Ferredoxin", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_2934", "Multicopper Oxidase", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_891", "Glutamate Synthase", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_332", "Chl A-B binding protein", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_417", "AMT", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_2501", "Fe-Mn SOD", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1525", "SLC-4",
                            ifelse (clusters4_aggregated$cluster ==  "clust_231", "Rhodopsin",
                            ifelse (clusters4_aggregated$cluster ==  "clust_1906", "Mn-cluster",
                            ifelse (clusters4_aggregated$cluster ==  "clust_812", "ZIP",
                            ifelse (clusters4_aggregated$cluster ==  "clust_16168", "TonB",
                            ifelse (clusters4_aggregated$cluster ==  "clust_10646", "TonB_2",
                            ifelse (clusters4_aggregated$cluster ==  "clust_1372", "Urea transporter",
                            ifelse (clusters4_aggregated$cluster ==  "clust_320", "Urea carboxylase",
                            ifelse (clusters4_aggregated$cluster ==  "clust_6019", "Ornithine cyclodeaminase",    
                            ifelse (clusters4_aggregated$cluster ==  "clust_655", "CA_clust_655",
                            ifelse (clusters4_aggregated$cluster ==  "clust_6573", "CA_clust_6573", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_35", "Ribosome recycling",
                            ifelse (clusters4_aggregated$cluster ==  "clust_2490", "Rubisco",
                            ifelse (clusters4_aggregated$cluster ==  "clust_2244", "PEPC",
                            ifelse (clusters4_aggregated$cluster ==  "clust_1830", "Acireductone dioxygenase Fe containing",
                                    clusters4_aggregated$cluster 
                    ))))))))))))))))))))))))))))))


clusters4_aggregated$zscore_BAnew <- ifelse (clusters4_aggregated$norm_abundance ==  "0", NA, 
                                             clusters4_aggregated$zscore_BA) #this is for z-score for each bioassay separately

proteinclusterorder <- c("ISIP-3",
                         "ISIP-2A",
                         "ISIP-1",
                         "Rhodopsin",
                         "Flavodoxin",
                         "Plastocyanin", 
                         "Flavodoxin-1",
                         "Cu-Zn SOD", 
                         "PSI",
                         "Ferredoxin", 
                         "Nitrate reductase", 
                        "Multicopper Oxidase",
                        "Glutamate Synthase", 
                        "Chl A-B binding protein", 
                        "AMT", 
                        "Fe-Mn SOD", 
                        "Cd Carbonic Anhydrase", 
                        "SLC-4",
                       
                        "Mn-cluster",
                        "ZIP",
                        "TonB",
                        "TonB_2",
                        "Urea transporter",
                        "Urea carboxylase",
                        "Ornithine cyclodeaminase",
                        "CA",
                        "Ribosome recycling",
                        "Rubisco",
                        "PEPC",
                        "Acireductone dioxygenase Fe containing")


clusters4_aggregated <- filter(clusters4_aggregated, grepl (bioassay_name, bioassay))

ggplot((filter(clusters4_aggregated, grepl(clusternames, clust_annotation))),
               aes(#x = factor(fulltreatment, level = treatment_orders),
                 x = factor(tempFerep, level = tempFerep_order),
  y = factor(clust_annotation, level = proteinclusterorder),
  fill = zscore_BAnew)) +
  geom_tile(color = "black", lwd = 0.7)+
  scale_fill_gradientn(colors = c("deepskyblue2","white","darkred"),
         values = rescale(c(-1.6, 0,3)),
         guide = "colorbar", limits=c(-1.6, 3),
         name = "Row Z-score")+
  scale_y_discrete(position = "right")+
labs (x = "", y = "")+ 
  theme(axis.text.x=element_blank(),
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        plot.title = element_text(face = "bold", size=10, vjust = 0, hjust = 0.02, margin = margin(0,0,-5,0)),
        axis.ticks  = element_blank()) +
 #facet_grid(.~bioassay+temperature_treatment+iron_treatment, scales = "free_x") +
      facet_nested(~fct_rev(temperature_treatment)+ ~fct_rev(iron_treatment), scales = "free_x") +
  theme(strip.background=element_rect(color=NA, fill=NA),
        #strip.background =element_rect(fill="white"),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.text = element_text(size = 13,  face = "bold"),
        panel.spacing=unit(c(0.5),"lines"), 
        # strip.text = element_blank()
        )+
  theme(legend.position = "none",
               legend.box.margin=margin(l=-10))
}

protein_clusters_hm_means <- function (taxonomic_groups, clusternames, bioassay_name) {
clusters <- filter (norm_proteomicsdata, grepl(taxonomic_groups, F)) 
clusters <- clusters [c(12, 3, 24:53)]
clusters2 <- clusters %>% mutate_at (3:32, funs (((. / sum(.)))))
clusters3 <- filter(clusters2, grepl("clust_2934|^clust_891$|^clust_332$|^clust_808$|^clust_55$|^clust_4660|^clust_1820$|^clust_1154$|^clust_343$|^clust_1814$|^clust_534$|^clust_5562$|^clust_417$|^clust_411$|clust_2501|clust_6573|^clust_1525$|clust_1552|^clust_231$|^clust_1906$|^clust_812$|^clust_16168$|^clust_10646$|^clust_1372$|^clust_320$|^clust_6019$|^clust_35$|^clust_1830$|^clust_655$|^clust_2490$|^clust_1156$|^clust_2244$", cluster)) 

clusters3 <- melt(clusters3, id.vars=c("cluster", "F")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

clusters3_aggregated <- aggregate(clusters3$norm_abundance, by=list(Category=clusters3$cluster, clusters3$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(cluster = Category) %>% 
                  separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

clusters3_aggregated$tempFerep <- paste (clusters3_aggregated$temperature_treatment, clusters3_aggregated$iron_treatment,  sep ='_') #add clusters3_aggregated$replicate if reps needed for heatmap

clusters4_aggregated <- filter (clusters3_aggregated, !grepl("T0", iron_treatment)) 
clusters4_aggregated$zscore_BA <- ave(clusters4_aggregated$norm_abundance, list(clusters4_aggregated$cluster, clusters4_aggregated$bioassay), FUN=scale) #z-score for each bioassay separately

#if we want to plot the average
clusters4_aggregated <- aggregate(list(zscore_BA =clusters4_aggregated$zscore_BA, norm_abundance = clusters4_aggregated$norm_abundance), by=list( clusters4_aggregated$tempFerep, clusters4_aggregated$bioassay, clusters4_aggregated$cluster), FUN=mean) %>%
  rename (tempFerep = Group.1,
          bioassay = Group.2,
          cluster = Group.3) %>%
separate(tempFerep, into=c("temperature_treatment", "iron_treatment"), sep = "_")


clusters4_aggregated$tempFe <- paste (clusters4_aggregated$temperature_treatment, clusters4_aggregated$iron_treatment,  sep ='_') 

clusters4_aggregated$clust_annotation <-   ifelse (clusters4_aggregated$cluster ==  "clust_343", "ISIP-2A", 
                   ifelse (clusters4_aggregated$cluster ==  "clust_411", "Nitrate reductase", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_534", "Flavodoxin", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1154", "ISIP-3", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1814", "ISIP-1", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1820", "Plastocyanin", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_4660", "Flavodoxin-1", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_5562", "Cu-Zn SOD", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_55", "PSI", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_808", "Ferredoxin", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_2934", "Multicopper Oxidase", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_891", "Glutamate Synthase", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_332", "Chl A-B binding protein", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_417", "AMT", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_2501", "Fe-Mn SOD", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1525", "SLC-4",
                            ifelse (clusters4_aggregated$cluster ==  "clust_231", "Rhodopsin",
                            ifelse (clusters4_aggregated$cluster ==  "clust_1906", "Mn-cluster",
                            ifelse (clusters4_aggregated$cluster ==  "clust_812", "ZIP",
                            ifelse (clusters4_aggregated$cluster ==  "clust_16168", "TonB",
                            ifelse (clusters4_aggregated$cluster ==  "clust_10646", "TonB_2",
                            ifelse (clusters4_aggregated$cluster ==  "clust_1372", "Urea transporter",
                            ifelse (clusters4_aggregated$cluster ==  "clust_320", "Urea carboxylase",
                            ifelse (clusters4_aggregated$cluster ==  "clust_6019", "Ornithine cyclodeaminase",    
                            ifelse (clusters4_aggregated$cluster ==  "clust_655", "CA_clust_655",
                            ifelse (clusters4_aggregated$cluster ==  "clust_6573", "CA_clust_6573", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_35", "Ribosome recycling",
                            ifelse (clusters4_aggregated$cluster ==  "clust_2490", "Rubisco",
                            ifelse (clusters4_aggregated$cluster ==  "clust_2244", "PEPC",
                            ifelse (clusters4_aggregated$cluster ==  "clust_1830", "Acireductone dioxygenase Fe containing",
                                    clusters4_aggregated$cluster 
                    ))))))))))))))))))))))))))))))


clusters4_aggregated$zscore_BAnew <- ifelse (clusters4_aggregated$norm_abundance ==  "0", NA, 
                                             clusters4_aggregated$zscore_BA) #this is for z-score for each bioassay separately

proteinclusterorder <- c("ISIP-3",
                         "ISIP-2A",
                         "ISIP-1",
                         "Rhodopsin",
                         "Flavodoxin",
                         "Plastocyanin", 
                         "Flavodoxin-1",
                         "Cu-Zn SOD", 
                         "PSI",
                         "Ferredoxin", 
                         "Nitrate reductase", 
                        "Multicopper Oxidase",
                        "Glutamate Synthase", 
                        "Chl A-B binding protein", 
                        "AMT", 
                        "Fe-Mn SOD", 
                        "Cd Carbonic Anhydrase", 
                        "SLC-4",
                       
                        "Mn-cluster",
                        "ZIP",
                        "TonB",
                        "TonB_2",
                        "Urea transporter",
                        "Urea carboxylase",
                        "Ornithine cyclodeaminase",
                        "CA",
                        "Ribosome recycling",
                        "Rubisco",
                        "PEPC",
                        "Acireductone dioxygenase Fe containing")


clusters4_aggregated <- filter(clusters4_aggregated, grepl (bioassay_name, bioassay))

ggplot((filter(clusters4_aggregated, grepl(clusternames, clust_annotation))),
               aes(#x = factor(fulltreatment, level = treatment_orders),
                 x = factor(tempFe, level = tempFe_order),
                 #x = tempFerep,
  y = factor(clust_annotation, level = proteinclusterorder),
  fill = zscore_BAnew)) +
  geom_tile(color = "black", lwd = 0.7)+
  scale_fill_gradientn(colors = c("deepskyblue2","white","darkred"),
         values = rescale(c(-1.6, 0,3)),
         guide = "colorbar", limits=c(-1.6, 3),
         name = "Row Z-score")+
  scale_y_discrete(position = "right")+
labs (x = "", y = "")+ 
  theme(#axis.text.x=element_blank(),
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        plot.title = element_text(face = "bold", size=10, vjust = 0, hjust = 0.02, margin = margin(0,0,-5,0)),
        axis.ticks  = element_blank()) +
 facet_grid(~bioassay, scales = "free_x") +
     # facet_nested(~fct_rev(temperature_treatment)+ ~fct_rev(iron_treatment), scales = "free_x") +
  theme(strip.background=element_rect(color=NA, fill=NA),
        #strip.background =element_rect(fill="white"),
        panel.border = element_blank(),
        panel.background = element_blank(),
        strip.text = element_text(size = 13,  face = "bold"),
        panel.spacing=unit(c(0.5),"lines"), 
        # strip.text = element_blank()
        )+
  theme(legend.position = "none",
               legend.box.margin=margin(l=-10))
}



alldata$bioassayFe <- paste (alldata$bioassay, alldata$iron_treatment, sep = "_")

alldata$tempFe <- paste (alldata$temperature_treatment, alldata$iron_treatment, sep = "_") 

Fe_order <- c('T0', 'noFe', 'Fe')
tempFe_order <- c('T0', 'LT_noFe', 'LT_Fe', 'HT_noFe', 'HT_Fe')
tempFerep_order <- c('LT_noFe_A, ','LT_noFe_B', 'LT_noFe_C', 
                     'LT_Fe_A',  'LT_Fe_B',  'LT_Fe_C',
                     'HT_noFe_A', 'HT_noFe_B', 'HT_noFe_C', 
                     'HT_Fe_A', 'HT_Fe_B', 'HT_Fe_C')

```

#Figure 1
map
```{r}
library(ggOceanMaps)
library(ggOceanMapsData)

#https://mran.microsoft.com/snapshot/2021-03-14/web/packages/ggOceanMaps/vignettes/ggOceanMaps.html
#https://cran.r-project.org/web/packages/ggOceanMaps/ggOceanMaps.pdf

#the numbers in the limits represent the range of the longs and lats 
#the first two numbers represent the longitude range to be shown, the secnd two numbers are the latitude range to be shown. 

#png("ps117_stationmap.png", width=45*0.75, height=25*0.75, units="cm", res=900)

bioassays <- data.frame(lon = c(-11.044,-0.011), lat =c(-70.25, -65), bioassay= c("BA2", "BA1")) 
#bioassays <- data.frame(lon = c(165.4133), lat =c(-77.61895), bioassay= c("")) # mcmurdo ross sea TFG incubations
map <- basemap(limits = c(-130, 100, -90,-55),
        bathymetry = TRUE, #this adds the bathymetry 
        glaciers = FALSE, 
        grid.col = "NA",
        grid.size = 0.2,
        lat.interval = 5)+ 
  scale_color_continuous()+
  #ylab("Latitude")+ 
  #xlab("Longitude")+
  ylab("")+ 
  xlab("")+
      theme(
        axis.text.x=element_text(face = "bold", color = "black", size = 15),
        axis.title.x=element_text(size = 16, face="bold"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        axis.title.y=element_text(size = 16,face="bold", color = "black"), 
        legend.title = element_text(size = 10, face = "bold"), 
        legend.text = element_text(face = "bold"), 
        legend.position = "top",
        legend.justification = "left",
        #plot.margin = unit(c(0,0,0,0), units = "cm"), 
        legend.key.height= unit(0.3, 'cm')) + 
        annotation_scale(location = "br", text_cex = 1, text_face = "bold") + 
        geom_spatial_point(data = bioassays, aes(x = lon, y = lat), color = "red", alpha = 0.5, size = 8) +
    geom_spatial_text(size = 6, aes(label=bioassays$bioassay, x = bioassays$lon, y = bioassays$lat), hjust =1.2, vjust = 0, color = "black") 



#dev.off()
```

legend
```{r}
alldata$fulltreatment <- paste (alldata$temperature_treatment, alldata$iron_treatment, sep = "_")

fig1_legend <- ggplot(filter (alldata, grepl ("T0|T8", full_treatment_name) ), 
       aes(x=iron_treatment, 
       y=fvfm,
       color = fulltreatment, 
       shape = fulltreatment))+
       #color = temperature_treatment, 
       #shape = iron_treatment)) + 
  geom_point(size = 4, alpha =0.65, stroke =1.5)+
  
    scale_color_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c('black', 'black', 'black','darkorange2', "darkorange2")) +
  
     scale_shape_manual(name="",
                      breaks = c("T0_T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      labels = c("T0", "LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
                      values = c(5, 1, 19,1, 19)) +
  
  # scale_color_manual(name="",
  #                     breaks = c("LT_noFe", "LT_Fe", "HT_noFe", "HT_Fe"),
  #                     labels = c("LT", "HT"),
  #                     values = c('black', 'darkorange2','black')) +
  # scale_shape_manual(name = "",
  #                      breaks = c("noFe","Fe"), #this removes the T0 from the legend
  #                      labels = c("noFe", "Fe"),
  #                      values = c(1,19,1)) +
  # scale_x_discrete(limits=c("T0","noFe","Fe"), 
  #                  labels = c("T0", "noFe", "Fe"))+
  theme_bw()+
  theme(legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12, margin = margin(r = 0.5, unit = 'cm')), 
        legend.position = "top") 


fig1_legend <- get_legend(fig1_legend)
fig1_legend <- as_ggplot(fig1_legend)

fig1_legend
```

T0 dissolved iron and Mn
```{r}
t0 <- filter (alldata, grepl ("T0", full_treatment_name))

t0 <- t0 [c(4, 24,26)]

t0_2 <- melt(t0, id.vars=c("bioassay")) %>%
            rename(metal = variable) %>%
            rename(concentration = value )
t0_2 <- t0_2 %>% separate(metal, c("metal", "thing")) 

t0_2$metal <-   ifelse (t0_2$metal ==  "DFe56", "DFe", t0_2$metal) 
                                    
                


t0_3 <- filter (t0_2, grepl ("BA", bioassay))


ggplot(filter (t0_3, !grepl ("0.58", concentration)), 
       aes (y= concentration,
       x= metal))+
    stat_summary(fun = mean, geom = "bar", size = 4)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1)+

  xlab (expression ("")) +
    ylab (expression ("nM")) +
    facet_wrap(~bioassay)+

  theme_bw()+
  theme(axis.text.x=element_text(size = 14, color = "black"),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text (size = 16, color = "black"), 
         strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 14, face = "bold"))+
     expand_limits(y = 0:0.4) 
       
        

```

nutrients, chla, fv/fmphysical oceanography stuff
```{r}
fvfm <- ggplot(filter (alldata, grepl ("T0|T8", full_treatment_name) ), 
       aes(x=iron_treatment, 
       y=fvfm,
       color = temperature_treatment, 
       shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  #geom_point(size = 5, alpha =0.5, stroke =1)+
  stat_summary(fun = mean, geom = "point", size = 4, stroke = 1.5, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1)+
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('black', 'darkorange2','black')) +
  scale_shape_manual(name = "Iron",
                       breaks = c("Fe","noFe", "T0"), #this removes the T0 from the legend
                       labels = c("Fe", "noFe", "T0"),
                       values = c(19,1,5)) +
  scale_x_discrete(limits=c("T0","noFe","Fe"), 
                   labels = c("T0", "noFe", "Fe"))+
  ylab (expression (F[v]/F[m])) + 
  xlab (expression ("")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 14,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 12, color = "black"),
        axis.title.y=element_text(size=18, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 12, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 12, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14), 
        legend.position = "none") 

POC <- ggplot(filter (alldata, grepl ("T0|T8", full_treatment_name) ), 
       aes(x=iron_treatment, 
       y=POC_uM,
       color = temperature_treatment, 
       shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  #geom_point(size = 5, alpha =0.5, stroke =1)+
  stat_summary(fun = mean, geom = "point", size = 4, stroke = 1.5, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1)+
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('black', 'darkorange2','black')) +
  scale_x_discrete(limits=c("T0","noFe","Fe"), 
                   labels = c("T0", "noFe", "Fe"))+
   scale_shape_manual(name = "Iron",
                       breaks = c("Fe","noFe", "T0"), #this removes the T0 from the legend
                       labels = c("Fe", "noFe", "T0"),
                       values = c(19,1,5)) +
  ylab (expression (POC~mu*M)) + 
  xlab (expression ("")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 12,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 12, color = "black"),
        axis.title.y=element_text(size=18, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 12, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 12, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14), 
        legend.position = "none") 


chla <- ggplot(filter (alldata, grepl ("T0|T8", full_treatment_name) ), 
       aes(x=iron_treatment, 
       y=total_chla_ug_L,
       color = temperature_treatment, 
       shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  #geom_point(size = 5, alpha =0.5, stroke =1)+
  stat_summary(fun = mean, geom = "point", size = 4, stroke = 1.5, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1)+
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('black', 'darkorange2','black')) +
  scale_x_discrete(limits=c("T0","noFe","Fe"), 
                   labels = c("T0", "noFe", "Fe"))+
   scale_shape_manual(name = "Iron",
                       breaks = c("Fe","noFe", "T0"), #this removes the T0 from the legend
                       labels = c("Fe", "noFe", "T0"),
                       values = c(19,1,5)) +
  ylab (expression (Chl-a~~mu*g~L^-1)) + 
  xlab (expression ("")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 12,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 12, color = "black"),
        axis.title.y=element_text(size=18, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 12, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 12, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14), 
        legend.position = "none") 


NO3 <- x_vs_time (alldata, daycount, NO3_uM, NO[3]~mu*M) + expand_limits(y = 0:30)+ xlab (expression ("")) +  theme(legend.position = "none") 

PO4 <- x_vs_time (alldata, daycount, PO4_uM, PO[4]~mu*M) + theme(legend.position = "none") + xlab (expression (""))

Si <- x_vs_time (alldata, daycount, Si_uM, Si~mu*M) + expand_limits(y = 0:80) +theme(legend.position = "none") 

  


```

plot
```{r}
##

x<- plot_grid(fvfm, NULL, NO3,
          chla, NULL,  PO4, 
          POC, NULL, Si,
           
          labels = c("B","",  "E",
                     "C","",  "F",
                     "D", "", "G"), 
           ncol = 3, 
          rel_widths = c(1,0.05,1),
          align = "vh")

map2 <- plot_grid(fig1_legend, map,
          nrow = 2, 
          rel_heights = c(0.1,1), 
          rel_widths = c(0.5,1), 
          labels = c("A"), 
          axis = "rbtl")

plot_grid(map2, NULL, x,
          ncol = 3, 
          rel_widths = c(1,0.02,1.25), 
          axis = "rbtl")
     

###


# map <- plot_grid(map, 
#           nrow= 1,
#           labels = c("A"))
# 
# 
# x <- plot_grid(NULL, fvfm, NULL, POC,
#           ncol = 2,
#           labels = c("B","", "C",""), 
#           align = "v", 
#           rel_widths = c(0.05,1,0.05,1))
# 
# col2 <- plot_grid(NO3, PO4, Si,
#           ncol = 1,
#           labels = c("D", "E", "F"),
#           align = "v")
# 
# col1 <- plot_grid(map, x,
#           nrow=2, 
#           rel_heights  = c(2,3))
# 
# 
# plot_grid(col1, NULL, col2,
#           ncol=3, 
#           rel_widths  = c(1,0.1 ),
#          align = "vh")

```



#Figure 2 
Figure 2 - pigments
```{r}
###
pigment_fract <- alldata [c(14,54:59)]
pigment_fract <- na.omit(pigment_fract)
pigment_fract2 <- melt(pigment_fract, id.vars=c("full_treatment_name")) %>%
            rename(taxon_group = variable) %>%
            rename(fract = value )

pigment_fract2 <- pigment_fract2 %>% separate(full_treatment_name, c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate")) 


pigment_fract2$fulltreatment <- paste(pigment_fract2$temperature_treatment,pigment_fract2$iron_treatment, sep = "_")

pigment_fract2 <- pigment_fract2 %>% separate(taxon_group, c("bla", "taxon_group")) 

pigment_fract2$taxon_group <- reorder(pigment_fract2$taxon_group, pigment_fract2$fract) #this reorders the data so when it's plotted, the most abundant fraction is on the bottom #only wroks when there are no NA's

treatment_order <- c('T0_T0', 'LT_noFe', 'LT_Fe', 'HT_noFe', 'HT_Fe')


#png("pigment_fraction_20220329.png", width=40*0.75, height=30*0.75, units="cm", res=600) 
ggplot (pigment_fract2, aes(
  x = factor(fulltreatment, level = treatment_order),
  y = fract, 
  fill = taxon_group)) +  
  geom_bar(position = "fill", stat = "identity") + #position = "fill" does percentage, stack does raw values
  facet_grid(.~ bioassay)+
theme_bw()+
   theme (axis.text.x = element_text(angle = 90)) +
   ylab ("Contribution to total pigment (fraction)")+
    xlab ("")+
   scale_fill_manual(name="",
                     values = c("diatoms" = "#117733",
                                "dinoflagellates" = "#0072B2",
                                "haptophytes" = "#CC79A7",
                                "prasinophytes" = "#88CCEE",
                                "cryptophytes" = "grey", #000000",
                                "pelagophytes" = "#E69F00"))+
  theme(axis.text.x=element_text(face = "bold", size = 18, color = "black", vjust = 1, hjust = 1, angle = 45),
        axis.title.y=element_text(size=20,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 15), 
        strip.text.x = element_text(size = 18, face = "bold"))
  

#dev.off()
#pigmentheatmap

pigment_means <- aggregate(pigment_fract2$fract, by=list( pigment_fract2$fulltreatment, pigment_fract2$bioassay, pigment_fract2$taxon_group), FUN=mean) %>%
  rename (fulltreatment = Group.1, 
          bioassay = Group.2, 
          taxon_group = Group.3, 
          fract = x)

pigment_means$taxon_group2 <-   ifelse (pigment_means$taxon_group ==  "diatoms", "Bacillariophyta", 
                   ifelse (pigment_means$taxon_group ==  "haptophytes", "Haptophyta", 
                   ifelse (pigment_means$taxon_group ==  "pelagophytes", "Pelagophyceae", 
                   ifelse (pigment_means$taxon_group ==  "cryptophytes", "Cryptophyta",
                   ifelse (pigment_means$taxon_group ==  "chlorophytes", "Chlorophyta", 
                   ifelse (pigment_means$taxon_group ==  "dinoflagellates", "Dinophyta",
                           "else" 
                                    
                    ))))))


pigment_means$taxon_group2 <- reorder(pigment_means$taxon_group2, pigment_means$fract) #this reorders the data so when it's plotted, the most abundant fraction is on the bottom #only wroks when there are no NA's


pigment_means$treatment2 <- ifelse (pigment_means$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (pigment_means$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (pigment_means$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (pigment_means$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))

fig2_pigmentmeans <-ggplot(pigment_means, aes(x = factor(treatment2, level = tempFe_order),
 y = taxon_group2, 
  fill = fract)) +  
    geom_tile(color = "black", lwd = 0.7)+
    #coord_fixed()+ #this makes it all squares 
   scale_fill_gradientn(colours = c("deepskyblue2","white","darkred"),
                         values = rescale(c(0.0000001,0.35,.9)),
                         guide = "colorbar", limits=c(0.0000001,0.9),
                       name = "Fraction")+
   ylab ("")+
   xlab ("")+
   theme(axis.text.x=element_text(face = "bold", size = 11, color = "black", angle = 45, vjust = 1.2, hjust = 1),
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank(), 
        axis.ticks  = element_blank()) +
    facet_grid(.~ bioassay, scales = "free_x") + 
    theme(strip.background =element_rect(fill="white"), 
        strip.text = element_text(size = 12, face = "bold"),
         legend.box.margin=margin(l=-10))

```

figure 2 protein heatmap
```{r}
allgroups <- norm_proteomicsdata [c(7,9,10, 24:53)]
allgroups <- melt(allgroups, id.vars=c("domain", "C","class")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

#test <- filter (norm_proteomicsdata, grepl("Hapto", C))

allgroups$class_a <-   ifelse (allgroups$class ==  "Bacillariophyta", "Bacillariophyta", 
                                 
                    ifelse (allgroups$C ==  "Dinophyta", "Dinophyta",
                                    
                    ifelse (allgroups$class ==  "Haptophyta", "Haptophyta",
                    ifelse (allgroups$class ==  "Prymnesiophyceae", "Haptophyta",
                    ifelse (allgroups$class ==  "Pavlovophyceae", "Haptophyta",
                                    
                    ifelse (allgroups$class ==  "Cryptophyceae", "Cryptophyta",
                    ifelse (allgroups$class ==  "Cryptophyta", "Cryptophyta",
                                    
                    ifelse (allgroups$class ==  "Pelagophyceae", "Pelagophyceae",
                                    
                    ifelse (allgroups$class ==  "Chlorophyceae", "Chlorophyta",
                    ifelse (allgroups$class ==  "Chlorophyta", "Chlorophyta",
                    ifelse (allgroups$class ==  "Pyramimonadales", "Chlorophyta",
                    ifelse (allgroups$class ==  "Prasinophyceae", "Chlorophyta",
                            
                    ifelse (allgroups$class ==  "Spirotrichea", "Ciliophora",
                    ifelse (allgroups$class ==  "Prostomatea", "Ciliophora",  
                    ifelse (allgroups$class ==  "Oligohymenophorea", "Ciliophora",
                    ifelse (allgroups$class ==  "Litostomatea", "Ciliophora",   
                    ifelse (allgroups$class ==  "Heterotrichea", "Ciliophora", 
                    ifelse (allgroups$class ==  "Colpodea", "Ciliophora", 
                    ifelse (allgroups$class ==  "Ciliophora", "Ciliophora", 
                            
                    ifelse (allgroups$class == "Eukaryota", "Other Eukaryota",
                    ifelse (allgroups$domain == "Bacteria", "Bacteria",
                    ifelse (allgroups$domain == "Viruses", "Viruses",
                    "Other stuff"))))))))))))))))))))))

allgroups_aggregated <- aggregate(allgroups$norm_abundance, by=list( allgroups$treatment, allgroups$class_a), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.1) %>%
                  rename(class = Group.2) 

allgroupshm <- filter (allgroups_aggregated, !grepl("Other stuff", class))  %>% separate(treatment, into =  c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"))  
 
allgroupshm$treatmentrep <- paste (allgroupshm$temperature_treatment, allgroupshm$iron_treatment, allgroupshm$replicate, sep = "_")

allgroupshm$treatment<- paste (allgroupshm$temperature_treatment, allgroupshm$iron_treatment, sep = "_")

allgroupshm$class <- reorder(allgroupshm$class , allgroupshm$norm_abundance )

treatmentrep_order <- c('T0_T0_A', 'T0_T0_B', 'T0_T0_C', 'T0_T0_D',
                     'LT_noFe_A', 'LT_noFe_B','LT_noFe_C',
                     'LT_Fe_A', 'LT_Fe_B', 'LT_Fe_C', 
                     'HT_noFe_A', 'HT_noFe_B','HT_noFe_C',
                     'HT_Fe_A', 'HT_Fe_B', 'HT_Fe_C')


allgroupstop <- filter (allgroupshm, grepl("Bacillar|Other|Hapto|Bacteria", class))

ggplot(allgroupstop, aes( x = factor(treatmentrep, level = treatmentrep_order), y = class, fill= abundance))+ 
  geom_tile(aes(fill = norm_abundance),color = "black", lwd = 0.7 ) + 
  #coord_fixed()+ #this makes it all squares 
  scale_fill_gradient2(low = "deepskyblue2", 
                       high = "darkred",
                       mid = "white", 
                       midpoint = 0.25,
                       name = "normalized abundance") +
   ylab ("")+
   xlab ("")+
   theme(axis.text.x=element_text(face = "bold", size = 8, color = "black", vjust = 1,            hjust = 1, angle = 45),
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank(), 
        axis.ticks.y  = element_blank()) +
   guides(fill = guide_colourbar(barwidth = 1, barheight = 15)) +
  facet_grid(.~ bioassay, scales = "free_x") #free_x removes replicates that aren't there


allgroupsbottom <- filter (allgroupshm, !grepl("Bacillar|Other|Hapto|Bacteria", class))
ggplot(allgroupsbottom, aes( x = factor(treatmentrep, level = treatmentrep_order), y = class, fill= abundance))+   geom_tile(aes(fill = norm_abundance),color = "black", lwd = 0.7 ) + 
  #coord_fixed()+ #this makes it all squares 
  scale_fill_gradient2(low = "deepskyblue2", 
                       high = "darkred",
                       mid = "white", 
                       midpoint = 0.003,
                       name = "normalized abundance") +
   ylab ("")+
   xlab ("")+
   theme(axis.text.x=element_text(face = "bold", size = 8, color = "black", vjust = 1,            hjust = 1, angle = 45),
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank(), 
        axis.ticks.y  = element_blank()) +
   guides(fill = guide_colourbar(barwidth = 1, barheight = 15)) +
  facet_grid(.~ bioassay, scales = "free_x") #free_x removes replicates that aren't there



###
### means
###

allgroupshm_means <- aggregate(allgroupshm$norm_abundance, by=list( allgroupshm$treatment, allgroupshm$bioass, allgroupshm$class), FUN=mean) %>%
  rename (treatment = Group.1, 
          bioassay = Group.2, 
          class = Group.3, 
          norm_abundance = x)

allgroupshm_means$treatment2 <- ifelse (allgroupshm_means$treatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (allgroupshm_means$treatment ==  "LT_Fe", "LT_Fe",
                            ifelse (allgroupshm_means$treatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (allgroupshm_means$treatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))

allgroups_bottommeans <- filter (allgroupshm_means, !grepl("Bacillar|Other|Hapto|Bacteria", class))

allgroups_topmeans <- filter (allgroupshm_means, grepl("Bacillar|Hapto|Bacteria", class))


fig2_proteintopmeans <- ggplot(allgroups_topmeans, aes(x = factor(treatment2, level = tempFe_order), y = class, fill= norm_abundance))+ 
  geom_tile(color = "black", lwd = 0.7) +
 # coord_fixed(ratio=2.5)+ #this makes it all squares 
  # scale_fill_gradient2(low = "deepskyblue2", 
  #                      high = "darkred",
  #                      mid = "white", 
  #                      midpoint = 0.2,
  #                      name = "Abundance") +
  
    scale_fill_gradientn(colours = c("deepskyblue2","white","darkred"),
                         values = rescale(c(0.0000001,0.13,.55)),
                         guide = "colorbar", limits=c(0.0000001,0.55),
                       name = "Fraction")+
   ylab ("")+
   xlab ("")+
   theme(axis.text.x=element_blank(),
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank(), 
        axis.ticks  = element_blank()) +
     #guides(fill = guide_colourbar(barwidth = 1, barheight = 8)) +
  facet_grid(.~ bioassay)+
  theme(strip.background =element_rect(fill="white"), 
        strip.text = element_text(size = 12, face = "bold"), 
        plot.margin=margin(b=0, unit="cm"), 
         legend.box.margin=margin(l=-10))


fig2_proteinbottommeans <- ggplot(allgroups_bottommeans, aes(x = factor(treatment2, level = tempFe_order), y = class, fill= norm_abundance))+ 
  geom_tile(color = "black", lwd = 0.7) +
  scale_fill_gradientn(colours = c("deepskyblue2","white","darkred"),
                         values = rescale(c(0.0000001,0.001,.035)),
                         guide = "colorbar", limits=c(0.0000001,0.035),
                       name = "Fraction")+
   ylab ("")+
   xlab ("")+
   theme(axis.text.x=element_text(face = "bold", size = 11, color = "black", vjust = 1.2, hjust = 1,  angle = 45),
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank()) +
       facet_grid(.~ bioassay)+
  theme(strip.background =element_rect(fill="white"), 
        strip.text = element_blank(),
        plot.margin=margin(t=-0.5, unit="cm"), 
        legend.box.margin=margin(l=-10),
        axis.ticks  = element_blank())
```

figure2 - nmds

The vegan package displays 'species' and 'sites'. The sites are the treatments, the 'species' are the variables (pigments, FC-groups, proteomics species etc) 
peptide nmds
```{r}
proteomics_taxonomy <- norm_proteomicsdata [c(2,24:53)]

proteomics_taxonomy <- melt(proteomics_taxonomy, id.vars=c("orf_id")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


proteomics_taxonomy2 <- dcast(proteomics_taxonomy, treatment~orf_id, fun.aggregate = sum,  value.var = "norm_abundance")

proteomics_taxonomy2 <- proteomics_taxonomy2 %>% separate(treatment, into=c("bioassay", "timepoint" ,"temperature_treatment", "iron_treatment", "replicate"), sep = "_")

proteomics_taxonomy2_nmds = proteomics_taxonomy2 [c(6:23175)] 

proteomics_taxonomy2_nmds = as.matrix(proteomics_taxonomy2_nmds)

dimcheckMDS(
  proteomics_taxonomy2_nmds,
  distance = "bray",
  k = 4,
  trymax = 1000,
  autotransform = TRUE
)

set.seed(123)

nmds = metaMDS(proteomics_taxonomy2_nmds, try = 1000, trymax = 1000, k = 2, distance = "bray") # try = 500, trymax = 500
nmds
scores(nmds, display = "sites")
stressplot(nmds) #for the suplement

data.scores <- as.data.frame(scores(nmds, "sites")) #can select sites (treatments) or species (variable)

# nmdsspecies <- dcast(proteomics_taxonomy_aggregated, genus+compartment~treatment)
#            # rename(treatment = variable) %>%
#             #rename(norm_abundance = value )

# data.scores$cluster <- nmdsspecies$compartment

data.scores$temperature_treatment <- proteomics_taxonomy2$temperature_treatment
data.scores$iron_treatment <- proteomics_taxonomy2$iron_treatment
data.scores$bioassay <- proteomics_taxonomy2$bioassay

data.scores$bioassaytempfe <- paste(data.scores$bioassay, data.scores$temperature_treatment,data.scores$iron_treatment, sep = "_")


fig2_peptide_nmds <- ggplot(data.scores, aes(x = NMDS1, 
                        y = NMDS2))+
  geom_point(size = 3, alpha = 0.75, stroke = 1.2, 
                aes(fill = temperature_treatment,
                   # alpha = iron_treatment, 
                   # shape = bioassay,
                    shape = bioassaytempfe,
                    color = temperature_treatment))+
  scale_shape_manual( name = "Bioassay",
                      values=c(  "BA1_T0_T0" = 1, 
                               "BA1_LT_noFe" = 1,
                               "BA1_LT_Fe" = 19,
                               "BA1_HT_Fe" = 19,
                                "BA1_HT_noFe" = 1, 
                        
                               "BA2_T0_T0" = 2, 
                               "BA2_LT_noFe" = 2,
                               "BA2_LT_Fe" = 17,
                               "BA2_HT_Fe" = 17,
                                "BA2_HT_noFe" =2 ))+
                        
  scale_color_manual(values = c("HT" = "darkorange", "LT" = "black", "T0" = "darkgreen"), 
                     name = "Temperature")+
  scale_fill_manual(values = c("HT" = "darkorange", "LT" = "black", "T0" = "blue"), guide = FALSE) +
  ggtitle("Proteins")+
    theme(axis.text.y = element_text(colour = "black", size = 11, face = "bold"), 
    axis.text.x = element_text(colour = "black", face = "bold", size = 11), 
    axis.title.y = element_text(face = "bold", size = 12), 
    legend.text = element_text(size = 10, face ="bold", colour ="black"), 
    legend.position = "none", 
    legend.justification='right',
    legend.box.margin=unit(0, "cm"),
        legend.direction='horizontal',
    legend.title = element_text(size = 12, colour = "black", face = "bold"), 
    axis.title.x = element_text(face = "bold", size = 12, colour = "black"), 
    panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
    legend.key=element_blank(), 
      plot.title = element_text(vjust = -8, hjust = 0.02))
```

flow cytometry nmds 2022-07-27
```{r}
FC <- alldata [c(14,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95,97,99,101,103,105)] %>% na.omit() %>% separate(full_treatment_name, c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate")) 

FC_nmds <- FC [c(6:28)] 

FC_nmds = as.matrix(FC_nmds)

set.seed(123)
nmds = metaMDS(FC_nmds, try = 10000, distance = "bray") # try = 500, trymax = 500
nmds

data.scores <- as.data.frame(scores(nmds, "sites"))
stressplot(nmds)

data.scores$temperature_treatment <- FC$temperature_treatment
data.scores$iron_treatment <- FC$iron_treatment
data.scores$bioassay <- FC$bioassay

data.scores$bioassaytempfe <- paste(data.scores$bioassay, data.scores$temperature_treatment,data.scores$iron_treatment, sep = "_")

fig2_FC_nmds <-  ggplot(data.scores, aes(x = NMDS1, 
                        y = NMDS2))+
  geom_point(size = 3, alpha = 0.75, stroke = 1.2, 
                aes(fill = temperature_treatment,
                   # alpha = iron_treatment, 
                   # shape = bioassay,
                    shape = bioassaytempfe,
                    color = temperature_treatment))+
  scale_shape_manual( name = "Bioassay",
                      values=c(  "BA1_T0_T0" = 1, 
                               "BA1_LT_noFe" = 1,
                               "BA1_LT_Fe" = 19,
                               "BA1_HT_Fe" = 19,
                                "BA1_HT_noFe" = 1, 
                        
                               "BA2_T0_T0" = 2, 
                               "BA2_LT_noFe" = 2,
                               "BA2_LT_Fe" = 17,
                               "BA2_HT_Fe" = 17,
                                "BA2_HT_noFe" =2 ))+
                        
  scale_color_manual(values = c("HT" = "darkorange", "LT" = "black", "T0" = "darkgreen"), 
                     name = "Temperature")+
  scale_fill_manual(values = c("HT" = "darkorange", "LT" = "black", "T0" = "blue"), guide = FALSE) +
  ggtitle("Flow Cytometry")+
    theme(axis.text.y = element_text(colour = "black", size = 11, face = "bold"), 
    axis.text.x = element_text(colour = "black", face = "bold", size = 11), 
    axis.title.y = element_text(face = "bold", size = 12), 
    legend.text = element_text(size = 10, face ="bold", colour ="black"), 
    legend.position = "none", 
    legend.justification='right',
    legend.box.margin=unit(0, "cm"),
        legend.direction='horizontal',
    legend.title = element_text(size = 12, colour = "black", face = "bold"), 
    axis.title.x = element_text(face = "bold", size = 12, colour = "black"), 
    panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
    legend.key=element_blank(), 
      plot.title = element_text(vjust = -8, hjust = 0.02))
```

pigment nmds 
```{r}
pigment_nmds <- alldata [c(14,41:46)] %>% na.omit() %>% separate(full_treatment_name, c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate")) 

pigment_nmds2 <- pigment_nmds [c(6:11)] 

pigment_nmds2 = as.matrix(pigment_nmds2)


set.seed(123)
nmds = metaMDS(pigment_nmds2, try = 1000, distance = "bray") # try = 500, trymax = 500
nmds

data.scores <- as.data.frame(scores(nmds, "sites"))
stressplot(nmds)

data.scores$temperature_treatment <- pigment_nmds$temperature_treatment
data.scores$iron_treatment <- pigment_nmds$iron_treatment
data.scores$bioassay <- pigment_nmds$bioassay

data.scores$bioassaytempfe <- paste(data.scores$bioassay, data.scores$temperature_treatment,data.scores$iron_treatment, sep = "_")

fig2_pigment_nmds <- ggplot(data.scores, aes(x = NMDS1, 
                        y = NMDS2))+
  geom_point(size = 3, alpha = 0.75, stroke = 1.2, 
                aes(fill = temperature_treatment,
                   # alpha = iron_treatment, 
                   # shape = bioassay,
                    shape = bioassaytempfe,
                    color = temperature_treatment))+
  scale_shape_manual( name = "Bioassay",
                      values=c(  "BA1_T0_T0" = 1, 
                               "BA1_LT_noFe" = 1,
                               "BA1_LT_Fe" = 19,
                               "BA1_HT_Fe" = 19,
                                "BA1_HT_noFe" = 1, 
                        
                               "BA2_T0_T0" = 2, 
                               "BA2_LT_noFe" = 2,
                               "BA2_LT_Fe" = 17,
                               "BA2_HT_Fe" = 17,
                                "BA2_HT_noFe" =2 ))+
                        
  scale_color_manual(values = c("HT" = "darkorange", "LT" = "black", "T0" = "darkgreen"), 
                     name = "Temperature")+
  scale_fill_manual(values = c("HT" = "darkorange", "LT" = "black", "T0" = "blue"), guide = FALSE) +
  ggtitle("Pigments")+
    theme(axis.text.y = element_text(colour = "black", size = 11, face = "bold"), 
    axis.text.x = element_text(colour = "black", face = "bold", size = 11), 
    axis.title.y = element_text(face = "bold", size = 12), 
    legend.text = element_text(size = 10, face ="bold", colour ="black"), 
    legend.position = "none", 
    legend.justification='right',
    legend.box.margin=unit(0, "cm"),
        legend.direction='horizontal',
    legend.title = element_text(size = 12, colour = "black", face = "bold"), 
    axis.title.x = element_text(face = "bold", size = 12, colour = "black"), 
    panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
    legend.key=element_blank(), 
      plot.title = element_text(vjust = -8, hjust = 0.02))

```

diatom-phaeo composition_genus 2022-04-17
```{r}
#https://europepmc.org/article/pmc/7287063
diatoms <- filter (norm_proteomicsdata, grepl("Bacill|", class)) #do this for Bacillar and Prymnes separately
diatoms <- diatoms [c(12, 24:53)]

diatoms <- diatoms %>% mutate_at (2:31, funs (((. / sum(.)))))

diatoms <- melt(diatoms, id.vars=c("F")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

diatoms$genus_a <-   ifelse (diatoms$F ==  "Polar-centric-Mediophyceae", "Centric diatoms",
                            ifelse (diatoms$F ==  "Raphid-pennate", "Pennate diatoms",  diatoms$F))
#ifelse (diatoms$F ==  "Radial-centric-basal-Coscinodiscophyceae", "Centric diatoms",
#ifelse (diatoms$F ==  "Araphid-pennate", "Pennate diatoms", diatoms$F
                           
# diatoms$genus_a <-   ifelse (diatoms$G ==  "Thalassiosira", "Thalassiosira", 
#                             ifelse (diatoms$G ==  "Chaetoceros", "Chaetoceros", 
#                             ifelse (diatoms$G ==  "Fragilariopsis", "Fragilariopsis",
#                             ifelse (diatoms$G ==  "Pseudo-nitzschia", "Pseudo-nitzschia",
#                             ifelse (diatoms$G ==  "Araphid-pennate", "Araphid-pennate",
#                             ifelse (diatoms$G ==  "Polar-centric-Mediophyceae", "Other Polar-centric-Mediophyceae",
#                             ifelse (diatoms$G ==  "Raphid-pennate", "Other Raphid-pennate",  
#                             ifelse (diatoms$G ==  "Bacillariophyta_X", "Ambiguous Diatom",   
#                             "Other Diatoms"))))))))

diatoms_aggregated <- aggregate(diatoms$norm_abundance, by=list(Category=diatoms$genus_a, diatoms$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(genus = Category) 

diatoms_aggregated <- diatoms_aggregated %>% separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment",  "replicate"), sep = "_")

diatoms_aggregated$treatment <- paste (diatoms_aggregated$temperature_treatment, diatoms_aggregated$iron_treatment, sep ='_')

diatoms_aggregated$treatment2 <- ifelse (diatoms_aggregated$treatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (diatoms_aggregated$treatment ==  "LT_Fe", "LT_Fe",
                            ifelse (diatoms_aggregated$treatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (diatoms_aggregated$treatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))


#phaocystis 

phaeo <- filter (norm_proteomicsdata, grepl("Prymnesio|", class)) #do this for Bacillar and Prymnes separately
phaeo <- phaeo [c(12, 24:53)]
phaeo <- phaeo %>% mutate_at (2:31, funs (((. / sum(.)))))
phaeo <- melt(phaeo, id.vars=c("F")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

phaeo$genus_a <-   ifelse (phaeo$F ==  "Phaeocystaceae", "Phaeocystis",
                            phaeo$F)
                           
phaeo_aggregated <- aggregate(phaeo$norm_abundance, by=list(Category=phaeo$genus_a, phaeo$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(genus = Category) 

phaeo_aggregated <- phaeo_aggregated %>% separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment",  "replicate"), sep = "_")

phaeo_aggregated$treatment <- paste (phaeo_aggregated$temperature_treatment, phaeo_aggregated$iron_treatment, sep ='_')

phaeo_aggregated$treatment2 <- ifelse (phaeo_aggregated$treatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (phaeo_aggregated$treatment ==  "LT_Fe", "LT_Fe",
                            ifelse (phaeo_aggregated$treatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (phaeo_aggregated$treatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))

diatoms_phae <- rbind(diatoms_aggregated, phaeo_aggregated)


diatomgroups <- ggplot ((filter(diatoms_aggregated, grepl("Pennate|Centric", genus))), aes(
  x = factor(treatment2, level = tempFe_order),
  y = norm_abundance, 
  color = genus)) +  
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 5,  position= position_dodge(width=0.4))+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1,  position= position_dodge(width=0.4))+ 
  facet_grid(.~ bioassay )+
  theme_bw()+
  ylab (expression(paste("Fraction contribution to total proteins")))+
  #ylab ("Fraction contribution to total diatom proteins")+
  xlab ("") +
  scale_y_log10()+
  scale_color_manual(name="",
                     values = c("Centric diatoms" = "grey50",
                                "Pennate diatoms" = "black"))+ 
 theme(axis.text.x=element_text(face = "bold", size = 11, color = "black", vjust = 1, hjust = 1,  angle = 45),
        axis.title.y=element_text(size=14,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 10, face = "bold"), 
        strip.text.x = element_text(size = 11, face = "bold"), 
       legend.position = "top", 
       legend.title = element_blank(),
       legend.justification = "left", 
       legend.box.margin=margin(-10,-10,-10,-10))+ 
       #plot.margin = unit(c(0.25,0.25,0.25,0), units = "cm"))+
  guides(shape = guide_legend(override.aes = list(size=5)))#this locks the legend in position so it doesnt move around as you zoom in and out of the plot)


phaeogroups <- ggplot ((filter(phaeo_aggregated, grepl("Phaeocystis", genus))), aes(
  x = factor(treatment2, level = tempFe_order),
  y = norm_abundance, 
  color = genus)) +  
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, alpha = 0.5, position= position_dodge(width=0.4))+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1,  position= position_dodge(width=0.4))+ 
  facet_grid(.~ bioassay )+
  theme_bw()+
  ylab (expression(paste("Fraction contribution to total proteins")))+
  #ylab ("Fraction contribution to total diatom proteins")+
  xlab ("") +
  scale_y_log10(labels = label_number((accuracy = 0.01)))+
  scale_color_manual(name="",
                     values = c("Phaeocystis" = "#CC79A7"))+ 
 theme(axis.text.x=element_text(face = "bold", size = 11, color = "black", vjust = 1, hjust = 1,  angle = 45),
        axis.title.y=element_text(size=14,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 10, face = "bold"), 
        strip.text.x = element_text(size = 11, face = "bold"), 
       legend.position = "top", 
       legend.title = element_blank(),
       legend.justification = "left", 
       legend.box.margin=margin(-10,-10,-10,-10))+ 
       #plot.margin = unit(c(0.25,0.25,0.25,0), units = "cm"))+
  guides(shape = guide_legend(override.aes = list(size=5))) 
  #scale_y_continuous(labels = label_number((accuracy = 0.01)))
#In BA2, if there was more in-situ iron, then the LT + and - Fe might be too different, but the HT might exacerbate Fe deficiency because things would have grown more to use the Fe. 


```

Si:NO3
```{r}
percentchange <- read.csv("LJ_ps117_si_consumption.csv", header = T)
percentchange_2 <- filter (percentchange, !grepl ("0|2|4", daycount) )

fig2_Sidrawdown <-  ggplot(percentchange_2, 
       aes(x=iron_treatment, 
       y=SiNO3ratio,
       color = temperature_treatment, 
       shape = iron_treatment))+ 
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 4, stroke = 1.5, alpha = 0.65)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1.3, width = 0.1) +
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("LT", "HT"),
                     values = c('black', 'darkorange2','black')) +
  scale_shape_manual(name = "Iron",
                       breaks = c("Fe","noFe", "T0"), #this removes the T0 from the legend
                       labels = c("Fe", "noFe", "T0"),
                       values = c(19,1,5)) +
  
  scale_x_discrete(limits=c("noFe","Fe"))+
  ylab (expression (Si:NO[3]~drawdown)) + 
  xlab (expression ("")) +
  theme_bw()+
  theme(axis.text.x=element_text(face = "bold", size = 11, color = "black"),
        axis.title.y=element_text(size=14, color = "black", face = "bold"), 
        axis.text.y=element_text(face = "bold", size = 10, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 11, face = "bold"),
        legend.position = "none")+
    facet_wrap(~bioassay)



```

legend
```{r}
data.scores$BAFe <- paste (data.scores$bioassay, data.scores$iron_treatment, sep = "_")

legend <- ggplot(data.scores, aes(x = NMDS1,
                        y = NMDS2))+
  geom_point(size=5,
             aes(shape = BAFe,
                 color = temperature_treatment) ) +
                                     
    scale_color_manual(values = c("HT" = "darkorange", "LT" = "black", "T0" = "darkgreen"),
                     name = "Temperature")+
 
      scale_shape_manual( name = "Iron",
                      values=c("BA1_noFe" = 1,
                               "BA1_Fe" = 19,
                               "BA2_noFe" = 2,
                               "BA2_Fe" = 17))+

    theme(legend.text = element_text(size = 9, face ="bold", colour ="black"),
    legend.position = "right",
    #legend.justification = "right",
    #legend.margin=unit(0,"cm"),
    legend.title = element_text(size = 10, colour = "black", face = "bold"),
    legend.key=element_blank(),
    legend.margin=margin(l = 0.2,unit ="cm"))+
   guides(shape = guide_legend(override.aes = list(size=4.5)))+ 
 guides(colour = guide_legend(override.aes = list(size=4.5)))
 

legend2 <- get_legend(legend)
fig2_legend <- as_ggplot(legend2)

fig2_legend
```

plot 
```{r}
hm <- plot_grid(fig2_pigmentmeans, fig2_proteintopmeans, fig2_proteinbottommeans, 
          ncol=1,
          #labels = c("A ","B", ""), 
          align = "v", 
          rel_heights = c(1.25,0.65,1.1))

nmds <- plot_grid(NULL, fig2_pigment_nmds,  
          NULL, fig2_FC_nmds,  
          NULL, fig2_peptide_nmds, 
          ncol=2, 
          rel_widths = c(0.05,1), 
          rel_heights = c(1,1,1))
          #labels = c("C")) 
  
col3 <- plot_grid (diatomgroups, phaeogroups, fig2_Sidrawdown, 
                   ncol = 1, 
                   align = "v", axis = "l", 
                  rel_heights =  c(1,1,0.8))
                  # labels = c("D","E", "F"))

plot_grid(hm, nmds, NULL, col3, fig2_legend,
          ncol=5, 
          rel_widths = c(3.7,2.3,0.2,2.8,0.6))
```


#Figure 3
To do - get DE for BA2, 
combine BA1 and BA2 DE, 
Plot temperature and Fe De
metals
```{r}
alldata$PFe57_nM <-  ifelse (alldata$PFe57_nM ==  "0.454", NA, 
                            alldata$PFe57_nM)

metals_Fe <- fig3_metals (PFe57_nM, PFe[57]:PON) 
metals_Mn <- fig3_metals (PMn_nM, PMn:PON)
metals_Cu <- fig3_metals (PCu_nM, PCu:PON)
metals_Zn <- fig3_metals (PZn_nM, PZn:PON)
metals_Co <- fig3_metals (PCo_nM, PCo:PON~(M:M))
metals_Ni <- fig3_metals (PNi_nM, PNi:PON)
```

DE_foldchange
```{r}
BA1 <- read.csv("ps117_BA1_taxonF_DE_20220912.csv", header = T)
colnames(BA1)[3:12] <- paste("BA1", colnames(BA1[,c(3:12)]), sep = "_")

BA2 <- read.csv("ps117_BA2_taxonF_DE_20220913.csv", header = T)
colnames(BA2)[3:12] <- paste("BA2", colnames(BA2[,c(3:12)]), sep = "_")

BA_DE <- merge (BA1, BA2, by = c("cluster", "taxon_F"), all = TRUE)
                                 
BA_DE2 <- filter (BA_DE, grepl ("centric|pennate|Phaeocyst", taxon_F))

BA_DE2$cluster <-   ifelse (BA_DE2$cluster ==  "clust_343", "ISIP-2A", 
                   ifelse (BA_DE2$cluster ==  "clust_411", "Nitrate reductase", 
                            ifelse (BA_DE2$cluster ==  "clust_1154", "ISIP-3", 
                            ifelse (BA_DE2$cluster ==  "clust_1814", "ISIP-1", 
                            ifelse (BA_DE2$cluster ==  "clust_1820", "Plastocyanin", 
                            ifelse (BA_DE2$cluster ==  "clust_534", "Flavodoxin", 
                            #ifelse (BA_DE2$cluster ==  "clust_4660", "Flavodoxin-1", 
                            ifelse (BA_DE2$cluster ==  "clust_5562", "Cu-Zn SOD", 
                            ifelse (BA_DE2$cluster ==  "clust_55", "PSI PsaA/PsaB", 
                            ifelse (BA_DE2$cluster ==  "clust_808", "Ferredoxin", 
                            ifelse (BA_DE2$cluster ==  "clust_2934", "Multicopper Oxidase", 
                            ifelse (BA_DE2$cluster ==  "clust_891", "Glutamate Synthase", 
                            ifelse (BA_DE2$cluster ==  "clust_332", "Chl A-B binding", 
                            ifelse (BA_DE2$cluster ==  "clust_5925", "Chl A-B binding", 
                            ifelse (BA_DE2$cluster ==  "clust_1044", "Chl A-B binding",
                            ifelse (BA_DE2$cluster ==  "clust_195", "Chl A-B binding",
                            ifelse (BA_DE2$cluster ==  "clust_417", "AMT", 
                            ifelse (BA_DE2$cluster ==  "clust_2501", "Fe-Mn SOD", 
                            ifelse (BA_DE2$cluster ==  "clust_1525", "SLC-4",
                            ifelse (BA_DE2$cluster ==  "clust_231", "Rhodopsin",
                            ifelse (BA_DE2$cluster ==  "clust_1906", "Mn-cluster",
                            ifelse (BA_DE2$cluster ==  "clust_812", "ZIP",
                            ifelse (BA_DE2$cluster ==  "clust_16168", "TonB",
                            ifelse (BA_DE2$cluster ==  "clust_10646", "TonB_2",
                            ifelse (BA_DE2$cluster ==  "clust_1372", "Urea transporter",
                            ifelse (BA_DE2$cluster ==  "clust_320", "Urea carboxylase",
                            ifelse (BA_DE2$cluster ==  "clust_6019", "Ornithine cyclodeaminase",    
                            ifelse (BA_DE2$cluster ==  "clust_655", "CA_clust_655",
                            ifelse (BA_DE2$cluster ==  "clust_6573", "CA_clust_6573", 
                            ifelse (BA_DE2$cluster ==  "clust_35", "Ribosome recycling",
                            ifelse (BA_DE2$cluster ==  "clust_2490", "Rubisco",
                            ifelse (BA_DE2$cluster ==  "clust_2244", "PEPC",
                            ifelse (BA_DE2$cluster ==  "clust_1830", "Acireductone dioxygenase Fe containing",
                            ifelse (BA_DE2$cluster ==  "clust_2088", "Fasciclin",
                            ifelse (BA_DE2$cluster ==  "clust_613", "Cyt b6",
                            ifelse (BA_DE2$cluster ==  "clust_2815", "PSI RC",
                            ifelse (BA_DE2$cluster ==  "clust_42", "PSII RC, D1",
                            ifelse (BA_DE2$cluster ==  "clust_1978", "PSII PsbU",
                            ifelse (BA_DE2$cluster ==  "clust_498", "PSII CP47",
                            ifelse (BA_DE2$cluster ==  "clust_1497", "Cyt f",
                            ifelse (BA_DE2$cluster ==  "clust_1886", "SLC",
                            BA_DE2$cluster 
                    ))))))))))))))))))))))))))))))))))))))))

cluster_order <- c('ISIP-1', 'ISIP-2A', 'ISIP-3', 'Fasciclin','Cu-Zn SOD', 'Fe-Mn SOD', 'Nitrate reductase', 'Plastocyanin',  'Rhodopsin', 'Flavodoxin', 'Ferredoxin', "Cyt b6", 'Cyt f', "Chl A-B binding",'PSI RC','PSI PsaA/PsaB', "PSII PsbU", 'PSII CP47','PSII RC, D1', 'Mn-cluster' )

#"clust_87", "clust_888", "clust_1702", "clust_12215", "clust_4938" 'Urea carboxylase', 'Urea transporter',

BA_DE3 <- filter (BA_DE2, grepl (paste(cluster_order, collapse = "|"), cluster)) 

BA_DE3$BA1_ironDE <-  ifelse (BA_DE3$BA1_Fe_FDR < 0.05, "*", "")
BA_DE3$BA1_tempDE <-  ifelse (BA_DE3$BA1_noFe_HT_LT_FDR < 0.05, "*", "")
BA_DE3$BA1_FextempDE <-  ifelse (BA_DE3$BA1_Fetempinteraction_FDR < 0.05, "*", "")


BA_DE3$BA2_ironDE <-  ifelse (BA_DE3$BA2_Fe_FDR < 0.05, "*", "")
BA_DE3$BA2_tempDE <-  ifelse (BA_DE3$BA2_noFe_HT_LT_FDR < 0.05, "*", "")
BA_DE3$BA2_FextempDE <-  ifelse (BA_DE3$BA2_Fetempinteraction_FDR < 0.05, "*", "")


DE_slide <- function (effect, significance, title) {
  ggplot (BA_DE3, aes(x= {{effect}}, y= factor(cluster, level = cluster_order),
                   color = taxon_F, 
                   shape = {{significance}}))+ 
                    geom_point (size = 3 , alpha=1, stroke = 1.5) + 
  # geom_text(aes (label = {{significance}}), show.legend = FALSE, size = 8, nudge_y = 0.2)+
    scale_color_manual(name="",
                       breaks = c("centric", "pennate","Phaeocystaceae"),
                       labels = c("Centric diatoms", "Pennate diatoms", "Phaeocystis"),
                       values = c("black", 'grey60', 'darkgreen')) +
    
    scale_shape_manual(name="",
                       breaks = c("*", ""),
                       values = c(19,1), 
                       guide = "none") +
    ylab("")+ 
    xlab(expression (paste("Log"[2]~fold~change)))+
    ggtitle(title)+
    theme_bw()+
    theme(plot.title = element_text(hjust = 0, size = 10, color = "black", face = "bold"))+
    theme(axis.text.x = element_text(size = 10, face = "bold", color = "black"),
          axis.title.x = element_text(size = 10, face = "bold", color = "black"), 
          axis.text.y = element_text(size = 10,face = "bold", color="black"), 
          legend.text = element_text(size = 9, color = "black", face = "bold")) +
          geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )}

BA1_Fe <- DE_slide (BA1_Fe_logFC, BA1_ironDE, "BA1 Fe Effect" )+theme (legend.position = c(0.95,0), legend.justification = c(0.95,0), legend.key.size = unit (0.4, "cm")) + guides(colour = guide_legend(override.aes = list(size=2))) + theme(legend.margin=margin(t=0, r=0, b=0, l=0, unit="cm")) 

BA1_temp <- DE_slide (BA1_noFe_HT_LT_logFC, BA1_tempDE, "BA1 Temperature Effect" ) + theme(legend.position = "none") 

BA2_Fe <- DE_slide (BA2_Fe_logFC, BA2_ironDE, "BA2 Fe Effect" ) + theme(legend.position = "none") 

BA2_temp <- DE_slide (BA2_noFe_HT_LT_logFC, BA2_tempDE, "BA2 Temperature Effect" ) + theme(legend.position = "none")

DE_slide (BA1_Fetempinteraction_logFC, BA1_FextempDE, "BA1 FexTemperature Effect" ) + theme(legend.position = "none") 

```


point size test
```{r}
#combine clusters first 


#combine DE and norm protein data 
BA_DE2$F <- BA_DE2$taxon_F
clustsize <- merge (norm_proteomicsdata, BA_DE2, by = c("cluster", "F"))

clustsize2 <- clustsize[c(1,2,37,38)]
clustsize2$BA1mean <- rowMeans(clustsize2[,c(3,4)])

#combine clusters


ggplot(clustsize2, aes(x=BA1mean)) + 
  geom_histogram(color = "black", fill = "white") 
 #geom_histogram(aes(y=..density..), colour="black", fill="white")+
 #geom_density(alpha=.2, fill="#FF6666") +
  scale_x_log10()
#get cluster abundance 
```


legends
```{r}
##legend
x<- c(1,2,3)
y <- c("a", "b", "c")
colr <- c (-1.2,0, 1.7)
z <- data.frame(x,y, colr)

legend3b <- ggplot(z, aes (x,y, fill = colr))+
   geom_point()+
    scale_fill_gradientn(colors = c("deepskyblue2","white","darkred"),
         values = rescale(c(-1.2,0,1.7)),
         guide = "colorbar", limits=c(-1.2,1.7),
         name = "Row Z-score")+
   theme(legend.text = element_text(face = "bold", size=13),
        legend.title = element_text (face = "bold", size = 13, angle = 90), 
        legend.title.align=0.5)+
   guides(fill = guide_colourbar(title.position = "left", barwidth = 1.5, barheight = 15))
  

legend3b <- get_legend(legend3b)
legend3b <- as_ggplot(legend3b)


###

#temperature legend 
metallegend <- ggplot((filter(alldata, grepl("8", daycount))) , aes(
  y = (PNi_nM/(1000*PON_uM)),
  x = iron_treatment,
  color = temperature_treatment))+
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     values = c( 'black','darkorange2', "black")) +
  stat_summary(fun = mean, geom = "point", size = 4, stroke = 1, alpha = 0.5)+
  theme_bw() + 
  theme(legend.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 12))

metallegend <- get_legend(metallegend)
metallegend <- as_ggplot(metallegend)
```

plot
```{r}
a <- plot_grid(metals_Fe, metals_Mn, metals_Cu, metals_Zn,
          nrow = 4, 
          align = 'hv')

b <- plot_grid(BA1_Fe,BA2_Fe,  BA1_temp,  BA2_temp,
          nrow = 2, 
          align = 'hv')
  
plot_grid(a, NULL, b, 
          nrow = 1, 
          rel_widths = c(0.35,0.01,1))
```


fasciclin
```{r}
fas <- read.table ("fasciclin_environmental_parameters.csv")

```

heatmaps
```{r}
protein_clusters_hm ("pennate", "ISIP|Rhodop|Flavodo|Ferredo|PSI|Nitrate reductase", "BA1" )
protein_clusters_hm ("centric", "ISIP|Rhodop|Flavodo|Ferredo|PSI|Nitrate reductase", "BA1" )
protein_clusters_hm ("Phaeo", "ISIP|Rhodop|Flavodo|Ferredo|PSI|Nitrate reductase", "BA1" )
```



#Figure 4 
This includes an exploration of ribosomes and chloroplast proteins

N:P
```{r}
PONPOP <- ggplot((filter(alldata, grepl("0|8", daycount))) , aes(
    x = as.numeric(temperature_C),
    y = ((1000*PON_uM)/POP_nM),
    shape = bioassayFe, 
    color = bioassayFe))+
 
   #geom_point(size = 4, alpha=0.65, stroke = 1.5)+
 scale_color_manual(name= "Bioassay",
                breaks = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                labels = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
               values = c('darkgreen', "black","black", "darkgreen", "black","black")) +
   stat_summary(fun = mean, geom = "point", show.legend = T, size = 4, stroke = 1.5, alpha = 0.65)+
   stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F,  size = 1, width = 0.1)+
    scale_shape_manual(name = "Bioassay",
                       breaks = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                       labels = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                       values = c(1, 19, 1, 2, 17, 2)) +
    ylab (expression (PON:POP~(M:M)))+
    xlab (expression("")) +
    theme_bw()+
    theme(axis.text.x=element_text(face = "bold", size = 12, color = "black", vjust = 0.5, hjust = 0.5),
          axis.title.x=element_text(size=16, color = "black"),
          axis.title.y=element_text(size=16, color = "black"),
          axis.text.y=element_text(face = "bold", size = 12, color = "black"),
          legend.position = "none",
          legend.title = element_text(size = 11, face = "bold"),
          legend.text = element_text(size = 10), 
          strip.text = element_text(size = 11, face = "bold"))+
   expand_limits(y = 0) 
  

  



npdrawdown <- read.csv("LJ_ps117_NO3_PO4_consumption.csv")
npdrawdown$bioassayFe <- paste (npdrawdown$bioassay, npdrawdown$iron_treatment, sep = "_")


npdrawdown$temperature <- ifelse (npdrawdown$bioassay ==  "BA1" & npdrawdown$temperature_treatment ==  "T0",  -0.3, 
                         ifelse (npdrawdown$bioassay ==  "BA1" & npdrawdown$temperature_treatment ==  "HT",  1.7, 
                         ifelse (npdrawdown$bioassay ==  "BA1" & npdrawdown$temperature_treatment ==  "LT",  -0.3, 
                         ifelse (npdrawdown$bioassay ==  "BA2" & npdrawdown$temperature_treatment ==  "T0",  -1.4, 
                         ifelse (npdrawdown$bioassay ==  "BA2" & npdrawdown$temperature_treatment ==  "HT",  1, 
                         ifelse (npdrawdown$bioassay ==  "BA2" & npdrawdown$temperature_treatment ==  "LT",  -1, 
                         "Other"))))))




NPdrawdown <- ggplot((filter(npdrawdown, grepl("8", daycount))) , aes(
    x = as.numeric(temperature),
    y = (total_Ndrawdown/total_Pdrawdown),
    shape = bioassayFe, 
    color = bioassayFe))+
   #geom_point(size = 4, alpha=0.65, stroke = 1.5)+
   scale_color_manual(name= "Bioassay",
                breaks = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                labels = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
               values = c('darkgreen', "black","black", "darkgreen", "black","black")) +
   stat_summary(fun = mean, geom = "point", show.legend = T, size = 4, stroke = 1.5, alpha = 0.65)+
   stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1, width = 0.1)+
    scale_shape_manual(name = "Bioassay",
                       breaks = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                       labels = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                       values = c(1, 19, 1, 2, 17, 2)) +
    ylab (expression (NO[3]:PO[4]~drawdown~(M:M))) + 
   #xlab (expression("Temperature Â°C")) +  
  xlab (expression("")) +  
  theme_bw()+
    theme(axis.text.x=element_text(face = "bold", size = 12, color = "black", vjust = 0.5, hjust = 0.5),
          axis.title.x=element_text(size=16, color = "black"),
          axis.title.y=element_text(size=16, color = "black"),
          axis.text.y=element_text(face = "bold", size = 12, color = "black"),
          legend.position = "none",
          legend.title = element_text(size = 11, face = "bold"),
          legend.text = element_text(size = 10), 
          strip.text = element_text(size = 11, face = "bold"))+
   expand_limits(y = 0) 
```

Ribosome + compartments 2022-07-28
```{r}
#to choose which proteins are 'ribosomes' I used the grpnorm_compartment annotation only. anything that's ribosome in there is not in a cluster. adding the clusters annotated as ribosomes doesn't do much really
#ribosome$newannotation <- paste(ribosome$cluster_annotation, ribosome$grpnorm_compartment, sep ="_") 
compartment <- norm_proteomicsdata [c(23:53)]

compartment <- melt(compartment, id.vars=c("grpnorm_compartment")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value ) 


compartment_aggregated <- aggregate(compartment$norm_abundance, by=list(Category=compartment$treatment, compartment$grpnorm_compartment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(compartment = Group.2) %>%
                  rename(treatment = Category) %>%
separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

compartment_aggregated$fulltreatment <- paste (compartment_aggregated$temperature_treatment, compartment_aggregated$iron_treatment , sep ='_')

compartment_aggregated$fulltreatment <- ifelse (compartment_aggregated$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (compartment_aggregated$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (compartment_aggregated$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (compartment_aggregated$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))


compartment_aggregated$temperature <- ifelse (compartment_aggregated$bioassay ==  "BA1" & compartment_aggregated$temperature_treatment ==  "T0",  -0.3, 
                         ifelse (compartment_aggregated$bioassay ==  "BA1" & compartment_aggregated$temperature_treatment ==  "HT",  1.7, 
                         ifelse (compartment_aggregated$bioassay ==  "BA1" & compartment_aggregated$temperature_treatment ==  "LT",  -0.3, 
                         ifelse (compartment_aggregated$bioassay ==  "BA2" & compartment_aggregated$temperature_treatment ==  "T0",  -1.4, 
                         ifelse (compartment_aggregated$bioassay ==  "BA2" & compartment_aggregated$temperature_treatment ==  "HT",  1, 
                         ifelse (compartment_aggregated$bioassay ==  "BA2" & compartment_aggregated$temperature_treatment ==  "LT",  -1, 
                         "Other"))))))

compartment_aggregated$fulltreatments <- paste (compartment_aggregated$bioassay, compartment_aggregated$iron_treatment, sep = "_")

# compartments<- ggplot(filter(compartment_aggregated, !grepl("ambi|mito",compartment)), aes(x = factor(fulltreatment, level = treatment_orders), 
#                 y=norm_abundance,
#                 color = compartment)) + 
#   facet_wrap(~bioassay)+
#   stat_summary(fun = mean, geom = "point", size = 4, stroke = 1.8, alpha = 0.5)+
#   stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F,  size = 1.3, width = 0.1) +
#      scale_color_manual(
#              name= "Compartment",
#               breaks = c(" chloro ", " ribosomal ", " nuclear ", " mito "),
#              labels = c("Chloroplast", "Ribosomal", "Nuclear", "Mictochonrial"),
#               values = c('darkgreen', '#377A98', "grey", "black")) +
#   ylab (expression (bold("Metaproteome mass fraction" ))) + 
#   xlab (expression ("")) +
#   theme_bw()+
#   scale_y_log10()+
#   theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
#         axis.text.x=element_text(face = "bold", size = 8, color = "black"),
#         axis.title.y=element_text(size=14, color = "black", face = "bold"), 
#         axis.text.y=element_text(face = "bold", size = 15, color = "black"),
#         strip.background =element_rect(fill = "white"),
#         strip.text.x = element_text(size = 11, face = "bold"),
#         legend.title = element_text(size = 11, face = "bold"),
#         legend.text = element_text(size = 10))


#metap

ribosomes <- ggplot(filter(compartment_aggregated, grepl("ribo",compartment)) , aes(
    x = as.numeric(temperature),
    y = norm_abundance,
     shape = fulltreatments, 
    color = fulltreatments))+
  
   #geom_point(size = 4, alpha=0.65, stroke = 1.5)+
  #scale_y_log10()+
scale_color_manual(name= "Bioassay",
                breaks = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                labels = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
               values = c('darkgreen', "black","black", "darkgreen", "black","black")) +
   stat_summary(fun = mean, geom = "point", show.legend = T, size = 4, stroke = 1.5, alpha = 0.65)+
   stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1, width = 0.1)+
    scale_shape_manual(name = "Bioassay",
                       breaks = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                       labels = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                       values = c(1, 19, 1, 2, 17, 2)) +
    ylab (substitute("Metaproteome \n ribosomal protein fraction")) +
    xlab (expression("Temperature Â°C")) +
    theme_bw()+
    theme(axis.text.x=element_text(face = "bold", size = 12, color = "black", vjust = 0.5, hjust = 0.5),
          axis.title.x=element_text(size=16, color = "black"),
          axis.title.y=element_text(size=16, color = "black"),
          axis.text.y=element_text(face = "bold", size = 12, color = "black"),
          legend.position = "none",
          legend.title = element_text(size = 11, face = "bold"),
          legend.text = element_text(size = 10), 
          strip.text = element_text(size = 11, face = "bold"))+
   expand_limits(y = c(0,0.2)) 


chloro <- ggplot(filter(compartment_aggregated, grepl("chloro",compartment)) , aes(
    x = as.numeric(temperature),
    y = norm_abundance,
    shape = fulltreatments, 
    color = fulltreatments))+
  
   #geom_point(size = 4, alpha=0.65, stroke = 1.5)+
  scale_color_manual(name= "Bioassay",
                breaks = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                labels = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
               values = c('darkgreen', "black","black", "darkgreen", "black","black")) +
     stat_summary(fun = mean, geom = "point", show.legend = T, size = 4, stroke = 1.5, alpha = 0.65)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1, width = 0.1)+
  scale_shape_manual(name = "Bioassay",
                       breaks = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                       labels = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                       values = c(1, 19, 1, 2, 17, 2)) +
      ylab (substitute("Metaproteome \n chloroplast protein fraction")) +
    xlab (expression("Temperature Â°C")) +
    theme_bw()+
    theme(axis.text.x=element_text(face = "bold", size = 12, color = "black", vjust = 0.5, hjust = 0.5),
          axis.title.x=element_text(size=16, color = "black"),
          axis.title.y=element_text(size=16, color = "black"),
          axis.text.y=element_text(face = "bold", size = 12, color = "black"),
          legend.position = "none",
          legend.title = element_text(size = 11, face = "bold"),
          legend.text = element_text(size = 10), 
          strip.text = element_text(size = 11, face = "bold"))+
   expand_limits(y = c(0, 0.6)) +
  scale_y_continuous(labels = function(x) format(x, nsmall = 2))

```

taxon-specific compartments 2022-07-28
```{r}

phaeo_ribosome <- taxon_compartment_temp("Phaeocyst", 'ribo', "Phaeocystis \n ribosomal protein fraction") + xlab("") + expand_limits(y = c(0,0.2))
phaeo_chloro <- taxon_compartment_temp("Phaeocyst", 'chloro', "Phaeocystis \n chloroplast protein fraction") + scale_y_continuous(labels = function(x) format(x, nsmall = 2)) + expand_limits(y = c(0,0.2))

pennate_ribosome <- taxon_compartment_temp("pennate", 'ribo', "Pennate diatom \n ribosomal protein fraction") +xlab("") + expand_limits(y = c(0,0.2))
pennate_chloro <- taxon_compartment_temp("pennate", 'chloro', "Pennate diatom \n chloroplast protein fraction") + scale_y_continuous(labels = function(x) format(x, nsmall = 2)) + expand_limits(y = c(0,0.6))

centric_ribosome <- taxon_compartment_temp("centric", 'ribo', "Centric diatom \n ribosomal protein fraction") +xlab("") + expand_limits(y = c(0,0.2)) +    xlab (expression("Temperature Â°C")) 

centric_chloro <- taxon_compartment_temp("centric", 'chloro', "Centric diatom \n chloroplast protein fraction") + scale_y_continuous(labels = function(x) format(x, nsmall = 2)) + expand_limits(y = c(0,0.6))

### older function for plotting
taxon_compartment <- function (taxon, proteincompartment, yaxislabel) {
ribosometaxon <- filter (norm_proteomicsdata, grepl(taxon, F))

ribosometaxon <- ribosometaxon [c(12, 23 ,24:53)]

ribosometaxon2 <- ribosometaxon %>% mutate_at (3:32, funs (((. / sum(.)))))

ribosometaxon3 <- melt(ribosometaxon2, id.vars=c("grpnorm_compartment", "F")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

ribosometaxon_aggregated <- aggregate(ribosometaxon3$norm_abundance, by=list(Category=ribosometaxon3$grpnorm_compartment, ribosometaxon3$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(compartment = Category) %>%
          separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

ribosometaxon_aggregated$fulltreatment <- paste (ribosometaxon_aggregated$temperature, ribosometaxon_aggregated$iron , sep ='_')
ribosometaxon_aggregated$fulltreatment <- ifelse (ribosometaxon_aggregated$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (ribosometaxon_aggregated$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (ribosometaxon_aggregated$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (ribosometaxon_aggregated$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))

ribosometaxon_aggregated <- filter (ribosometaxon_aggregated, !grepl("T0", fulltreatment))

ggplot(filter(ribosometaxon_aggregated, grepl(proteincompartment,compartment)), 
             aes(x = factor(fulltreatment, level = tempFe_order), 
                y=norm_abundance, 
                color = temperature_treatment, 
                shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  stat_summary(fun = mean, geom = "point", size = 4, stroke = 1.5, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F,  size = 1, width = 0.1) +
  scale_shape_manual(name = "Iron",
                       breaks = c("Fe","noFe", "T0"), #this removes the T0 from the legend
                       labels = c("Fe", "noFe", "T0"),
                       values = c(19,1,5)) +
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     values = c( 'black','darkorange2', "black")) +
  xlab (expression ("")) +
  ylab (substitute(yaxislabel)) +

  theme_bw()+
  theme(axis.text.x=element_text(face = "bold", size = 10, color = "black"),
        axis.title.y=element_text(size=16, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 12, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 12, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14), 
        legend.position = "none")+
  expand_limits(y = 0)

}

```

legend
```{r}
fig3legend <- ggplot((filter(alldata, grepl("0|8", daycount))) , aes(
    x = as.numeric(temperature_C),
    y = ((1000*PON_uM)/POP_nM),
    shape = bioassayFe, 
    color = bioassayFe))+
   geom_point(size = 4, alpha=0.5, stroke = 1.5)+
  scale_color_manual(name= "",
                breaks = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                labels = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
               values = c('darkgreen', "black","black", "darkgreen", "black","black")) +
     scale_shape_manual(name = "",
                       breaks = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                       labels = c("BA1_T0", "BA1_Fe", "BA1_noFe", "BA2_T0", "BA2_Fe", "BA2_noFe"),
                       values = c(1, 19, 1, 2, 17, 2)) +
   theme_bw()+
             theme(legend.title = element_text(size = 13, face = "bold"),
          legend.text = element_text(size = 10), 
          legend.position = "top", 
          legend.direction='horizontal')  +
  guides(colour = guide_legend(nrow = 1))
 


legend3 <- get_legend(fig3legend)
fig3legend <- as_ggplot(legend3)

fig3legend
```

plot
```{r}
x <- plot_grid(NPdrawdown, NULL,  phaeo_ribosome,
          PONPOP, NULL,  pennate_ribosome,
          ribosomes, NULL,  centric_ribosome,
          nrow = 3,  
           rel_widths = c(1, 0.05, 1), 
          align = 'hv')
                      

plot_grid (fig3legend, x,
           ncol  = 1,
           rel_heights  = c(0.05,1), 
           align = 'hv')


### chloroplasts and ribosomes
# fullplot <- plot_grid(PONPOP, NULL, ribosomes, NULL,   phaeo_ribosome, NULL, pennate_ribosome, NULL, centric_ribosome,
#           NPdrawdown, NULL, chloro,NULL,  phaeo_chloro, NULL, pennate_chloro, NULL, centric_chloro, 
#           nrow = 2, 
#           align = 'hv',
#           rel_widths = c(1, 0.05, 1, 0.1,1,0.05,1,0.05,1))
# 
# 
# plot_grid (fullplot, fig3legend, 
#            rel_widths = c(1,0.08))

```



#Figure 5
```{r}
metals_Cd <- fig3_metals (PCd_nM, PCd:PON)

CA_centric <- protein_clusters ("centric|Bacill", "CA_clust_6573", protein = c("CA_clust_6573"), shape = c(19,15)) + ggtitle ("Centric diatoms - clust_6573") + theme(legend.position = "none") 

CA_pennate <- protein_clusters ("pennate", "CA_clust_655", protein = c("CA_clust_655"), shape = c(19,15)) + ggtitle ("Pennate diatoms - clust_655") + theme(legend.position = "none") 

CA_phaeo <- protein_clusters ("Phaeo", "CA_clust_655", protein = c("CA_clust_655"), shape = c(19,15)) + ggtitle ("Phaeocystis - clust_655") + theme(legend.position = "none")

#protein_clusters ("pennate", "Cation Efflux protein", protein = c("Cation Efflux protein"), shape = c(19,15)) look into efflux proteins

#protein_clusters_hm ("centric|pennate", "Carbonic|CA", "BA1")

#protein_clusters_hm ("centric|pennate", "Carbonic|CA", "BA2")

#Maybe look atr SLC4 ? 

```

plot
```{r}
plot_grid( metals_Cd,  CA_centric,  
           CA_pennate, CA_phaeo, 
          nrow = 4, 
        
          align = 'hv')
         

#and heatmap
```


#Misc 
proteins (see chucnk with same name below without a function) 2022-08-30
```{r}

protein_clusters ("centric|pennate|Bacillariop", "Plastocyanin", protein = c("Plastocyanin", "Fe-Mn SOD"), shape = c(1,4))

protein_clusters ("centric|pennate|Bacillariop", "", "Fe-Mn|Mn-clus" )

protein_clusters ("centric|pennate|Bacillariop", "", "ISIP" ) +geom_point(size =2, position= position_dodge(width=0.3) )
Mnproteins_phaeo <- protein_clusters ("Phaeo|Prymn", "", "Fe-MnSOD|Mn-clus")

Cuproteins_diatoms <- protein_clusters ("centric|pennate|Bacillariop", "", "Plastocy|Cu-Zn" )
Cuproteins_phaeo <- protein_clusters ("Phaeo|Prymn", "", "Plastocy|Cu-Zn")

Znproteins_diatoms <- protein_clusters ("centric|pennate|Bacillariop", "", "ZIP|RNA" )
Znproteins_phaeo <- protein_clusters ("Phaeo|Prymn", "", "ZIP|RNA")

Cdproteins_diatoms <- protein_clusters ("centric|pennate|Bacillariop", "", "CA" )
Cdproteins_phaeo <- protein_clusters ("Phaeo|Prymn", "", "CA")

```

legend
```{r}
x<- c(1,2,3)
y <- c("a", "b", "c")
colr <- c (-1.2,0, 1.7)
z <- data.frame(x,y, colr)

legend3b <- ggplot(z, aes (x,y, fill = colr))+
   geom_point()+
    scale_fill_gradientn(colors = c("deepskyblue2","white","darkred"),
         values = rescale(c(-1.2,0,1.7)),
         guide = "colorbar", limits=c(-1.2,1.7),
         name = "Row Z-score")+
   theme(legend.text = element_text(face = "bold", size=13),
        legend.title = element_text (face = "bold", size = 13, angle = 90), 
        legend.title.align=0.5)+
   guides(fill = guide_colourbar(title.position = "left", barwidth = 1.5, barheight = 15))
  

legend3b <- get_legend(legend3b)
legend3b <- as_ggplot(legend3b)
```

proteins heatmap
```{r}
protein_clusters_hm ("pennate", "Fe-Mn|Mn-clus|Anhyd", "BA" )
protein_clusters ("Phaeo|Prymn", "", "Fe-Mn|Mn-clus|Anhyd", "BA2")

Cuproteins_centric <- protein_clusters ("centric", "Plastocy|Cu-Zn", "BA" )
Cuproteins_pennate <- protein_clusters ("pennate", "Plastocy|Cu-Zn", "BA" )
Cuproteins_phaeo <- protein_clusters ("Phaeo|Prymn", "Plastocy|Cu-Zn", "BA")

Znproteins_diatoms <- protein_clusters ("centric|pennate|Bacillariop", "", "ZIP|RNA" )
Znproteins_phaeo <- protein_clusters ("Phaeo|Prymn", "", "ZIP|RNA")

Cdproteins_diatoms <- protein_clusters ("centric|pennate|Bacillariop", "", "CA" )
Cdproteins_phaeo <- protein_clusters ("Phaeo|Prymn", "", "CA")

```

plot
```{r}
row1 <- plot_grid(NULL, metals_Mn, NULL, Mnproteins_diatoms, NULL, Mnproteins_phaeo, 
          nrow = 1,
          rel_widths = c(0.05, 1, 0.1, 1.3, 0.05, 1.3),
          labels = c("", "A", "", "B", "", "C"), 
          align = 'h')
row2 <- plot_grid(NULL, metals_Cu, NULL, Cuproteins_diatoms, NULL, Cuproteins_phaeo, 
          nrow = 1,
          rel_widths = c(0.05, 1, 0.1, 1.3, 0.05, 1.3),
          labels = c("", "D", "", "E", "", "F"), 
          align = 'h')
row3 <- plot_grid(NULL, metals_Zn, NULL, Znproteins_diatoms, NULL, Znproteins_phaeo, 
          nrow = 1,
          rel_widths = c(0.05, 1, 0.1, 1.3, 0.05, 1.3),
          labels = c("", "G", "", "H", "", "I"), 
          align = 'h')

plot_grid(row1, row2, row3, 
          nrow = 3)
```

plot_v2
```{r}
row1 <- plot_grid(NULL, metals_Mn, NULL, Mnproteins_diatoms, NULL, Mnproteins_phaeo, 
          nrow = 1,
          rel_widths = c(0.05, 1, 0.1, 1.3, 0.05, 1.3),
          labels = c("", "A", "", "B", "", "C"), 
          align = 'h')

plot_grid(metals_Cu, Cuproteins_diatoms, Cuproteins_phaeo, 
          nrow = 3, align = 'hv', 
          rel_heights = c(1,0.5,0.5), 
          axis = 'l')


           
row3 <- plot_grid(NULL, metals_Zn, NULL, Znproteins_diatoms, NULL, Znproteins_phaeo, 
          nrow = 1,
          rel_widths = c(0.05, 1, 0.1, 1.3, 0.05, 1.3),
          labels = c("", "G", "", "H", "", "I"), 
          align = 'h')


plot_grid(row1, row2, row3, 
          nrow = 3)
```






#DE data
```{r}
BA1 <- read.csv("ps117_BA1_taxonF_DE_20220912.csv", header = T)

#BA2 <- read.csv("ps117_BA2_DE_20220824.csv", header = T)

#colnames(BA1)[2:11] <- paste("BA1", colnames(BA1[,c(2:11)]), sep = "_")

#colnames(BA2)[2:11] <- paste("BA2", colnames(BA2[,c(2:11)]), sep = "_")

#DE <- merge (BA1, BA2, by = 'genes', all = TRUE) %>%  separate(genes, into=c("cluster", "F"), sep = ";")

#DE_2 <- merge (DE, norm_proteomicsdata, by = c('cluster', 'F'))


#DE[is.na(DE)] <- 0


x<- ggplot (BA1, aes (x = Fe_logFC, 
                  y = -log10(Fe_FDR), 
        color = cluster))+
  geom_point(size = 2.5, alpha = 0.5)+
  geom_hline (yintercept = 1.301, col = "blue", lty = 2, lwd=0.8 )+
  geom_vline (xintercept = c(-1,1), col = "blue", lty = 2, lwd=0.8 )+
  ggtitle("Fe effect")+
   xlab("Log2 FC")+
   ylab("-log10 FDR")+ 
    theme_bw()+
    theme(axis.title.x = element_text (color= "black", face = "bold", size = 16), 
          axis.title.y = element_text (color= "black", face = "bold", size = 16),
          legend.position = "none")
          
  
ggplotly(x) 
```

```{r}
BA1_2 <- filter (BA1, grepl ("centric|pennate|Phaeocyst", taxon_F))

BA1_2$cluster <-   ifelse (BA1_2$cluster ==  "clust_343", "ISIP-2A", 
                   ifelse (BA1_2$cluster ==  "clust_411", "Nitrate reductase", 
                            ifelse (BA1_2$cluster ==  "clust_534", "Flavodoxin", 
                            ifelse (BA1_2$cluster ==  "clust_1154", "ISIP-3", 
                            ifelse (BA1_2$cluster ==  "clust_1814", "ISIP-1", 
                            ifelse (BA1_2$cluster ==  "clust_1820", "Plastocyanin", 
                            ifelse (BA1_2$cluster ==  "clust_4660", "Flavodoxin-1", 
                            ifelse (BA1_2$cluster ==  "clust_5562", "Cu-Zn SOD", 
                            ifelse (BA1_2$cluster ==  "clust_55", "PSI", 
                            ifelse (BA1_2$cluster ==  "clust_808", "Ferredoxin", 
                            ifelse (BA1_2$cluster ==  "clust_2934", "Multicopper Oxidase", 
                            ifelse (BA1_2$cluster ==  "clust_891", "Glutamate Synthase", 
                            ifelse (BA1_2$cluster ==  "clust_332", "Chl A-B binding protein", 
                            ifelse (BA1_2$cluster ==  "clust_5925", "Chl A-B binding protein2", 
                            ifelse (BA1_2$cluster ==  "clust_417", "AMT", 
                            ifelse (BA1_2$cluster ==  "clust_2501", "Fe-Mn SOD", 
                            ifelse (BA1_2$cluster ==  "clust_1525", "SLC-4",
                            ifelse (BA1_2$cluster ==  "clust_231", "Rhodopsin",
                            ifelse (BA1_2$cluster ==  "clust_1906", "Mn-cluster",
                            ifelse (BA1_2$cluster ==  "clust_812", "ZIP",
                            ifelse (BA1_2$cluster ==  "clust_16168", "TonB",
                            ifelse (BA1_2$cluster ==  "clust_10646", "TonB_2",
                            ifelse (BA1_2$cluster ==  "clust_1372", "Urea transporter",
                            ifelse (BA1_2$cluster ==  "clust_320", "Urea carboxylase",
                            ifelse (BA1_2$cluster ==  "clust_6019", "Ornithine cyclodeaminase",    
                            ifelse (BA1_2$cluster ==  "clust_655", "CA_clust_655",
                            ifelse (BA1_2$cluster ==  "clust_6573", "CA_clust_6573", 
                            ifelse (BA1_2$cluster ==  "clust_35", "Ribosome recycling",
                            ifelse (BA1_2$cluster ==  "clust_2490", "Rubisco",
                            ifelse (BA1_2$cluster ==  "clust_2244", "PEPC",
                            ifelse (BA1_2$cluster ==  "clust_1830", "Acireductone dioxygenase Fe containing",
                            ifelse (BA1_2$cluster ==  "clust_2088", "Fasciclin",
                                    BA1_2$cluster 
                    ))))))))))))))))))))))))))))))))

cluster_order <- c('ISIP-1', 'ISIP-2A', 'ISIP-3', 'Fasciclin','Cu-Zn SOD','Fe-Mn SOD', 'Plastocyanin',  'Rhodopsin', 'Flavodoxin', 'Flavodoxin-1', 'Ferredoxin', "clust_613", 'Nitrate reductase', "Chl A-B binding protein2", "Chl A-B binding protein",'clust_1044', 'clust_195', 'clust_2815', 'PSI', 'clust_1497', 'Mn-cluster', 'Urea transporter', 'Urea carboxylase', 'SLC-4' )

BA1_3 <- filter (BA1_2, grepl (paste(cluster_order, collapse = "|"), cluster)) 

BA1_3$iron_shapes <-  ifelse (BA1_3$Fe_FDR < 0.05, "*", "")

ggplot (BA1_3, aes(x= Fe_logFC, y= factor(cluster, level = cluster_order),
                   color = taxon_F))+ 
                   #shape = iron_shapes)) +
    geom_point (size = 5, alpha=0.7, stroke = 1.5) + 
   geom_text(aes (label = iron_shapes), show.legend = FALSE, size = 8, nudge_y = 0.2)+
  
  scale_color_manual(name="",
                       breaks = c("centric", "pennate","Phaeocystaceae"),
                       labels = c("Centric diatoms", "Pennate Diatoms", "Phaeocystis"),
                       values = c("black", 'grey', '#9E3C0E')) +
    ylab("")+ 
    xlab(expression (paste("Log"[2]~fold~change)))+
    ggtitle("Iron Effect")+
    theme_bw()+
    theme(plot.title = element_text(hjust = 0.5))+
    theme(axis.text.x = element_text(size = 12, face = "bold", color = "black"),
          axis.title.x = element_text(size = 16, face = "bold", color = "black"), 
          axis.text.y = element_text(size = 12,face = "bold", color="black"), 
          
          plot.title = element_text(hjust = 0, face = "bold", color ="black")) +
    # theme(legend.position = "top")+
    geom_vline (xintercept = c(0), col = "black", lty = 2, lwd=0.3 )



```






#proteomics taxonomy
domains (eukaryotes, bacteria, archae, virus etc.) 2022-04-17
```{r}
all_domains <- norm_proteomicsdata [c(1,7, 24:53)]

all_domains <- melt(all_domains, id.vars=c("peptide", "domain" )) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value)

all_domains_aggregated <- aggregate(all_domains$norm_abundance,by=list(Category=all_domains$domain, all_domains$treatment), FUN=sum)%>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(domain = Category)

all_domains_aggregated <- all_domains_aggregated %>% separate(treatment, into=c("bioassay", "timepoint" , "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

all_domains_aggregated$treatment <- paste (all_domains_aggregated$temperature, all_domains_aggregated$iron, sep ='_')

all_domains_aggregated$domain_a <- ifelse (all_domains_aggregated$domain ==  "Archaea", "Archaea", 
                            ifelse (all_domains_aggregated$domain ==  "Viruses", "Viruses",
                            ifelse (all_domains_aggregated$domain ==  "Bacteria", "Bacteria",
                            ifelse (all_domains_aggregated$domain ==  "Eukaryota", "Eukaryotes",
                            ifelse (all_domains_aggregated$domain ==  "unassigned_tax", "No annotation",
                           ifelse (all_domains_aggregated$domain ==  "ambiguous_domain", "Ambiguous",
                            "Other"))))))


all_domains_aggregated$domain_a <- reorder(all_domains_aggregated$domain_a , all_domains_aggregated$norm_abundance )



#T0
#png("domain_taxa_T0.png", width=40*0.75, height=30*0.75, units="cm", res=600) 

ggplot ((filter(all_domains_aggregated, grepl("T0", treatment))), aes(
  x = domain_a,
  y = norm_abundance)) +  
  geom_point (size = 5, alpha = 0.5)+
  #stat_summary(fun = mean, geom = "point", show.legend = F, size = 5, stroke = 1.8, alpha = 0.5)+
  #stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1.3, width = 0.1)+
  facet_grid(.~ bioassay)+
  theme_bw()+
  theme (axis.text.x = element_text(angle = 90)) +
  ylab ("Normalized abundance")+
  xlab ("") +
  scale_y_log10(labels = function(x) format(x, scientific = FALSE, drop0trailing = TRUE))+ #this ensures that the decimal points are exact every time. 
    #scale_y_log10()+ 
  theme(axis.text.x=element_text(face = "bold", size = 18, color = "black", vjust = 1, hjust = 1, angle = 45),
        axis.title.y=element_text(size=20,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 15), 
        strip.text.x = element_text(size = 18, face = "bold"))+
   expand_limits(y = 0:1)  
#dev.off()


#png("domain_taxa_alltreatments.png", width=40*0.75, height=30*0.75, units="cm", res=600) 

ggplot (all_domains_aggregated, aes(
  x = factor(treatment, level = treatment_order),
  y = norm_abundance, 
  color = domain_a)) +  
  geom_point (size = 5, alpha = 0.5)+
  #stat_summary(fun = mean, geom = "point", show.legend = F, size = 5, stroke = 1.8, alpha = 0.5)+
  #stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1.3, width = 0.1)+
  facet_grid(.~ bioassay)+
  theme_bw()+
  theme (axis.text.x = element_text(angle = 90)) +
  ylab ("normalized abundance")+
  xlab ("") +
  scale_y_log10(labels = function(x) format(x, scientific = FALSE, drop0trailing = TRUE))+
  scale_color_manual(name="",
                       values = c("Archaea" = "#E69F00",
                     "Viruses" = "#0072B2",
                     "No annotation" = "grey", 
                     "Bacteria" = "#CC79A7", 
                     "Ambiguous" = "black", 
                     "Eukaryotes" = "#D55E00"))+ 
  theme(axis.text.x=element_text(face = "bold", size = 18, color = "black", vjust = 1, hjust = 1, angle = 45),
        axis.title.y=element_text(size=20,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 15), 
        strip.text.x = element_text(size = 18, face = "bold"))+
    expand_limits(y = 0:1)  
#dev.off()


```

amongthe eukaryotes, we'll focus on the most abundant classes (Bacillariophytes, Haptophytes etc.) 2022-04-17
```{r}
####### 
#because we show in the proteomics data that diatoms and haptophytes are the most common Eukaryotes by far, we'll focus on them here. 
#The following bit of code shows that Bacillariophyta (diatoms) and Prymnesiophytes (haptopytes) are by far the most common euks
all_class_euks <- filter (norm_proteomicsdata, grepl('Euk',domain)) 
#all_class_euks$class <- gsub(" ","", as.character (all_class_euks$class)) #removes extra white space from taxonomy names

#let's see which are the most common 'classes' of eukaryotes 
all_class_euks2 <- all_class_euks [c(1,10, 24:53)]

all_class_euks3 <- all_class_euks2 %>% mutate_at (3:32, funs (((. / sum(.))))) #normalize the abundance to total eukaryotes



all_class_euks3 <- melt(all_class_euks3, id.vars=c("peptide", "class")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value ) 

all_class_euks2_aggregated <- aggregate(all_class_euks3$norm_abundance, by=list(Category=all_class_euks3$class, all_class_euks3$treatment), FUN=sum)%>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(class = Category)

all_class_euks2_aggregated <- all_class_euks2_aggregated %>% separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

all_class_euks2_aggregated$treatment <- paste (all_class_euks2_aggregated$temperature, all_class_euks2_aggregated$iron, sep ='_')

all_class_euks2_aggregated$class <- reorder(all_class_euks2_aggregated$class , all_class_euks2_aggregated$norm_abundance )


all_class_euks2_aggregated2 <- filter(all_class_euks2_aggregated, norm_abundance > 0.005)

#might want to display on top most abundant groups
ggplot ((filter(all_class_euks2_aggregated2, grepl("T0", treatment))), aes(
  x = class,
  y = norm_abundance))  +  
  geom_point (size = 5, alpha = 0.5)+
  facet_grid(bioassay ~.)+
  theme_bw()+
  theme (axis.text.x = element_text(angle = 90)) +
  ylab ("Fraction contribution to eukaryote proteins T0")+
  xlab ("") +
  #expand_limits(y = 30) + 
  theme(axis.text.x=element_text(face = "bold", size = 12, color = "black", vjust = 1, hjust = 1, angle = 45),
        axis.title.y=element_text(size=20,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 15), 
        strip.text.x = element_text(size = 18, face = "bold"))

############

all_class <- all_class_euks [c(1,10, 24:53 )]

all_class$class_a <- ifelse (all_class$class ==  "Bacillariophyta", "Diatoms", 
                            ifelse (all_class$class ==  "Haptophyta", "Haptophytes",
                            ifelse (all_class$class ==  "Prymnesiophyceae", "Haptophytes",
                            ifelse (all_class$class ==  "Pavlovophyceae", "Haptophytes",
                                                       "Other"))))

all_class_normalized <- all_class %>% mutate_at (3:32, funs (((. / sum(.)))))

all_class_normalized <- all_class_normalized [c(1, 33, 3:32)]

all_class_normalized <- melt(all_class_normalized, id.vars=c("peptide", "class_a")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value ) 

all_class_normalized_aggregated <- aggregate(all_class_normalized$norm_abundance, by=list(Category=all_class_normalized$class_a, all_class_normalized$treatment), FUN=sum)%>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(class = Category)

all_class_normalized_aggregated <- all_class_normalized_aggregated %>% separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

all_class_normalized_aggregated$treatment <- paste (all_class_normalized_aggregated$temperature, all_class_normalized_aggregated$iron, sep ='_')


#png("diatom_hapto_alltreatments.png", width=40*0.75, height=30*0.75, units="cm", res=600) 

ggplot ((filter(all_class_normalized_aggregated, !grepl("Other", class))), aes(
  x = factor(treatment, level = treatment_order),
  y = norm_abundance, 
  shape = class)) +  
  #geom_point (size = 5, alpha = 0.5, stroke = 1.8)+
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.5, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1.3, width = 0.1)+ 
  facet_grid(.~ bioassay)+
  theme_bw()+
  theme (axis.text.x = element_text(angle = 90)) +
  ylab ("Fraction contribution to all eukaryote proteins")+
  xlab ("") +
  scale_shape_manual(name="",
                     values = c("Diatoms" = 5, 
                                "Haptophytes" = 16, 
                                "Dynophytes" = 4))+  
 theme(axis.text.x=element_text(face = "bold", size = 18, color = "black", vjust = 1, hjust = 1, angle = 45),
        axis.title.y=element_text(size=20,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 15), 
        strip.text.x = element_text(size = 18, face = "bold"))+
   expand_limits(y = 0:0.8)  

#dev.off()
```

amongthe bacteria, we'll focus on the most abundant classes 2022-08-08
```{r}
####### 
#because we show in the proteomics data that diatoms and haptophytes are the most common Eukaryotes by far, we'll focus on them here. 
#The following bit of code shows that Bacillariophyta (diatoms) and Prymnesiophytes (haptopytes) are by far the most common euks
all_class_euks <- filter (norm_proteomicsdata, grepl('Bact',domain)) 
#all_class_euks$class <- gsub(" ","", as.character (all_class_euks$class)) #removes extra white space from taxonomy names

#let's see which are the most common 'classes' of eukaryotes 
all_class_euks2 <- all_class_euks [c(1,10, 24:53)]

all_class_euks3 <- all_class_euks2 %>% mutate_at (3:32, funs (((. / sum(.))))) #normalize the abundance to total eukaryotes



all_class_euks3 <- melt(all_class_euks3, id.vars=c("peptide", "class")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value ) 

all_class_euks2_aggregated <- aggregate(all_class_euks3$norm_abundance, by=list(Category=all_class_euks3$class, all_class_euks3$treatment), FUN=sum)%>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(class = Category)

all_class_euks2_aggregated <- all_class_euks2_aggregated %>% separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

all_class_euks2_aggregated$treatment <- paste (all_class_euks2_aggregated$temperature, all_class_euks2_aggregated$iron, sep ='_')

all_class_euks2_aggregated$class <- reorder(all_class_euks2_aggregated$class , all_class_euks2_aggregated$norm_abundance )


all_class_euks2_aggregated2 <- filter(all_class_euks2_aggregated, norm_abundance > 0.005)

#might want to display on top most abundant groups
ggplot ((filter(all_class_euks2_aggregated2, grepl("T0", treatment))), aes(
  x = class,
  y = norm_abundance))  +  
  geom_point (size = 5, alpha = 0.5)+
  facet_grid(bioassay ~.)+
  theme_bw()+
  theme (axis.text.x = element_text(angle = 90)) +
  ylab ("Fraction contribution to eukaryote proteins T0")+
  xlab ("") +
  #expand_limits(y = 30) + 
  theme(axis.text.x=element_text(face = "bold", size = 12, color = "black", vjust = 1, hjust = 1, angle = 45),
        axis.title.y=element_text(size=20,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 15), 
        strip.text.x = element_text(size = 18, face = "bold"))


ggplot ((filter(all_class_euks2_aggregated2, grepl("Gamma", class))), aes(
  x = iron_treatment,
  y = norm_abundance, 
  color = temperature_treatment))  +  
  #geom_point (size = 5, alpha = 0.5)+
  facet_grid(bioassay ~.)+
  theme_bw()+
 stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.5, alpha = 0.5)+ 
  theme (axis.text.x = element_text(angle = 90)) +
  ylab ("Fraction contribution to eukaryote proteins T0")+
  xlab ("") +
  #expand_limits(y = 30) + 
  theme(axis.text.x=element_text(face = "bold", size = 12, color = "black", vjust = 1, hjust = 1, angle = 45),
        axis.title.y=element_text(size=20,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 15), 
        strip.text.x = element_text(size = 18, face = "bold"))

############

all_class <- all_class_euks [c(1,10, 24:53 )]

all_class$class_a <- ifelse (all_class$class ==  "Bacillariophyta", "Diatoms", 
                            ifelse (all_class$class ==  "Haptophyta", "Haptophytes",
                            ifelse (all_class$class ==  "Prymnesiophyceae", "Haptophytes",
                            ifelse (all_class$class ==  "Pavlovophyceae", "Haptophytes",
                                                       "Other"))))

all_class_normalized <- all_class %>% mutate_at (3:32, funs (((. / sum(.)))))

all_class_normalized <- all_class_normalized [c(1, 33, 3:32)]

all_class_normalized <- melt(all_class_normalized, id.vars=c("peptide", "class_a")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value ) 

all_class_normalized_aggregated <- aggregate(all_class_normalized$norm_abundance, by=list(Category=all_class_normalized$class_a, all_class_normalized$treatment), FUN=sum)%>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(class = Category)

all_class_normalized_aggregated <- all_class_normalized_aggregated %>% separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

all_class_normalized_aggregated$treatment <- paste (all_class_normalized_aggregated$temperature, all_class_normalized_aggregated$iron, sep ='_')


#png("diatom_hapto_alltreatments.png", width=40*0.75, height=30*0.75, units="cm", res=600) 

ggplot ((filter(all_class_normalized_aggregated, !grepl("Other", class))), aes(
  x = factor(treatment, level = treatment_order),
  y = norm_abundance, 
  shape = class)) +  
  #geom_point (size = 5, alpha = 0.5, stroke = 1.8)+
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.5, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1.3, width = 0.1)+ 
  facet_grid(.~ bioassay)+
  theme_bw()+
  theme (axis.text.x = element_text(angle = 90)) +
  ylab ("Fraction contribution to all eukaryote proteins")+
  xlab ("") +
  scale_shape_manual(name="",
                     values = c("Diatoms" = 5, 
                                "Haptophytes" = 16, 
                                "Dynophytes" = 4))+  
 theme(axis.text.x=element_text(face = "bold", size = 18, color = "black", vjust = 1, hjust = 1, angle = 45),
        axis.title.y=element_text(size=20,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 15), 
        strip.text.x = element_text(size = 18, face = "bold"))+
   expand_limits(y = 0:0.8)  

#dev.off()
```

diatom composition_broad groups 2022-04-17
```{r}
#https://europepmc.org/article/pmc/7287063

diatoms <- filter (norm_proteomicsdata, grepl("Bacillar", class))

diatoms <- diatoms [c(12, 24:53)]


diatoms <- diatoms %>% mutate_at (2:31, funs (((. / sum(.)))))

diatoms <- melt(diatoms, id.vars=c("F")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

diatoms_aggregated <- aggregate(diatoms$norm_abundance, by=list(Category=diatoms$F, diatoms$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(genus = Category) 

diatoms_aggregated <- diatoms_aggregated %>% separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment",  "replicate"), sep = "_")

diatoms_aggregated$treatment <- paste (diatoms_aggregated$temperature, diatoms_aggregated$iron, sep ='_')

#!grepl("Bacill|Araphid|Radial"

#png("diatom_hapto_alltreatments.png", width=40*0.75, height=30*0.75, units="cm", res=600) 
ggplot ((filter(diatoms_aggregated, !grepl("nothing", genus))), aes(
  x = factor(treatment, level = treatment_order),
  y = norm_abundance, 
  fill = genus)) +  
  #geom_point (size = 5, alpha = 0.5)+
  #stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, alpha = 0.5)+
  #stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1.3, width = 0.1)+ 
  facet_grid(.~ bioassay)+
  theme_bw()+
  theme (axis.text.x = element_text(angle = 90)) +
  ylab ("contribution to total diatom proteins")+
  xlab ("") +
   geom_bar(position = "fill", stat = "identity") + #position = "fill" does percentage, stack does raw values
  scale_fill_manual(name="",
                     breaks = c("Raphid-pennate", "Araphid-pennate", "Polar-centric-Mediophyceae", "Radial-centric-basal-Coscinodiscophyceae", "Bacillariophyta_X"),
                     labels = c("Pennate - raphid", "Pennate - araphid", "Centric - polar Mediophyceae", "Centric - radial basal Coscinodiscophyceae", "Ambiguous diatom"),
                     values = c('darkorange2', 'brown', 'green', 'darkgreen', 'black')) +
   
 theme(axis.text.x=element_text(face = "bold", size = 18, color = "black", vjust = 1, hjust = 1, angle = 45),
        axis.title.y=element_text(size=20,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 15), 
        strip.text.x = element_text(size = 18, face = "bold"))
#dev.off()

#png("diatom_class.png", width=40*0.75, height=30*0.75, units="cm", res=600) 

ggplot ((filter(diatoms_aggregated, grepl("Polar|Raphid", genus))), aes(
  x = factor(iron_treatment, level = Fetreatment_order),
  y = norm_abundance, 
  shape = genus)) +  
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1.3, width = 0.1)+ 
  facet_grid(.~ bioassay)+
  theme_bw()+
  theme (axis.text.x = element_text(angle = 90)) +
  ylab ("Fraction contribution to all diatom proteins")+
  xlab ("") +
  scale_shape_manual(name="",
                     values = c("Raphid-pennate" = 15, 
                                "Polar-centric-Mediophyceae" = 16))+ 
 theme(axis.text.x=element_text(face = "bold", size = 18, color = "black", vjust = 1, hjust = 1, angle = 45),
        axis.title.y=element_text(size=20,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 15), 
        strip.text.x = element_text(size = 18, face = "bold"))
#dev.off()  
```


Haptophyte composition_genus 2022-04-17
```{r}
#https://europepmc.org/article/pmc/7287063

haptophytes <- filter (norm_proteomicsdata, grepl("Haptoph", C))

haptophytes <- haptophytes [c(13, 24:53)]

haptophytes2 <- haptophytes %>% mutate_at (2:31, funs (((. / sum(.)))))

haptophytes <- melt(haptophytes2, id.vars=c("G")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

haptophytes$genus <-   ifelse (haptophytes$G ==  "Phaeocystis", "Phaeocystis", 
                              "Other haptophytes")


haptophytes_aggregated <- aggregate(haptophytes$norm_abundance, by=list(Category=haptophytes$genus, haptophytes$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(genus = Category) 


haptophytes_aggregated <- haptophytes_aggregated %>% separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment",  "replicate"), sep = "_")


haptophytes_aggregated$treatment <- paste (haptophytes_aggregated$temperature_treatment, haptophytes_aggregated$iron_treatment, sep ='_')


haptophytes_aggregated$treatment2 <- ifelse (haptophytes_aggregated$treatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (haptophytes_aggregated$treatment ==  "LT_Fe", "LT_Fe",
                            ifelse (haptophytes_aggregated$treatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (haptophytes_aggregated$treatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))
  
fig_haptophytespecies <- ggplot ((filter(haptophytes_aggregated, grepl("Phaeo", genus))), aes(
  x = factor(treatment2, level = treatment_orders),
  y = norm_abundance, 
  shape = genus)) +  
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 4, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1.1, width = 0.1)+ 
  facet_grid(.~ bioassay )+
  theme_bw()+
  ylab (expression(paste("Phaeocystis sp. mass fraction contribution to \n total Haptophyte proteins")))+
  #ylab ("Fraction contribution to total diatom proteins")+
  xlab ("") +
  theme(axis.text.x=element_text(face = "bold", size = 11, color = "black", vjust = 1, hjust = 1,  angle = 45),
        axis.title.y=element_text(size=14,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 10, face = "bold"), 
        strip.text.x = element_text(size = 11, face = "bold"), 
       legend.position = "none", 
       legend.title = element_blank(),
       legend.justification = "left", 
       legend.box.margin=margin(-10,-10,-10,-10))+ 
       #plot.margin = unit(c(0.25,0.25,0.25,0), units = "cm"))+
  guides(shape = guide_legend(override.aes = list(size=5))) #this locks the legend in position so it doesnt move around as you zoom in and out of the plot)  


#In BA2, if there was more in-situ iron, then the LT + and - Fe might be too different, but the HT might exacerbate Fe deficiency because things would have grown more to use the Fe. 

#dev.off()
```

show similar groups to pigment data 2022-04-19
```{r}
euks <- filter (norm_proteomicsdata, grepl("Euka", domain))


class <- euks [c(10, 24:53)]

class$class <- gsub(" ","", as.character (class$class)) #removes extra white space from taxonomy names


class <- class %>% mutate_at (2:31, funs ((100*(. / sum(.))))) #normalize the abundance to total eukaryotes


class2 <- melt(class, id.vars=c("class")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

class2$class_a <-   ifelse (class2$class ==  "Bacillariophyta", "Diatoms", 
                                 
                    ifelse (class2$class ==  "Dinophyceae", "Dinoflagellates", 
                    ifelse (class2$class ==  "Dinophyta", "Dinoflagellates",
                                    
                    ifelse (class2$class ==  "Haptophyta", "Haptophytes",
                    ifelse (class2$class ==  "Prymnesiophyceae", "Haptophytes",
                    ifelse (class2$class ==  "Pavlovophyceae", "Haptophytes",
                                    
                    ifelse (class2$class ==  "Cryptophyceae", "Cryptophytes",
                    ifelse (class2$class ==  "Cryptophyta", "Cryptophytes",
                                    
                    ifelse (class2$class ==  "Pelagophyceae", "Pelagophytes",
                                    
                    ifelse (class2$class ==  "Chlorophyceae", "Prasinophytes",
                    ifelse (class2$class ==  "Chlorophyta", "Prasinophytes",
                    ifelse (class2$class ==  "Pyramimonadales", "Prasinophytes",
                    ifelse (class2$class ==  "Prasinophyceae", "Prasinophytes",
                            
                            "Other")))))))))))))

class2_aggregated <- aggregate(class2$norm_abundance, by=list(Category=class2$class_a, class2$treatment), FUN=sum)%>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(class = Category) 
                 

class2_aggregated <- class2_aggregated %>% separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")


class2_aggregated$treatment <- paste (class2_aggregated$temperature_treatment, class2_aggregated$iron_treatment, sep ='_')

class2_aggregated$class <- reorder(class2_aggregated$class , class2_aggregated$norm_abundance )

#png("proteome_taxon_fraction_20220329.png", width=40*0.75, height=30*0.75, units="cm", res=600) 

ggplot ((filter(class2_aggregated, !grepl(" asxa", class))), aes(
  x = factor(treatment, level = treatment_order),
  #x = factor(iron_treatment, level = Fetreatment_order),
  y = norm_abundance, 
  fill = class))+ 
  #color = class)) +  
  #geom_point(size = 4)+
  geom_bar(position = "fill", stat = "identity") + #position = "fill" does percentage, stack does raw values
  facet_grid(.~ bioassay)+
theme_bw()+
   theme (axis.text.x = element_text(angle = 90)) +
   ylab ("Contribution to protein abundance (fraction)")+
    xlab ("")+
   scale_fill_manual(name="",
                      values = c("Diatoms" = "#117733",
                                 "Dinoflagellates" = "#0072B2",
                                "Haptophytes" = "#CC79A7", 
                                 "Prasinophytes" = "#88CCEE", 
                                 "Cryptophytes" = "grey", #000000", 
                                 "Pelagophytes" = "#E69F00", 
                                "Other" = "black"))+
  theme(axis.text.x=element_text(face = "bold", size = 18, color = "black", vjust = 1, hjust = 1, angle = 45),
        axis.title.y=element_text(size=20,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 15), 
        strip.text.x = element_text(size = 18, face = "bold"))
  
#dev.off()

###
###
#T0 protein heatmap 

proteinheatmap <- filter (class2_aggregated, grepl("T0", treatment)) 

proteinheatmap$full_treatment <- paste(proteinheatmap$bioassay, proteinheatmap$replicate, sep = "_")


proteinheatmap1 <- proteinheatmap [c(1,7,9)]

proteinheatmap2 <- dcast(proteinheatmap1, class ~ full_treatment, value.var = "norm_abundance")

proteinheatmap2 <- data.frame(proteinheatmap2[,-1], row.names = proteinheatmap2 [,1])

proteinheatmap3 <- proteinheatmap2 [c(1,3:7)]

proteinheatmap3 <- proteinheatmap3 [c(1,2,3,4,5,7),]

proteinheatmap3$BA1 <- rowMeans(proteinheatmap3[,c(1:3)])
proteinheatmap3$BA2 <- rowMeans(proteinheatmap3[,c(4:6)])
proteinheatmap4 <- proteinheatmap3 [c(7,8)]
  
  
par(cex.main=6, cex.lab=1, cex.axis=1.5) 
heatmap.2 (as.matrix(proteinheatmap4/100), 
             trace = "none",
             density = "none", 
             col = greenred(100),
             breaks = seq(0,0.6, length.out = 101),
             symbreaks = FALSE,#changes the scale of the color key
             cexRow = 1.8, 
             cexCol =1.8,
             margins = c(8,20),
             scale = "none" ,
             dendrogram = 'none',  
             Colv = FALSE, 
             Rowv = FALSE,
             keysize = 0.75,
             key.title = NA,
             rowsep = c(1,2,3,4,5),
             colsep=c(3),
             sepcolor = "white",
             sepwidth = c(0.1, 0.05),
             na.rm = FALSE,
             na.color = "grey") #just need to figure out how to dcast NAs
```


#proteomics functional groups
all proteins clusters heatmap 
```{r}
#clusters <- filter (norm_proteomicsdata, grepl("pennate|centric|Bacill", F)) 

clusters <- filter (norm_proteomicsdata, grepl("Phaeo|Prymn", F)) #clust_5911 , Cd carbonic anhydrase

#|^clust_3290$

#clust_999 transaminase, also involved in methionine recylcling

clusters <- clusters [c(12, 3, 24:53)]

clusters2 <- clusters %>% mutate_at (3:32, funs (((. / sum(.)))))

clusters3 <- filter(clusters2, grepl("clust_2934|^clust_891$|^clust_332$|^clust_808$|^clust_55$|^clust_4660|^clust_1820$|^clust_1154$|^clust_343$|^clust_1814$|^clust_534$|^clust_5562$|^clust_417$|^clust_411$|clust_2501|clust_6573|^clust_1525$|clust_1552|^clust_231$|^clust_1906$|^clust_812$|^clust_16168$|^clust_10646$|^clust_1372$|^clust_320$|^clust_6019$|^clust_35$|^clust_1830$|^clust_655$|^clust_2490$|^clust_1156$|^clust_2244$", cluster)) 

clusters3 <- melt(clusters3, id.vars=c("cluster", "F")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


clusters3_aggregated <- aggregate(clusters3$norm_abundance, by=list(Category=clusters3$cluster, clusters3$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(cluster = Category) %>% 
                    separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

clusters3_aggregated$fulltreatment <- paste (clusters3_aggregated$temperature, clusters3_aggregated$iron , sep ='_')
clusters3_aggregated$fulltreatment <- ifelse (clusters3_aggregated$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))


# ggplot(clusters3_aggregated, 
#              aes(x = factor(fulltreatment, level = treatment_orders), 
#                 y=norm_abundance, 
#                 color = cluster)) + 
#   facet_wrap(~bioassay)+
#   geom_point(size=3)+
#   #stat_summary(fun = mean, geom = "point", size = 6, stroke = 1, alpha = 0.5)+
#   #stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F,  size = 1.3, width = 0.1) +
#   ylab (expression ("Fraction contribution to total diatom proteins" )) + 
#   xlab (expression ("")) +
#   theme_bw()+
#   theme(axis.text.x=element_text(face = "bold", size = 15, color = "black"),
#         axis.title.y=element_text(size=20, color = "black"), 
#         axis.text.y=element_text(face = "bold", size = 15, color = "black"),
#         strip.background =element_rect(fill = "white"),
#         strip.text.x = element_text(size = 15, face = "bold"),
#         legend.title = element_text(size = 14, face = "bold"),
#         legend.text = element_text(size = 14))



clusters4_aggregated <- filter (clusters3_aggregated, !grepl("T077", iron_treatment))

clusters4_aggregated$zscore <- ave(clusters4_aggregated$norm_abundance, clusters4_aggregated$cluster, FUN=scale)


clusters4_aggregated$zscore_BA <- ave(clusters4_aggregated$norm_abundance, list(clusters4_aggregated$cluster, clusters4_aggregated$bioassay), FUN=scale)


clusters4_aggregated_means <- aggregate(list(zscore_BA =clusters4_aggregated$zscore_BA, zscore =clusters4_aggregated$zscore, norm_abundance = clusters4_aggregated$norm_abundance), by=list( clusters4_aggregated$fulltreatment, clusters4_aggregated$bioassay, clusters4_aggregated$cluster), FUN=mean) %>%
  rename (fulltreatment = Group.1, 
          bioassay = Group.2, 
          cluster = Group.3)

clusters4_aggregated_means$clust_annotation <-   ifelse (clusters4_aggregated_means$cluster ==  "clust_343", "ISIP-2A", 
                   ifelse (clusters4_aggregated_means$cluster ==  "clust_411", "Nitrate reductase", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_534", "Flavodoxin", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1154", "ISIP-3", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1814", "ISIP-1", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1820", "Plastocyanin", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_4660", "Flavodoxin-1", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_5562", "Cu-Zn SOD", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_55", "PSI", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_808", "Ferredoxin", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_2934", "Multicopper Oxidase", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_891", "Glutamate Synthase", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_332", "Chl A-B binding protein", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_417", "AMT", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_2501", "Fe-Mn SOD", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1525", "SLC-4",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_231", "Rhodopsin",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1906", "Mn-cluster",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_812", "ZIP",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_16168", "TonB",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_10646", "TonB_2",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1372", "Urea transporter",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_320", "Urea carboxylase",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_6019", "Ornithine cyclodeaminase",    
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_655", "CA_clust_655",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_6573", "CA_clust_6573", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_35", "Ribosome recycling",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_2490", "Rubisco",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_2244", "PEPC",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1830", "Acireductone dioxygenase Fe containing",
                                    clusters4_aggregated_means$cluster 
                    ))))))))))))))))))))))))))))))

clusters4_aggregated_means$zscorenew <-   ifelse (clusters4_aggregated_means$norm_abundance ==  "0", -20, 
                   
                                              clusters4_aggregated_means$zscore)

clusters4_aggregated_means$zscore_BAnew <-   ifelse (clusters4_aggregated_means$norm_abundance ==  "0", -20, 
                   
                                              clusters4_aggregated_means$zscore_BA)


proteinclusterorder <- c("ISIP-1", "ISIP-2A", "ISIP-3",
                         
                         "Flavodoxin", "Flavodoxin-1", "Plastocyanin",  "PSI","Chl A-B binding protein", "Ferredoxin",  "Mn-cluster", "Rubisco", "Rhodopsin", "Cd Carbonic Anhydrase",
                         
                         "Fe-Mn SOD", "Cu-Zn SOD", 
                         
                         "CA", "SLC-4", 
                         
                         "Nitrate reductase",  "AMT", "Urea transporter", "Urea carboxylase", "Ornithine cyclodeaminase",
                         "Multicopper Oxidase", "Glutamate Synthase",
                         
                         "ZIP", "TonB", "TonB_2",  "Ribosome recycling", "Acireductone dioxygenase Fe containing",
                         "Viral capsid protein", "Large Euk DNA virus capsid", "PEPC"  )

ggplot(clusters4_aggregated_means, aes(x = factor(fulltreatment, level = treatment_orders),
 y = factor(clust_annotation, level = proteinclusterorder), 
  fill = zscore_BAnew)) +  
    geom_tile(color = "black", lwd = 0.7)+

  scale_fill_gradientn(colors = c("deepskyblue2","white","darkred"),
    #colours = c("#0CB702","white", "#C77CFF"),  "#0CB702","white", "#C77CFF"
                        values = rescale(c(-1.3,0,1.85)),
                        guide = "colorbar", limits=c(-1.3,1.85),
                      name = "Row Z-score")+

    # scale_fill_gradientn(colors = c("deepskyblue2","white","darkred"),
    # #colours = c("#0CB702","white", "#C77CFF"),  "#0CB702","white", "#C77CFF"
    #                     values = rescale(c(0.00001,0.005,0.06)),
    #                     guide = "colorbar", limits=c(0.00001,0.06),
    #                   name = "Row Z-score")+
   ylab ("")+
   xlab ("")+
   theme(axis.text.x=element_text(face = "bold", size = 10, color = "black"),
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank(), 
        axis.ticks  = element_blank()) +
    facet_grid(.~ bioassay, scales = "free_x") + 
    theme(strip.background =element_rect(fill="white"), 
        strip.text = element_text(size = 12, face = "bold"),
         legend.box.margin=margin(l=-10))

# https://stackoverflow.com/questions/66304713/how-to-add-geom-tile-outside-of-axis
```

zscore 
```{r}
clusters <- filter (norm_proteomicsdata, grepl("Bacill", class))  #clust_5911 , Cd carbonic anhydrase

clusters <- clusters [c(10, 3, 24:53)]

clusters2 <- clusters %>% mutate_at (3:32, funs (((. / sum(.)))))

clusters3 <- filter(clusters2, grepl("clust_", cluster)) 

clusters3 <- melt(clusters3, id.vars=c("cluster", "class")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


clusters3_aggregated <- aggregate(clusters3$norm_abundance, by=list(Category=clusters3$cluster, clusters3$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(cluster = Category) %>% 
                    separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

clusters3_aggregated$fulltreatment <- paste (clusters3_aggregated$temperature, clusters3_aggregated$iron , sep ='_')
clusters3_aggregated$fulltreatment <- ifelse (clusters3_aggregated$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))





clusters4_aggregated <- filter (clusters3_aggregated, !grepl("T0", iron_treatment))

clusters4_aggregated$zscore <- ave(clusters4_aggregated$norm_abundance, clusters4_aggregated$cluster, FUN=scale)






ggplot(clusters4_aggregated, 
             aes(y = zscore, 
                x=cluster, 
                color = fulltreatment))  
  facet_wrap(~bioassay)
  
ggplotly(x)

geom_point(size=3)
  #stat_summary(fun = mean, geom = "point", size = 6, stroke = 1, alpha = 0.5)+
  #stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F,  size = 1.3, width = 0.1) +
  ylab (expression ("Fraction contribution to total diatom proteins" )) + 
  xlab (expression ("")) +
  theme_bw()+
  theme(axis.text.x=element_text(face = "bold", size = 15, color = "black"),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))





clusters4_aggregated_means <- aggregate(list(zscore_BA =clusters4_aggregated$zscore_BA, zscore =clusters4_aggregated$zscore, norm_abundance = clusters4_aggregated$norm_abundance), by=list( clusters4_aggregated$fulltreatment, clusters4_aggregated$bioassay, clusters4_aggregated$cluster), FUN=mean) %>%
  rename (fulltreatment = Group.1, 
          bioassay = Group.2, 
          cluster = Group.3)

```




```{r}
####### class normalized ribosomes (Haptophytes and Diatoms)
ribosome_class <- norm_proteomicsdata [c(23,10, 24:53 )] 

ribosome_class$class <- ifelse (ribosome_class$class ==  "Bacillariophyta", "Diatoms", 
                      ifelse (ribosome_class$class ==  "Haptophyta", "Haptophytes",
                      ifelse (ribosome_class$class ==  "Prymnesiophyceae", "Haptophytes",
                      ifelse (ribosome_class$class ==  "Pavlovophyceae", "Haptophytes",
                      "Other"))))


ribosome_diatom <- filter (ribosome_class, grepl ("Diat", class))
ribosome_diatom <- ribosome_diatom %>% mutate_at (3:32, funs (((. / sum(.)))))


ribosome_diatom2 <- melt(ribosome_diatom, id.vars=c("grpnorm_compartment", "class")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


ribosome_diatom3 <- aggregate(ribosome_diatom2$norm_abundance, 
                  by=list(Category = ribosome_diatom2$grpnorm_compartment, 
                  ribosome_diatom2$treatment, 
                  ribosome_diatom2$class), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(compartment = Category) %>%
                  rename(class = Group.3) %>%
  separate(treatment, into=c("bioassay","timepoint", "temperature_treatment", "iron_treatment",  "replicate"), sep = "_")

ribosome_diatom3$treatment <- paste (ribosome_diatom3$temperature_treatment, ribosome_diatom3$iron_treatment, sep = "_")


x_vs_treatment ((filter(ribosome_diatom3, grepl("ribo", compartment))),
                treatment, 
                treatment_order, 
                norm_abundance, 
                "Ribosomal fraction - Diatoms") +
  expand_limits(y = 0:0.2) +  
  theme (axis.text.x = element_text(angle = 45))  



ribosome_diatom3$temperature <- ifelse (ribosome_diatom3$bioassay ==  "BA1" & ribosome_diatom3$temperature_treatment ==  "T0",  -0.3, 
                         ifelse (ribosome_diatom3$bioassay ==  "BA1" & ribosome_diatom3$temperature_treatment ==  "HT",  1.7, 
                         ifelse (ribosome_diatom3$bioassay ==  "BA1" & ribosome_diatom3$temperature_treatment ==  "LT",  -0.3, 
                         ifelse (ribosome_diatom3$bioassay ==  "BA2" & ribosome_diatom3$temperature_treatment ==  "T0",  -1.4, 
                         ifelse (ribosome_diatom3$bioassay ==  "BA2" & ribosome_diatom3$temperature_treatment ==  "HT",  1, 
                         ifelse (ribosome_diatom3$bioassay ==  "BA2" & ribosome_diatom3$temperature_treatment ==  "LT",  -1, 
                         "Other"))))))


```

``` {r}
ribosome_genus <- norm_proteomicsdata [c(23,10:13, 24:53 )] 


ribosome_genus$genus_a <-   ifelse (ribosome_genus$G ==  "Thalassiosira", "Centric", 
                            ifelse (ribosome_genus$G ==  "Chaetoceros", "Centric", 
                            ifelse (ribosome_genus$G ==  "Fragilariopsis", "Pennate",
                            ifelse (ribosome_genus$G ==  "Pseudo-nitzschia", "Pennate",
                            ifelse (ribosome_genus$G ==  "Araphid-pennate", "Pennate",
                            ifelse (ribosome_genus$G ==  "Polar-centric-Mediophyceae", "Centric",
                            ifelse (ribosome_genus$G ==  "Raphid-pennate", "Pennate",  
                            ifelse (ribosome_genus$G ==  "Bacillariophyta_X", "Ambiguous Diatom",   
                            ifelse (ribosome_genus$G ==  "Phaeocystis", "Phaeocystis",  
                            ifelse (ribosome_genus$class ==  "Dinophyceae", "Dinophyte",
                            ifelse (ribosome_genus$class ==  "Dinophyta", "Dinophyte",

                            "Other Diatoms")))))))))))


ribosome_genus2 <- filter (ribosome_genus, grepl ("Phaeo", genus_a))


ribosome_genus2 <- ribosome_genus2 %>% mutate_at (6:35, funs (((. / sum(.)))))

ribosome_genus2 <- ribosome_genus2 [c(1, 6:36)]


ribosome_genus2 <- melt(ribosome_genus2, id.vars=c("grpnorm_compartment", "genus_a")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


ribosome_genus3 <- aggregate(ribosome_genus2$norm_abundance, 
                  by=list(Category = ribosome_genus2$grpnorm_compartment, 
                  ribosome_genus2$treatment, 
                  ribosome_genus2$genus_a), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(compartment = Category) %>%
                  rename(genus = Group.3) %>%
  separate(treatment, into=c("bioassay","timepoint", "temperature_treatment", "iron_treatment",  "replicate"), sep = "_")

ribosome_genus3$treatment <- paste (ribosome_genus3$temperature_treatment, ribosome_genus3$iron_treatment, sep = "_")



ribosome_genus3$temperature <- ifelse (ribosome_genus3$bioassay ==  "BA1" & ribosome_genus3$temperature_treatment ==  "T0",  -0.3, 
                         ifelse (ribosome_genus3$bioassay ==  "BA1" & ribosome_genus3$temperature_treatment ==  "HT",  1.7, 
                         ifelse (ribosome_genus3$bioassay ==  "BA1" & ribosome_genus3$temperature_treatment ==  "LT",  -0.3, 
                         ifelse (ribosome_genus3$bioassay ==  "BA2" & ribosome_genus3$temperature_treatment ==  "T0",  -1.4, 
                         ifelse (ribosome_genus3$bioassay ==  "BA2" & ribosome_genus3$temperature_treatment ==  "HT",  1, 
                         ifelse (ribosome_genus3$bioassay ==  "BA2" & ribosome_genus3$temperature_treatment ==  "LT",  -1, 
                         "Other"))))))

ggplot((filter(ribosome_genus3, grepl("ribo", compartment))) , aes(
    x = as.numeric(temperature),
    y = norm_abundance,
    color = bioassay,
    shape = iron_treatment)) +
   geom_point(size = 4, alpha=0.25)+
    scale_color_manual(name="bioassay",
                       breaks = c("T0", "BA1","BA2"),
                       labels = c("T0", "BA1", "BA2"),
                       values = c('#009E73', 'black', 'blue')) +
    scale_shape_manual(name = "Iron",
                       breaks = c("Fe","noFe"), #this removes the T0 from the legend
                       labels = c("Fe", "noFe"),
                       values = c(19, 1, 1)) +

    #ylab ("fraction") +
    xlab (expression ("Temperature")) +

    theme_bw()+
    theme(axis.text.x=element_text(face = "bold", size = 12, angle = 45, color = "black", vjust = 0.5, hjust = 0.5),
          axis.title.y=element_text(size=20, color = "black"),
          axis.text.y=element_text(face = "bold", size = 15, color = "black"),
          strip.background =element_rect(fill = "white"),
          strip.text.x = element_text(size = 15, face = "bold"),
          legend.title = element_text(size = 14, face = "bold"),
          legend.text = element_text(size = 14))+
   expand_limits(y = 0) 



#x<- lm(norm_abundance ~ (as.numeric(temperature))*iron_treatment, data = (filter(ribosome_genus3, grepl("ribo", compartment))))

#summary(x)

#https://academic.oup.com/jxb/article/72/6/2165/6161194
#maybe do a heatmap?
```



Copper + Plastocyanin 2022-08-01
Look for 'copper' things in the data
```{r}

###PCYN###
####metaP PCYN

meta_pcyn <- filter(norm_proteomicsdata, grepl("^clust_6573$", cluster)) 

meta_pcyn <- meta_pcyn [c(3, 24:53)]

meta_pcyn <- melt(meta_pcyn, id.vars=c("cluster")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

meta_pcyn_aggregated <- aggregate(meta_pcyn$norm_abundance, by=list(Category=meta_pcyn$cluster, meta_pcyn$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(cluster = Category) %>% 
                  separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")


meta_pcyn_aggregated$fulltreatment <- paste (meta_pcyn_aggregated$temperature, meta_pcyn_aggregated$iron , sep ='_')
meta_pcyn_aggregated$fulltreatment <- ifelse (meta_pcyn_aggregated$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (meta_pcyn_aggregated$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (meta_pcyn_aggregated$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (meta_pcyn_aggregated$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))


ggplot(meta_pcyn_aggregated, 
             aes(x = factor(fulltreatment, level = treatment_orders),
               #x = factor(iron_treatment, level = irontreatment_order), 
                y=norm_abundance, 
                color = temperature_treatment)) + 
  facet_wrap(~bioassay)+
  geom_point()+
  stat_summary(fun = mean, geom = "point", size = 6, stroke = 1, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F,  size = 1.3, width = 0.1) +
  ylab (expression ("PCYN Metaproteome mass fraction" )) + 
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     values = c( 'black','darkorange2', "black")) +
  xlab (expression ("")) +
  theme_bw()+
  theme(axis.text.x=element_text(face = "bold", size = 15, color = "black"),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))


#Then I want to show taxon-normalized protein abundance. So for example, Pnitzchia PCYN normalized to total Pnitzchia protein. This is different from total PCYN in the metaproteome that is allocated to PN

#First, I  normalize everything to the taxonomic group of interest  

pcyntaxon_diatom <- filter (norm_proteomicsdata, grepl("Bacill", class))

pcyntaxon_diatom <- pcyntaxon_diatom [c(10, 3, 24:53)]

pcyntaxon_diatom2 <- pcyntaxon_diatom %>% mutate_at (3:32, funs (((. / sum(.)))))

pcyn_diatom <- filter(pcyntaxon_diatom2, grepl("^clust_655$", cluster)) #clust_1820
pcyn_diatom <- melt(pcyn_diatom, id.vars=c("cluster", "class")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

pcyn_diatom_aggregated <- aggregate(pcyn_diatom$norm_abundance, by=list(Category=pcyn_diatom$cluster, pcyn_diatom$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(cluster = Category) %>% 
                    separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

pcyn_diatom_aggregated$fulltreatment <- paste (pcyn_diatom_aggregated$temperature, pcyn_diatom_aggregated$iron , sep ='_')
pcyn_diatom_aggregated$fulltreatment <- ifelse (pcyn_diatom_aggregated$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (pcyn_diatom_aggregated$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (pcyn_diatom_aggregated$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (pcyn_diatom_aggregated$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))

pcyn_diatom_aggregated$taxon <- "Diatoms"

ggplot(filter(pcyn_diatom_aggregated, !grepl("0.0588412",norm_abundance)), 
             aes(x = factor(fulltreatment, level = treatment_orders), 
                y=norm_abundance, 
                color = temperature_treatment)) + 
  facet_wrap(~bioassay)+
  #geom_point(size=3)+
  stat_summary(fun = mean, geom = "point", size = 6, stroke = 1, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F,  size = 1.3, width = 0.1) +
  ylab (expression ("Fraction contribution to total diatom proteins" )) + 
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     values = c( 'black','darkorange2', "black")) +
  xlab (expression ("")) +
  theme_bw()+
  theme(axis.text.x=element_text(face = "bold", size = 15, color = "black"),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))


## haptophyte
pcyntaxon_hapto <- filter (norm_proteomicsdata, grepl("Prymn", class))

pcyntaxon_hapto <- pcyntaxon_hapto [c(10, 3, 24:53)]

pcyntaxon_hapto2 <- pcyntaxon_hapto %>% mutate_at (3:32, funs (((. / sum(.)))))

pcyn_hapto <- filter(pcyntaxon_hapto2, grepl("clust_1820", cluster)) #clust_1820
pcyn_hapto <- melt(pcyn_hapto, id.vars=c("cluster", "class")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

pcyn_hapto_aggregated <- aggregate(pcyn_hapto$norm_abundance, by=list(Category=pcyn_hapto$cluster, pcyn_hapto$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(cluster = Category) %>% 
                    separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

pcyn_hapto_aggregated$fulltreatment <- paste (pcyn_hapto_aggregated$temperature, pcyn_hapto_aggregated$iron , sep ='_')
pcyn_hapto_aggregated$fulltreatment <- ifelse (pcyn_hapto_aggregated$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (pcyn_hapto_aggregated$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (pcyn_hapto_aggregated$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (pcyn_hapto_aggregated$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))

pcyn_hapto_aggregated$taxon <- "Phaeocystis"


ggplot(filter(pcyn_hapto_aggregated, !grepl("0.0588412",norm_abundance)), 
             aes(x = factor(fulltreatment, level = treatment_orders), 
                y=norm_abundance, 
                color = temperature_treatment)) + 
  facet_wrap(~bioassay)+
  #geom_point(size=3)+
  stat_summary(fun = mean, geom = "point", size = 6, stroke = 1, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F,  size = 1.3, width = 0.1) +
  ylab (expression ("Fraction contribution to total Phaeocystis proteins" )) + 
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     values = c( 'black','darkorange2', "black")) +
  xlab (expression ("")) +
  theme_bw()+
  theme(axis.text.x=element_text(face = "bold", size = 15, color = "black"),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))


pcyntaxon <-  rbind(pcyn_diatom_aggregated, pcyn_hapto_aggregated)

#fig3_PCYN-taxon <- 
ggplot ((filter(pcyntaxon, grepl("Diatom|Phaeo", taxon))), aes(
  x = factor(fulltreatment, level = treatment_orders),
  y = norm_abundance, 
  #shape = taxon, 
  fill = taxon)) +  
  #stat_summary(fun = mean, color = "black", geom = "point", show.legend = T, size = 5, alpha = 0.8, stroke = 1.5 )+
  stat_summary(show.legend = T, fun = mean, geom = "col",
               width=0.7,    
           position=position_dodge(0.7))+ 
  stat_summary(fun.data =  mean_se, geom = "errorbar", position=position_dodge(width=0.7),show.legend = F, size = 1.1, width = 0.1)+ 
  facet_grid(.~ bioassay )+
  theme_bw()+
  ylab (expression(paste("PCYN taxon-specific protein mass fraction contribution")))+
  xlab ("") +
  #scale_y_log10()+
  #scale_shape_manual(name="",
                     #values = c("Diatoms" = 15, 
                               # "Phaeocystis" = 0))+ 
  scale_fill_manual(name="dh",
                     values = c("Diatoms" = "grey30", 
                                "Phaeocystis" = "grey"))+ 
 theme(axis.text.x=element_text(face = "bold", size = 11, color = "black", vjust = 1, hjust = 1,  angle = 45),
        axis.title.y=element_text(size=14,face = "bold", color = "black"), 
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        strip.background =element_rect(fill = "white"),
        legend.text = element_text(size = 10, face = "bold"), 
        strip.text.x = element_text(size = 11, face = "bold"), 
       legend.position = "top", 
       legend.title = element_blank(),
       legend.justification = "left", 
       legend.box.margin=margin(-10,-10,-10,-10))+ 
       #plot.margin = unit(c(0.25,0.25,0.25,0), units = "cm"))+
  guides(shape = guide_legend(override.aes = list(size=5)))#this locks the legend in position so it doesnt move around as you zoom in and out of the plot)
```


ISIPS
```{r}
#ISIP2A shows increase with low iron , but increases under warming in PN, not in Frag!
```


rubisco 2022-04-22
```{r}
rubisco_class <- norm_proteomicsdata [c(4,10, 24:53 )] 

rubisco_class$class <- ifelse (rubisco_class$class ==  "Bacillariophyta", "Diatoms", 
                      ifelse (rubisco_class$class ==  "Haptophyta", "Haptophytes",
                      ifelse (rubisco_class$class ==  "Prymnesiophyceae", "Haptophytes",
                      ifelse (rubisco_class$class ==  "Pavlovophyceae", "Haptophytes",
                      "Other"))))


rubisco_class2 <- filter (rubisco_class, grepl ("Diatom", class))
rubisco_class2 <- rubisco_class2 %>% mutate_at (3:32, funs (((. / sum(.)))))


rubisco_class2 <- melt(rubisco_class2, id.vars=c("cluster_annotation", "class")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


rubisco_class3 <- aggregate(rubisco_class2$norm_abundance, 
                  by=list(Category = rubisco_class2$cluster_annotation, 
                  rubisco_class2$treatment, 
                  rubisco_class2$class), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(protein = Category) %>%
                  rename(class = Group.3) %>%
  separate(treatment, into=c("bioassay","timepoint", "temperature_treatment", "iron_treatment",  "replicate"), sep = "_")

rubisco_class3$treatment <- paste (rubisco_class3$temperature_treatment, rubisco_class3$iron_treatment, sep = "_")

protein <- filter(rubisco_class3, grepl("Ribulose bis", protein))

#rubisco clust_annotation: go_287_ (large subunit), Ribulose ..(small subunit)

x_vs_treatment ((filter(rubisco_class3, grepl("Ribulose", protein))),
                treatment, 
                treatment_order, 
                norm_abundance, 
                "rubisco fraction - Diatoms") +
  expand_limits(y = 0:0.2) +  
  theme (axis.text.x = element_text(angle = 45)) +
  geom_point(size =3)

```

LHCs 2022-04-22
```{r}
lhcs_class <- norm_proteomicsdata [c(4,10, 24:53 )] 

lhcs_class$class <- ifelse (lhcs_class$class ==  "Bacillariophyta", "Diatoms", 
                      ifelse (lhcs_class$class ==  "Haptophyta", "Haptophytes",
                      ifelse (lhcs_class$class ==  "Prymnesiophyceae", "Haptophytes",
                      ifelse (lhcs_class$class ==  "Pavlovophyceae", "Haptophytes",
                      "Other"))))


lhcs_class2 <- filter (lhcs_class, grepl ("Diatom", class))
lhcs_class2 <- lhcs_class2 %>% mutate_at (3:32, funs (((. / sum(.)))))


lhcs_class2 <- melt(lhcs_class2, id.vars=c("cluster_annotation", "class")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


lhcs_class3 <- aggregate(lhcs_class2$norm_abundance, 
                  by=list(Category = lhcs_class2$cluster_annotation, 
                  lhcs_class2$treatment, 
                  lhcs_class2$class), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(protein = Category) %>%
                  rename(class = Group.3) %>%
  separate(treatment, into=c("bioassay","timepoint", "temperature_treatment", "iron_treatment",  "replicate"), sep = "_")

lhcs_class3$treatment <- paste (lhcs_class3$temperature_treatment, lhcs_class3$iron_treatment, sep = "_")

x_vs_treatment ((filter(lhcs_class3, grepl("pfam_PF11264_Thylakoid", protein))),
                treatment, 
                treatment_order, 
                norm_abundance, 
                "LHC fraction - Diatoms") +
  expand_limits(y = 0:0.2) +  
  theme (axis.text.x = element_text(angle = 45))

###########################
lhcs_genus <- norm_proteomicsdata [c(13,3, 24:53 )] 


lhcs_genus$G <-   ifelse (lhcs_genus$G ==  "Thalassiosira", "Thalassio", 
                            ifelse (lhcs_genus$G ==  "Chaetoceros", "Chaeto", 
                            ifelse (lhcs_genus$G ==  "Fragilariopsis", "Fragilariopsis",
                            ifelse (lhcs_genus$G ==  "Pseudo-nitzschia", "Pseudo-nitzschia",
                            ifelse (lhcs_genus$G ==  "Araphid-pennate", "Araphid-pennate",
                            ifelse (lhcs_genus$G ==  "Polar-centric-Mediophyceae", "Polar centric",
                            ifelse (lhcs_genus$G ==  "Raphid-pennate", "Raphid pennate",  
                            ifelse (lhcs_genus$G ==  "Bacillariophyta_X", "Ambiguous Diatom",   
                            ifelse (lhcs_genus$G ==  "Phaeocystis", "Phaeocystis",   
                            "Other Diatoms")))))))))


lhcs_genus2 <- filter (lhcs_genus, grepl ("Chaet", G))
lhcs_genus2 <- lhcs_genus2 %>% mutate_at (3:32, funs (((. / sum(.)))))


lhcs_genus2 <- melt(lhcs_genus2, id.vars=c("cluster", "G")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


lhcs_genus3 <- aggregate(lhcs_genus2$norm_abundance, 
                  by=list(Category = lhcs_genus2$cluster, 
                  lhcs_genus2$treatment, 
                  lhcs_genus2$G), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(protein = Category) %>%
                  rename(genus = Group.3) %>%
  separate(treatment, into=c("bioassay","timepoint", "temperature_treatment", "iron_treatment",  "replicate"), sep = "_")

lhcs_genus3$treatment <- paste (lhcs_genus3$temperature_treatment, lhcs_genus3$iron_treatment, sep = "_")

#clust80 <- filter (lhcs_genus3, grepl("clust_402", protein))


x_vs_treatment ((filter(lhcs_genus3, grepl("clust_1919$", protein))),
                treatment, 
                treatment_order, 
                norm_abundance, 
                "LHC fraction - Diatoms") +
  expand_limits(y = 0) +  
  theme (axis.text.x = element_text(angle = 45))

```

Iron stress proteins (ISIPs, Flavodoxin, Plastocyanin etc.)
```{r}
#pnitz <- filter (norm_proteomicsdata, grepl ("Pseudo-nit", grpnorm_taxgrp))

nitrogenpnitz <- filter(norm_proteomicsdata, grepl("clust_1820", cluster))

nitrogenpnitz2 <- filter(nitrogenpnitz, grepl("Bacilla", class))

nitrogenpnitz3 <- nitrogenpnitz2 [c(3,24:53)]

#ubiquitin <- filter(alldata, grepl('ubiquitin|ubiquit|Ubiquit|clust_270 |clust_1965 |clust_2019 |clust_1057 ',taxon_function)) 

#urea <- filter(alldata, grepl('urea |clust_320 |clust_1530 |clust_4613 |ornith',taxon_function)) 

#initiationrepressor <- filter(alldata, grepl('struct|clust_2019|clust_1192',taxon_function)) 

#silicon <- filter(allproteindata, grepl('clust_202 |clust_1814 ',taxon_function)) 

#protein_name_map <- c("clust_1814" = "ISIP-1", 
                      # "clust_343" = "ISIP-2A", 
                      # "clust_1154" = "ISIP-3",
                      # "clust_1820" = "Plastocyanin",
                      # "clust_534" = "Flavodoxin",
                      # "clust_4660" = "Flavodoxin-1",
                      # "clust_411" = "Nitrate Reductase", 
                      # "clust_417" = "Ammonium Transporter",
                      # "clust_55" = "PSI-P700",
                      # "clust_211" = "PSII-psbC",
                      # "clust_42" = "PSII-psbA")

isisp_flavo <- filter(alldata, grepl('clust_1814 |clust_343 |clust_1154 |clust_1820 |clust_534 |clust_411 |clust_417 |clust_55 |clust_211 |clust_42 ',taxon_function)) 

proteinhm <- melt(nitrogenpnitz3, id.vars=c("cluster", "peptide")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

proteinhm <- proteinhm %>% separate(treatment, into=c("bioassay", "temperature", "iron", "timepoint", "replicate", "injection"), sep = "_")

proteinhm <- proteinhm %>% separate(taxon_function, into=c("cluster", "x", "y", "z", "zz", "zzz", "a", "compartment", "c", "d"), sep = " ; ")

proteinhm$replicate <- str_sub(proteinhm$replicate, -2,-2)
proteinhm$injection <- str_sub(proteinhm$injection, 2)

proteinhm$treatment <- paste (proteinhm$temperature, proteinhm$iron, proteinhm$replicate , sep ='_')

#isip1 <- filter(isip1, grepl('ribosomal',compartment)) 

hm_aggregated <- aggregate(proteinhm$norm_abundance, by=list(Category=proteinhm$cluster, proteinhm$treatment, proteinhm$bioassay), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(cluster = Category) %>%
                  rename(bioassay = Group.3)

#hm_aggregated$cluster <- protein_name_map [hm_aggregated$cluster]

hm_aggregated1 <- filter(hm_aggregated, grepl('BA1',bioassay)) 

hm_aggregated2 <- dcast(hm_aggregated1, cluster ~ treatment)

#BA2
hm_aggregated2 <- hm_aggregated2[c(1,14:16,11:13,8:10,5:7,2:4)]

#BA1
hm_aggregated2 <- hm_aggregated2[c(1,13:16,11:12,8:10,5:7,2:4)]

#combined
hm_aggregated2 <- hm_aggregated2[c(1,14:16,11:13,8:10,5:7,2:4)]

hm_aggregated2 <- data.frame(hm_aggregated2[,-1], row.names = hm_aggregated2 [,1])

hm_aggregated3 <- hm_aggregated2[c(3,4,5,8,9,10,2,7,1,6),]


#png("BA1_allproteins.png", width=50*0.75, height=30*0.75, units="cm", res=600) 
par(cex.main=2.5, cex.lab=1.8, cex.axis=1.5)
heatmap.2 (as.matrix(hm_aggregated2), 
             trace = "none",
             density = "none", 
             col = bluered(100),
             breaks = seq(-2,2, length.out = 101),
          # symbreaks = TRUE,#changes the scale of the color key
             cexRow = 1.8, 
             cexCol =1.2,
             margins = c(8,20),
             scale = "row" ,
             dendrogram = 'none',  
             Colv = FALSE, 
             Rowv = FALSE,
             keysize = 0.75,
             key.title = NA,
             #density.info = "density",
          #BA2
            #colsep = c(3,6,9,12),
            rowsep = c(3,8),
          #BA1
          colsep=c(4,6,9,12),
             sepcolor = "black",
             sepwidth = c(0.05, 0.05))

#dev.off()
ggplot(proteinhm, aes(x=timepoint, 
                y=norm_abundance,
                color = temperature, 
                shape = iron)) + 
  facet_wrap(~bioassay)+
  #geom_point(size =3, alpha =0.5)
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.8, alpha = 0.5)+
  #stat_summary(fun = mean, geom = "line", size = 1.1, aes(linetype = iron_treatment))+
  stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1.3, width = 0.1)+
  
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('darkorange2', 'black', 'black')) +
  scale_linetype_manual(name="Iron",
                     breaks = c("Fe","noFe"),
                     labels = c("Fe", "noFe"),
                     values = c(1,9,1 )) +
  scale_shape_manual(name = "Iron",
                     breaks = c("noFe","Fe"), #this removes the T0 from the legend
                     labels = c("noFe", "addedFe"),
                     values = c(19, 1, 1)) + #this keeps the T0 formatting that I want
  #scale_y_continuous(expand = c(0,5))+
  ylab (expression (NO[3]~uM)) + 
  xlab (expression ("Day Count")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))
           #guides(color = guide_legend(override.aes = list(size= 2)))
          #guides(linetype = guide_legend(override.aes = list(size= 2)))

#dev.off()


```

Silica
```{r}
silicon <- filter(norm_proteomicsdata, grepl('clust_202 |clust_19688',cluster)) 


proteinhm <- melt(silicon, id.vars=c("taxon_function", "peptide")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

proteinhm <- proteinhm %>% separate(treatment, into=c("bioassay", "temperature_treatment", "iron_treatment", "timepoint", "replicate", "injection"), sep = "_")

proteinhm <- proteinhm %>% separate(taxon_function, into=c("cluster", "x", "y", "z", "zz", "zzz", "a", "compartment", "c", "d"), sep = " ; ")

proteinhm$replicate <- str_sub(proteinhm$replicate, -2,-2)
proteinhm$injection <- str_sub(proteinhm$injection, 2)

proteinhm$treatment <- paste (proteinhm$temperature, proteinhm$iron, proteinhm$replicate , sep ='_')


hm_aggregated <- aggregate(proteinhm$norm_abundance, by=list(Category=proteinhm$cluster, proteinhm$treatment, proteinhm$bioassay), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(cluster = Category) %>%
                  rename(bioassay = Group.3)

hm_aggregated <- hm_aggregated %>% separate(treatment, into=c("temperature_treatment", "iron_treatment", "replicate"), sep = "_")

hm_aggregatednoT0 <- filter(hm_aggregated, !grepl("T0",iron_treatment))


Fetreatment_order <- c('noFe', 'Fe')


#png("SITprotein_20220331.png", width=40*0.75, height=30*0.75, units="cm", res=600) 

x_vs_treatment (hm_aggregatednoT0, iron_treatment, norm_abundance, "Relative Abndance - Silicon transporter protein")

#dev.off()


```

proteases 2022-04-28
```{r}
#https://www.ncbi.nlm.nih.gov/pmc/articles/PMC1326381/
#https://pubmed.ncbi.nlm.nih.gov/10849347/
#https://academic.oup.com/plcell/article/12/3/419/6008767
#some proteases function to stabalize the Mn cluster or keep the Mn cluster good and going. So if we have an increase in Mn cluster proteins, we'll need more proteases to maintain these proteins. 

ribosome <- norm_proteomicsdata [c(3, 14,  24:53 )] 


ribosome$genus <-   ifelse (ribosome$genus ==  "Thalassiosira", "Centric", 
                            ifelse (ribosome$genus ==  "Chaetoceros", "Chaetoceros", 
                            ifelse (ribosome$genus ==  "Fragilariopsis", "Fragilariopsis",
                            ifelse (ribosome$genus ==  "Pseudo-nitzschia", "Pseudo-nitzschia",
                            ifelse (ribosome$genus ==  "Araphid-pennate", "Pennate",
                            ifelse (ribosome$genus ==  "Polar-centric-Mediophyceae", "Centric",
                            ifelse (ribosome$genus ==  "Raphid-pennate", "Pennate",  
                            ifelse (ribosome$genus ==  "Bacillariophyta_X", "Ambiguous Diatom",   
                            ifelse (ribosome$genus ==  "Phaeocystis", "Phaeocystis",   
                            "Other Diatoms")))))))))


ribosome_genus2 <- filter (ribosome, grepl ("Chaeto", genus))
ribosome_genus2 <- ribosome_genus2 %>% mutate_at (3:32, funs (((. / sum(.)))))


ribosome_genus2 <- melt(ribosome_genus2, id.vars=c("cluster", "genus")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

# ribosome3 <- melt(ribosome, id.vars=c("cluster")) %>%
#             rename(treatment = variable) %>%
#             rename(norm_abundance = value )

ribosome4 <- aggregate(ribosome_genus2$norm_abundance, 
                  by=list(Category = ribosome_genus2$treatment, ribosome_genus2$cluster), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Category) %>% 
                  rename(cluster = Group.2) %>%
              separate(treatment, into=c("bioassay","timepoint", "temperature_treatment", "iron_treatment",  "replicate"), sep = "_")

ribosome4$treatment <- paste (ribosome4$temperature_treatment, ribosome4$iron_treatment, sep = "_")


x_vs_treatment ((filter(ribosome4, grepl("clust_231$", cluster))), 
                treatment, 
                treatment_order, 
                norm_abundance,
                "Ribosomal fraction - entire metaproteome") +
  expand_limits(y = 0) + 
  theme (axis.text.x = element_text(angle = 45))
```


other plots 
```{r}
silicon <- filter(alldata, grepl('clust_202 |clust_13423|clust_15873',taxon_function)) 


proteinhm <- melt(silicon, id.vars=c("taxon_function", "peptide")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


proteinhm <- proteinhm %>% separate(treatment, into=c("bioassay", "temperature", "iron", "timepoint", "replicate", "injection"), sep = "_")

si_pn <- filter(proteinhm, grepl("Frag", taxon_function))

ggplot(proteinhm, aes(x=temperature, 
                y=norm_abundance,
                color = temperature))+ 
                #shape = iron)) +
  
  #geom_point(size = 3, alpha= 0.5)+
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.8, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1.3, width = 0.1) +
  
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('darkorange2', 'black', 'black')) +
    scale_shape_manual(name = "Iron",
                     breaks = c("noFe","Fe"), #this removes the T0 from the legend
                     labels = c("noFe", "addedFe"),
                     values = c(19, 1, 1)) + #this keeps the T0 formatting that I want
  #ylab (expression (Si:NO3~drawdown)) + 
  #xlab (expression ("Day Count")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))+
    facet_wrap(~bioassay)
```

T0 compartments 2022-04-18
```{r}
compartments <- norm_proteomicsdata [c(10, 23,24:53 )] 

compartments$class <- ifelse (compartments$class ==  "Bacillariophyta", "Diatoms", 
                            ifelse (compartments$class ==  "Haptophyta", "Haptophytes",
                            ifelse (compartments$class ==  "Prymnesiophyceae", "Haptophytes",
                            ifelse (compartments$class ==  "Pavlovophyceae", "Haptophytes",
                            "Other"))))

compartments_2 <- melt(compartments, id.vars=c("grpnorm_compartment", "class")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

compartments_3 <- aggregate(compartments_2$norm_abundance, 
                                       by=list(Category = compartments_2$grpnorm_compartment, 
                                               compartments_2$treatment, 
                                               compartments_2$class), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(compartment = Category) %>%
                  rename (class =  Group.3)
                

compartments_3 <- compartments_3 %>% separate(treatment, into=c("bioassay","timepoint", "temperature_treatment", "iron_treatment",  "replicate"), sep = "_")

compartments_3$treatment <- paste (compartments_3$temperature_treatment, compartments_3$iron_treatment, sep = "_")




## get the T0 compartments to see what they look like. 
compartments_treemap <- aggregate(compartments_3$norm_abundance, 
                                       by=list(Category = compartments_3$compartment, 
                                               compartments_3$treatment, 
                                               compartments_3$bioassay), FUN=mean) %>% #important that the FUN = mean here
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(compartment = Category) %>%
                  rename (bioassay = Group.3)
                
ggplot ((filter(compartments_treemap, grepl("T0", treatment))), aes (
 area = norm_abundance, 
 fill = compartment,
 label = compartment)) + 
  geom_treemap() +
  theme (legend.position = "none")+
  geom_treemap_text (color = "white", 
                     place = "centre",
                     #size = 15,
                     grow = TRUE) + 
      facet_grid(.~ bioassay) 



ggplot ((filter(compartments_3, grepl("ribo", compartment))), aes(
        x = factor(treatment, level = treatment_order),
        y = norm_abundance, 
        color = class)) + 
      #geom_point() +
    facet_wrap(~bioassay)+
    stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.8, alpha = 0.5)+
    stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1.3, width = 0.1) +

      #geom_point(size = 2, alpha=0.5, stroke = 1.8)+

    
    scale_color_manual(name="Temperature",
                       breaks = c("LT","HT"),
                       labels = c("ambient", "warm"),
                       values = c('darkorange2', 'black', 'black')) +
    scale_shape_manual(name = "Iron",
                       breaks = c("Fe","noFe"), #this removes the T0 from the legend
                       labels = c("Fe", "noFe"),
                       values = c(19, 1, 1)) + 
    
    ylab (expression("% Ribosome contribution to total proteins" )) + 
    xlab (expression ("")) +
    
    theme_bw()+
    theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
          axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 0.5),
          axis.title.y=element_text(size=20, color = "black"), 
          axis.text.y=element_text(face = "bold", size = 15, color = "black"),
          strip.background =element_rect(fill = "white"),
          strip.text.x = element_text(size = 15, face = "bold"),
          legend.title = element_text(size = 14, face = "bold"),
          legend.text = element_text(size = 14)) + 
    expand_limits(y = 0) 





x_vs_treatment((filter(compartments_3, grepl("Hapto", class))), treatment, treatment_order, norm_abundance, compartments)
```

cluster nmds 
```{r}
proteomics_taxonomy <- norm_proteomicsdata [c(3, 4, 24:53)]

proteomics_taxonomy <- melt(proteomics_taxonomy, id.vars=c("cluster", "cluster_annotation")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

proteomics_taxonomy_aggregated <- aggregate(proteomics_taxonomy$norm_abundance, by=list(Category=proteomics_taxonomy$cluster, proteomics_taxonomy$treatment, proteomics_taxonomy$cluster_annotation), FUN=sum)%>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(genus = Category) %>%
                  rename (compartment = Group.3)

p <- proteomics_taxonomy_aggregated[c(1,2,4)]

proteomics_taxonomy_aggregated_2 <- dcast(p, treatment~genus, value.var = "norm_abundance", fun.aggregate = sum)

proteomics_taxonomy_aggregated_2 <- proteomics_taxonomy_aggregated_2 %>% separate(treatment, into=c("bioassay", "timepoint" ,"temperature_treatment", "iron_treatment", "replicate"), sep = "_")

proteomics_taxonomy_aggregated_2_nmds = proteomics_taxonomy_aggregated_2 [c(6:3742)] #select only the class/genus..etc

proteomics_taxonomy_aggregated_2_nmds = as.matrix(proteomics_taxonomy_aggregated_2_nmds)

set.seed(123)

nmds = metaMDS(proteomics_taxonomy_aggregated_2_nmds, try = 10000, trymax = 10000, distance = "bray") # try = 500, trymax = 500
nmds
scores(nmds, display = "sites")
stressplot(nmds) #for the suplement

data.scores <- as.data.frame(scores(nmds, "sites")) #can select sites (treatments) or species (variable)


# nmdsspecies <- dcast(proteomics_taxonomy_aggregated, genus+compartment~treatment)
#            # rename(treatment = variable) %>%
#             #rename(norm_abundance = value )

# data.scores$cluster <- nmdsspecies$compartment

data.scores$temperature_treatment <- proteomics_taxonomy_aggregated_2$temperature_treatment
data.scores$iron_treatment <- proteomics_taxonomy_aggregated_2$iron_treatment
data.scores$bioassay <- proteomics_taxonomy_aggregated_2$bioassay

data.scores$tempfe <- paste(data.scores$temperature_treatment,data.scores$iron_treatment)

ggplot(data.scores, aes(x = NMDS1, 
                        y = NMDS2))+
  geom_point(size = 3.5, 
                aes(fill = temperature_treatment,
                    alpha = iron_treatment, 
                    shape = bioassay,
                    color = temperature_treatment))+
  geom_point(size=3.5, 
             aes(shape = bioassay, 
                 color = temperature_treatment) ) +
  scale_shape_manual(values=c(21, 24),
                     name = "Bioassay")+
  scale_color_manual(values = c("HT" = "darkorange", "LT" = "black", "T0" = "blue"), 
                     name = "Temperature")+
  scale_alpha_manual(values = c("Fe" = 1, "noFe" = 0.2, "T0" = 0.2) , 
                    name = "Iron") +
  scale_fill_manual(values = c("HT" = "darkorange", "LT" = "black", "T0" = "blue"),    guide = FALSE) +
  ggtitle("Metaproteome - all MCL clusters")+
    theme(axis.text.y = element_text(colour = "black", size = 11, face = "bold"), 
    axis.text.x = element_text(colour = "black", face = "bold", size = 11), 
    legend.text = element_text(size = 12, face ="bold", colour ="black"), 
    legend.position = 'right', axis.title.y = element_text(face = "bold", size = 12), 
    axis.title.x = element_text(face = "bold", size = 12, colour = "black"), 
    legend.title = element_text(size = 14, colour = "black", face = "bold"), 
    panel.background = element_blank(), panel.border = element_rect(colour = "black",     fill = NA, size = 1.2),
    legend.key=element_blank(), 
   plot.title = element_text(vjust = -8, hjust = 0.02))
```









#nutrients
macros
```{r}
x_vs_time (alldata, daycount, (Si_uM), Si~mu*M)

x_vs_time (alldata, daycount, (PON_uM), PON~mu*M)

x_vs_time (alldata, daycount, (POC_uM), POC~mu*M)

x_vs_time (alldata, daycount, (POP_nM), POP~nM)

x_vs_time (alldata, daycount, (total_cells_mL), POP~nM)
```

metals
```{r}
x_vs_time (alldata, daycount, (DMn_nM), DFe57~n*M) +
  expand_limits(y = 0:0.5) 

x_vs_time (alldata, daycount, (PFe57_nM), PFe57~n*M) +
  expand_limits(y = 0:0.5) 

x_vs_treatment (alldata, temperature_treatment, temptreatment_order,  ((POC_uM)/((PAl_nM))), PFe[total]:POC~(mol:mol))+
  geom_point(size = 3)+
   #expand_limits(y = 0:0.00002) +
  scale_y_continuous(labels = function(x) format(x, scientific = FALSE, drop0trailing = TRUE)) #this ensures that the decimal points are exact every time. 
 

#png("Fe56_20220406.png", width=40*0.75, height=30*0.75, units="cm", res=600) 
ggplot(alldata, aes(
                x = temperature_C, #factor (iron_treatment, level = Fetreatment_order),
                y = ((PON_uM)/((POP_nM))),
                color = bioassay,
                shape = iron_treatment)) +
   geom_point(size = 5, alpha=0.5, stroke = 1.8)
 # stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.8, alpha = 0.5)+
 # stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1.3, width = 0.1)+
  
  scale_color_manual(name="Temperature",
                     breaks = c("T0", "LT","HT"),
                     labels = c("T0", "ambient", "warm"),
                     values = c('blue', 'black', 'orange')) +
    scale_shape_manual(name = "Iron",
                     breaks = c("T0", "noFe","Fe"), #this removes the T0 from the legend
                     labels = c("T0", "noFe", "addedFe"),
                     values = c(1, 1, 19)) + #this keeps the T0 formatting that I want
  ylab (expression ("PFe56")) + 
 xlab (expression ("DFe56")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))+
     expand_limits(y = 0) 

#dev.off()
```


ratios
```{r}
ggplot(alldata , aes(
    #y = (NO3_uM/PO4_uM),
    y = ((1000*PON_uM)/POP_nM),
    #x = iron_treatment,
    x = factor(iron_treatment, level = irontreatment_order),
    #shape = iron_treatment, 
    color = temperature_treatment))+
  #geom_point()+
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.8, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1.3, width = 0.1) +
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("LT", "HT"),
                     values = c('black', 'darkorange2','black')) +
  #scale_x_discrete(limits=c("noFe","Fe"))+
  #ylab (expression (NO[3]:PO[4]~(M:M))) + 
  #ylab (expression (NO[3]:PO[4]~drawdown)) + 
  ylab (expression (PON:POP~(M:M)))+
  xlab (expression ("")) +
  theme_bw()+
  theme(axis.text.x=element_text(face = "bold", size = 11, color = "black"),
        axis.title.y=element_text(size=14, color = "black", face = "bold"), 
        axis.text.y=element_text(face = "bold", size = 10, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 11, face = "bold"),
        legend.position = "none")+
    facet_wrap(~bioassay) +
  expand_limits(y = 10)
#####
  
  ggplot(alldata, aes(x=(PFe57_nM), 
                y=fvfm,
                color = temperature_treatment, 
                shape = iron_treatment)) + 
   
  facet_wrap(~bioassay)+
  geom_point(alpha =0.5, stroke =1, size = 6)+
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('darkorange2', 'black', 'black')) +
       scale_shape_manual(name = "PFe57",
                     breaks = c("Fe","noFe"), #this removes the T0 from the legend
                     labels = c("Fe", "noFe"),
                     values = c(19, 1, 1)) + #this keeps the T0 formatting that I want
  #scale_y_continuous(expand = c(0,5))+
  ylab (expression (F[v]/F[m])) + 
  xlab (expression ("Iron (nM)")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14)) +
       expand_limits(y = 0:0.6) 

   #geom_point(size = 5, alpha=0.5, stroke = 1 )
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.8, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = T, size = 1.3, width = 0.1)+
   scale_shape_manual(name = "",
                       breaks = c("T0", "Fe","noFe"), #this removes the T0 from the legend
                       labels = c("T0", "Fe", "noFe"),
                       values = c(2, 19, 1)) 

    ylab ("Metaproteome ribosomal fraction") +
    xlab (expression ("Temperature Â°C")) +
    theme_bw()+
theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 12), 
        legend.position = c(0.01,0.01), 
      legend.justification = c(0,0)) + #this locks the legend in position so it doesnt move around as you zoom in and out of the plot
     expand_limits(y = 0)+
     expand_limits( x = -1.5:1.5)
```


#cellcounts, cellsize
cellcounts
```{r}
#png("total_cellcount_vs_time_20220329.png", width=40*0.75, height=30*0.75, units="cm", res=600) 

x_vs_time (alldata, daycount, total_cells_mL,  Total~cells~(mL^-1)) + scale_y_log10()

x_vs_time (alldata, daycount, total_cells_mL,  Total~cells~(mL^-1)) + scale_y_log10()

#dev.off()


day6 <- filter (alldata, grepl("6", daycount))

Fe_treatment_order <- c('T0', 'noFe', 'Fe')

temp_treatment_order <- c('T0', 'LT', 'HT')

ggplot(day6, aes(
                x = (factor(temperature_treatment, level = temptreatment_order)),
                y = total_cells_mL,
                color = temperature_treatment,
                shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.8, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1.3, width = 0.1)+
  
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('black','darkorange2', 'black')) +
    scale_shape_manual(name = "Iron",
                     breaks = c("noFe","Fe"), #this removes the T0 from the legend
                     labels = c("noFe", "Fe"),
                     values = c(1,19, 1)) + #this keeps the T0 formatting that I want
  #ylab (expression (PFe[total]~~nM)) + 
  xlab (expression ("")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))
         
#dev.off()

```

cellsize
```{r}
ggplot(filter(alldata, !grepl("9", daycount)), aes(x=daycount, 
              # y=cellnumber_normalized_volume,
              #y=  (total_cell_volume_um3_mL/total_cells_mL),
              y=  total_cell_volume_um3_mL,
                color = temperature_treatment, 
                shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  #geom_point(size =3, alpha =0.5)+
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.8, alpha = 0.5)+
  #stat_summary(fun = mean, geom = "line", size = 1.1, aes(linetype = iron_treatment))+
  stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1.3, width = 0.1)+
  
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('black', 'darkorange2', 'black')) +
  
  scale_shape_manual(name = "Iron",
                     breaks = c("noFe","Fe", "T0"), #this removes the T0 from the legend
                     labels = c("noFe", "Fe", "T0"),
                     values = c(1,19, 1)) + #this keeps the T0 formatting that I want
  #scale_y_continuous(expand = c(0,5))+
  ylab (expression ("biovolume normalized to cell count")) + 
  xlab (expression ("Day Count")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))

####
cellsize <- alldata [c(14,61,63,65,67,69,71,73,75,77,79,81,83,85,87,89,91,93,95,97,99,101,103,105)]


meltedsize <- melt(cellsize, id.vars=c("full_treatment_name"))

cellsizemap <- c ("Pico2_cells_mL" = "1.18",
                  "Pico3_cells_mL" = "1.99",
                  "Pico4_cells_mL" = "2.03",
                  "Pico5_cells_mL" = "2.53",
                  "Pico6_cells_mL" = "3",
                  "Pico7_cells_mL" = "3",
                  "Nano1_cells_mL" = "3.23",
                  "Nano2_cells_mL" = "3.71",
                  "Nano3_cells_mL" = "4.31",
                  "Nano4_cells_mL" = "4.93",
                  "Nano7_cells_mL" = "5.17",
                  "Nano8_cells_mL" = "5.52",
                  "Nano9_cells_mL" = "6",
                  "Nano11_cells_mL" = "6.7",
                  "Nano12_cells_mL" = "7.06",
                  "Nano13_cells_mL" = "7.32",
                  "Nano14_cells_mL" = "7.75",
                  "Nano17_cells_mL" = "8.81",
                  "Nano19_cells_mL" = "10.36",
                  "Nano20_cells_mL" = "13.27",
                  "Nano22_cells_mL" = "15.39",
                  "Nano23_cells_mL" = "19",
                  "Nano24_cells_mL" = "19.78")


meltedsize$estimated_size <- cellsizemap [meltedsize$variable]

meltedsize <- meltedsize %>% separate(full_treatment_name, c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate")) 

meltedsize_day6 <- filter(meltedsize, grepl("T6", timepoint))

meltedsize_day6 <- filter(meltedsize_day6, grepl("BA1", bioassay))

ggplot(meltedsize_day6, aes(
  #x = variable,
  x = estimated_size,
  #x=as.numeric(estimated_size), 
                y=value,
              #y=  total_cell_volume_um3_mL,
               # color = temperature_treatment))+ 
                shape = iron_treatment)) + 
  facet_wrap(~temperature_treatment, ncol=1)+
  geom_point(size =3, alpha =0.5)+
  #stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.8, alpha = 0.5)+
  stat_summary(fun = mean, geom = "line", size = 1.1, aes(linetype = iron_treatment))+
  #stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1.3, width = 0.1)+
  
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('darkorange2', 'black', 'black')) +
  scale_shape_manual(name = "Iron",
                     breaks = c("noFe","Fe"), #this removes the T0 from the legend
                     labels = c("noFe", "addedFe"),
                     values = c(1,19, 1)) +
   #this keeps the T0 formatting that I want
  #scale_y_continuous(expand = c(0,5))+
 # ylab (expression ("norm relative size")) + 
  #xlab (expression ("Day Count")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 10, color = "black" , angle = 90),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))


#meltedsize$rounded <- round (meltedsize$value, -2)

meltedsize2 <- subset (meltedsize, rounded <200000 )

#https://aslopubs.onlinelibrary.wiley.com/doi/10.1002/lno.12023 - anderson et al. 

# https://adrienne-marshall.github.io/ggplot2_workshop/
#no big effect of temp or iron on community cell size ? 
```


POC-chla
```{r}
alldata$PONchla <- alldata$PON_ug_L/alldata$total_chla_ug_L


ggplot(filter(alldata, grepl("9|", daycount)), aes(x=daycount, 
                y=  PONchla,
                color = temperature_treatment, 
                shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  #geom_point(size =3, alpha =0.5)+
  stat_summary(fun = mean, geom = "point", show.legend = T, size = 5, stroke = 1.8, alpha = 0.5)+
  #stat_summary(fun = mean, geom = "line", size = 1.1, aes(linetype = iron_treatment))+
  stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1.3, width = 0.1)+
  
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('black', 'darkorange2', 'black')) +
  
  scale_shape_manual(name = "Iron",
                     breaks = c("noFe","Fe", "T0"), #this removes the T0 from the legend
                     labels = c("noFe", "Fe", "T0"),
                     values = c(1,19, 1)) + #this keeps the T0 formatting that I want
  ylab (expression ("PON:Chla (ug:ug)")) + 
  xlab (expression ("Day Count")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))


```


#pigments
```{r}
#chla
x_vs_time (alldata, daycount, (total_chla_ug_L), total~chla~mu*g~L^-1)

fig1_chla <- ggplot(filter (alldata, grepl ("T0|T8", full_treatment_name) ), 
       aes(x=iron_treatment, 
       y=total_chla_ug_L,
       color = temperature_treatment))+ 
       #shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  #geom_point(size = 5, alpha =0.5, stroke =1)+
  stat_summary(fun = mean, geom = "point", size = 4, stroke = 1.4, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F, size = 1, width = 0.1)+
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('darkorange2', 'black', 'black')) +
  scale_x_discrete(limits=c("T0","noFe","Fe"), 
                   labels = c("T0", "noFe", "Fe"))+
  ylab (expression (Chl-a~mu*g~L^-1)) + 
  xlab (expression ("")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black"),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14), 
        legend.position = "none") 



```

#photophysiology
raw PAM curve fitting
```{r}
setwd("D:/School/PhD/PS117/data/photophysiology/")


rawpam <- read.csv("ps117_rawPAM.csv", header = T)
rawpam <- filter (rawpam, !grepl ("blank", iron_treatment) )

rawpam$fvfm <-  (rowMeans(rawpam[,c(47:72)]) - rowMeans(rawpam[,c(17:32)]))/ rowMeans(rawpam[,c(47:72)])

rawpam2 <- rawpam [c(143,14)]


ggplot(rawpam, aes(x=fvfm, 
                y=YII)) + 
  geom_point(size =3, alpha =0.4, aes(color = temperature_treatment, 
                shape = iron_treatment))+
  geom_abline(intercept = 0, slope = 1, color = "Blue", size = 1, linetype = 2)+
  scale_x_continuous(expand = c(0, 0), limits = c(0, 0.7)) + 
  scale_y_continuous(expand = c(0, 0), limits = c(0, 0.7))+
  #geom_smooth(method='lm', formula= y~x, color = "green")+

    scale_shape_manual(name = "Iron",
                     breaks = c("T0", "noFe","Fe"), #this removes the T0 from the legend
                     labels = c("T0","noFe", "addedFe"),
                     values = c(1, 1, 19)) + #this keeps the T0 formatting that I want
  scale_color_manual(name="Temperature",
                     breaks = c("T0","LT","HT"),
                     labels = c("T0","ambient", "warm"),
                     values = c('black', 'black', "red")) +
  ylab (expression (PAM~~F[v]/F[m])) + 
  xlab (expression (Loay~~F[v]/F[m])) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"), 
        #legend.position = "top", # c(0.04,0.88), 
        legend.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 12))+ 
        
   guides(color = guide_legend(override.aes = list(size= 4)))+
   guides(shape = guide_legend(override.aes = list(size= 4)))


ggplot(rawpam, aes(x=daycount, 
                y=YII,
                color = temperature_treatment, 
                shape = iron_treatment)) + 
  #geom_point(size =5, alpha =0.5)+
  stat_summary(fun = mean, geom = "point", size = 2, stroke = 2, alpha = 0.6)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", size = 0.8, width = 0.2)+
  #geom_boxplot()+
  scale_shape_manual(name = "Iron",
                     breaks = c("T0", "noFe","Fe"), #this removes the T0 from the legend
                     labels = c("T0","noFe", "addedFe"),
                     values = c(1, 1, 19)) + #this keeps the T0 formatting that I want
  scale_color_manual(name="Temperature",
                     breaks = c("T0","LT","HT"),
                     labels = c("T0","ambient", "warm"),
                     values = c('black', 'black', 'red')) +
  #ylab (expression (F[v]/F[m])) + 
  xlab (expression ("Day Count")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"), 
        #legend.position = "top", # c(0.04,0.88), 
        legend.title = element_text(size = 12, face = "bold"),
        legend.text = element_text(size = 12)) +
   guides(color = guide_legend(override.aes = list(size= 4)))+
   guides(shape = guide_legend(override.aes = list(size= 4)))
```

plots
```{r}
#photophysiology
#png("fvfm_timeseries.png", width=40*0.75, height=30*0.75, units="cm", res=600) 
ggplot(alldata, aes(x=daycount, 
                y=fvfm,
                color = temperature_treatment, 
                shape = iron_treatment)) + 
  facet_wrap(~bioassay)+
  geom_point(size = 5, alpha =0.5, stroke =1)+
  #stat_summary(fun = mean, geom = "point", size = 5, stroke = 1.8, alpha = 0.5)+
   #stat_summary(fun.data =  mean_se, geom = "errorbar", size = 1.3, width = 0.1)+
    scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('darkorange2', 'black', 'black')) +
   
    scale_shape_manual(name = "Iron",
                     breaks = c("Fe","noFe"), #this removes the T0 from the legend
                     labels = c("Fe", "noFe"),
                     values = c(19, 1, 1)) + #this keeps the T0 formatting that I want
    #scale_y_continuous(expand = c(0,5))+
  ylab (expression (F[v]/F[m])) + 
  xlab (expression ("Day Count")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))

#dev.off()

#############

photophysi <- filter (alldata, grepl ("T0", full_treatment_name) )

#(DFe56_nM+DFe57_nM)

#png("fvfm_vs_iron.png", width=40*0.75, height=30*0.75, units="cm", res=600) 
ggplot(alldata, aes(x=(PFe57_nM), 
                y=fvfm,
                color = temperature_treatment, 
                shape = iron_treatment)) + 
   #stat_summary(fun = mean, geom = "point", size = 2, stroke = 2, alpha = 0.6)+
#  stat_summary(fun.data =  mean_se, geom = "errorbar", size = 0.8, width = 0.2)+
  facet_wrap(~bioassay)+
  geom_point(alpha =0.5, stroke =1, size = 6)+
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     labels = c("ambient", "warm"),
                     values = c('darkorange2', 'black', 'black')) +
       scale_shape_manual(name = "PFe57",
                     breaks = c("Fe","noFe"), #this removes the T0 from the legend
                     labels = c("Fe", "noFe"),
                     values = c(19, 1, 1)) + #this keeps the T0 formatting that I want
  #scale_y_continuous(expand = c(0,5))+
  ylab (expression (F[v]/F[m])) + 
  xlab (expression ("Iron (nM)")) +
  theme_bw()+
  theme(axis.title.x=element_text(size = 20,face = "bold", color = "black"), 
        axis.text.x=element_text(face = "bold", size = 15, color = "black", vjust = 0.5, hjust = 1),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14)) +
       expand_limits(y = 0:0.6) 
```



#old junk 
Figure 3 heatmap
protein clusters heatmap
```{r}
clusters <- filter (norm_proteomicsdata, grepl("pennate|centric|Bacill", F)) 
#clusters <- filter (norm_proteomicsdata, grepl("Phaeo|Prymn", F)) #clust_5911 , Cd carbonic anhydrase
#|^clust_3290$ #clust_999 transaminase, also involved in methionine recylcling

clusters <- clusters [c(12, 3, 24:53)]

clusters2 <- clusters %>% mutate_at (3:32, funs (((. / sum(.)))))

clusters3 <- filter(clusters2, grepl("clust_2934|^clust_891$|^clust_332$|^clust_808$|^clust_55$|^clust_4660|^clust_1820$|^clust_1154$|^clust_343$|^clust_1814$|^clust_534$|^clust_5562$|^clust_417$|^clust_411$|clust_2501|clust_6573|^clust_1525$|clust_1552|^clust_231$|^clust_1906$|^clust_812$|^clust_16168$|^clust_10646$|^clust_1372$|^clust_320$|^clust_6019$|^clust_35$|^clust_1830$|^clust_655$|^clust_2490$|^clust_1156$|^clust_2244$", cluster)) 



clusters3 <- melt(clusters3, id.vars=c("cluster", "F")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


clusters3_aggregated <- aggregate(clusters3$norm_abundance, by=list(Category=clusters3$cluster, clusters3$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(cluster = Category) %>% 
                    separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

clusters3_aggregated$fulltreatment <- paste (clusters3_aggregated$temperature, clusters3_aggregated$iron , sep ='_')
clusters3_aggregated$fulltreatment <- ifelse (clusters3_aggregated$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))

ggplot(clusters3_aggregated,
             aes(x = factor(fulltreatment, level = treatment_orders),
                y=norm_abundance,
                color = cluster)) +
  facet_wrap(~bioassay)+
  geom_point(size=2)+
  stat_summary(fun = mean, geom = "point", size = 6, stroke = 1, alpha = 0.5)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F,  size = 1.3, width = 0.1) +
  ylab (expression ("Fraction contribution to total diatom proteins" )) +
  xlab (expression ("")) +
  theme_bw()+
  #scale_y_log10()+
  theme(axis.text.x=element_text(face = "bold", size = 15, color = "black"),
        axis.title.y=element_text(size=20, color = "black"),
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))

clusters4_aggregated <- filter (clusters3_aggregated, !grepl("T0", iron_treatment)) #incase I want to do the heatmap with T0, I can add some random thing instead of T0 here

#clusters4_aggregated$zscore <- ave(clusters4_aggregated$norm_abundance, clusters4_aggregated$cluster, FUN=scale) #zscore for both bioassays together


clusters4_aggregated$zscore_BA <- ave(clusters4_aggregated$norm_abundance, list(clusters4_aggregated$cluster, clusters4_aggregated$bioassay), FUN=scale) #z-score for each bioassay separately


# clusters4_aggregated_means <- aggregate(list(zscore_BA =clusters4_aggregated$zscore_BA, norm_abundance = clusters4_aggregated$norm_abundance), by=list( clusters4_aggregated$fulltreatment, clusters4_aggregated$bioassay, clusters4_aggregated$cluster), FUN=mean) %>%
#   rename (fulltreatment = Group.1, 
#           bioassay = Group.2, 
#           cluster = Group.3)

#changed things here to keep replicates instead of means 

clusters4_aggregated$clust_annotation <-   ifelse (clusters4_aggregated$cluster ==  "clust_343", "ISIP-2A", 
                   ifelse (clusters4_aggregated$cluster ==  "clust_411", "Nitrate reductase", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_534", "Flavodoxin", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1154", "ISIP-3", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1814", "ISIP-1", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1820", "Plastocyanin", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_4660", "Flavodoxin-1", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_5562", "Cu-Zn SOD", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_55", "PSI", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_808", "Ferredoxin", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_2934", "Multicopper Oxidase", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_891", "Glutamate Synthase", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_332", "Chl A-B binding protein", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_417", "AMT", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_2501", "Fe-Mn SOD", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_1525", "SLC-4",
                            ifelse (clusters4_aggregated$cluster ==  "clust_231", "Rhodopsin",
                            ifelse (clusters4_aggregated$cluster ==  "clust_1906", "Mn-cluster",
                            ifelse (clusters4_aggregated$cluster ==  "clust_812", "ZIP",
                            ifelse (clusters4_aggregated$cluster ==  "clust_16168", "TonB",
                            ifelse (clusters4_aggregated$cluster ==  "clust_10646", "TonB_2",
                            ifelse (clusters4_aggregated$cluster ==  "clust_1372", "Urea transporter",
                            ifelse (clusters4_aggregated$cluster ==  "clust_320", "Urea carboxylase",
                            ifelse (clusters4_aggregated$cluster ==  "clust_6019", "Ornithine cyclodeaminase",    
                            ifelse (clusters4_aggregated$cluster ==  "clust_655", "CA_clust_655",
                            ifelse (clusters4_aggregated$cluster ==  "clust_6573", "CA_clust_6573", 
                            ifelse (clusters4_aggregated$cluster ==  "clust_35", "Ribosome recycling",
                            ifelse (clusters4_aggregated$cluster ==  "clust_2490", "Rubisco",
                            ifelse (clusters4_aggregated$cluster ==  "clust_2244", "PEPC",
                            ifelse (clusters4_aggregated$cluster ==  "clust_1830", "Acireductone dioxygenase Fe containing",
                                    clusters4_aggregated$cluster 
                    ))))))))))))))))))))))))))))))
    
#clusters4_aggregated_means$zscorenew <- ifelse (clusters4_aggregated_means$norm_abundance ==  "0", -20, 
                                          #clusters4_aggregated_means$zscore)  #this is if doing z-score for both bioassays at one

clusters4_aggregated$zscore_BAnew <- ifelse (clusters4_aggregated$norm_abundance ==  "0", NA, 
                                             clusters4_aggregated$zscore_BA) #this is for z-score for each bioassay separately


proteinclusterorder <- c("Plastocyanin", 
                         "Multi-Cu Oxidase",  
                         "Cyt Oxidase",

                         "Cu-Zn SOD",  
                         "ZIP", 
                         "RNA polymerase",
                         
                         "Fe-Mn SOD", 
                         "Mn-cluster",
                         
                         "CDCA - centric", 
                         "CDCA - pennate", 
                         
                         "Urea transporter",
                         "Urea Carboxylase")

ggplot(clusters4_aggregated, aes(x = factor(fulltreatment, level = treatment_orders),
  y = factor(clust_annotation, level = proteinclusterorder), 
  fill = zscore_BAnew)) +  
  geom_tile(color = "black", lwd = 0.7)+
  scale_fill_gradientn(colors = c("deepskyblue2","white","darkred"),
         values = rescale(c(-2.4, 0,3.1)),
         guide = "colorbar", limits=c(-2.4, 3.1),
         name = "Row Z-score")+
  scale_y_discrete(position = "right")+
  labs (x = "", y = "", title = "Phaeocysitis")+
  theme(axis.text.x=element_text(face = "bold", size = 13, color = "black"),
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank(),
        plot.title = element_text(face = "bold", size=10, vjust = 0, hjust = 0.02, margin = margin(0,0,-5,0)),
        axis.ticks  = element_blank()) +
  facet_grid(.~ bioassay, scales = "free_x") + 
  theme(strip.background =element_rect(fill="white"), 
        legend.position = "none",
        strip.text = element_text(size = 13,  face = "bold"),
        legend.box.margin=margin(l=-10))



ggplot((filter(clusters4_aggregated, grepl("Plastocyanin|Mn-cluster", clust_annotation))), aes(x = factor(fulltreatment, level = treatment_orders),
  y = zscore_BAnew, 
  shape   = clust_annotation)) +  
  
 # geom_point(size = 3, position= position_dodge(width=0.5))+ 
 # geom_boxplot()
  stat_summary(fun = mean, geom = "point", size = 6, stroke = 1, alpha = 0.5, position= position_dodge(width=0.5))+
  stat_summary(fun.data =  mean_se, geom = "errorbar", show.legend = F,  size = 1.3, width = 0.1, position= position_dodge(width=0.5)) 
  



# https://www.annualreviews.org/doi/pdf/10.1146/annurev-marine-121211-172322
# https://stackoverflow.com/questions/66304713/how-to-add-geom-tile-outside-of-axis
```

figure3 heatmap. protein clusters heatmap (see chucnk with same name below without a function)
this is the original one where I made heatmaps and plotted the means. 
```{r}
#run function below first
heatmap_diatom <- clusters_heatmap_top ("centric|pennate|Bacillariop", "Diatoms")
heatmap_phaeo <- clusters_heatmap_bottom ("Phaeo|Prymn", "Phaeocystis")

clusters_heatmap_top <- function (taxonomic_groups, plot_title) {
clusters <- filter (norm_proteomicsdata, grepl(taxonomic_groups, F)) 
clusters <- clusters [c(12, 3, 24:53)]
clusters2 <- clusters %>% mutate_at (3:32, funs (((. / sum(.)))))
clusters3 <- filter(clusters2, grepl("^clust_1820$|^clust_5562$|^clust_2934$|^clust_2501$|^clust_6573|^clust_1906$|^clust_812$|^clust_655$|^clust_225$|^clust_1217$|^clust_1372$|^clust_320$", cluster)) 


clusters3 <- melt(clusters3, id.vars=c("cluster", "F")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


clusters3_aggregated <- aggregate(clusters3$norm_abundance, by=list(Category=clusters3$cluster, clusters3$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(cluster = Category) %>% 
                    separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

clusters3_aggregated$fulltreatment <- paste (clusters3_aggregated$temperature, clusters3_aggregated$iron , sep ='_')
clusters3_aggregated$fulltreatment <- ifelse (clusters3_aggregated$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))

clusters4_aggregated <- filter (clusters3_aggregated, !grepl("T0", iron_treatment)) 
clusters4_aggregated$zscore_BA <- ave(clusters4_aggregated$norm_abundance, list(clusters4_aggregated$cluster, clusters4_aggregated$bioassay), FUN=scale) #z-score for each bioassay separately

clusters4_aggregated_means <- aggregate(list(zscore_BA =clusters4_aggregated$zscore_BA, norm_abundance = clusters4_aggregated$norm_abundance), by=list( clusters4_aggregated$fulltreatment, clusters4_aggregated$bioassay, clusters4_aggregated$cluster), FUN=mean) %>%
  rename (fulltreatment = Group.1, 
          bioassay = Group.2, 
          cluster = Group.3)

clusters4_aggregated_means$clust_annotation <-   ifelse ( clusters4_aggregated_means$cluster ==  "clust_1820", "Plastocyanin", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_2934", "Multi-Cu Oxidase", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_2501", "Fe-Mn SOD", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_5562", "Cu-Zn SOD - centric", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_6573", "CA_clust_6573-centric", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_655", "CA_clust_655-pennate",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1906", "Mn-cluster",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_812", "ZIP",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_225", "Cyt Oxidase",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1217", "RNA polymerase",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1372", "Urea transporter",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_320", "Urea Carboxylase",
                            clusters4_aggregated_means$cluster))))))))))))
    
clusters4_aggregated_means$zscore_BAnew <- ifelse (clusters4_aggregated_means$norm_abundance ==  "0", -20, 
                                             clusters4_aggregated_means$zscore_BA) #this is for z-score for each bioassay separately

proteinclusterorder <- c("Plastocyanin", 
                         "Multi-Cu Oxidase",  
                         "Cyt Oxidase",
                         "Cu-Zn SOD - centric",  
                         "ZIP", 
                         "RNA polymerase",
                         "Fe-Mn SOD", 
                         "Mn-cluster",
                         "CDCA - centric", 
                         "CDCA - pennate", 
                         "Urea transporter",
                         "Urea Carboxylase")

ggplot(clusters4_aggregated_means, aes(x = factor(fulltreatment, level = treatment_orders),
  y = factor(clust_annotation, level = proteinclusterorder), 
  fill = zscore_BAnew)) +  
  geom_tile(color = "black", lwd = 0.7)+
  scale_fill_gradientn(colors = c("deepskyblue2","white","darkred"),
         values = rescale(c(-1.2,0,1.7)),
         guide = "colorbar", limits=c(-1.2,1.7),
         name = "Row Z-score")+
  scale_y_discrete(position = "right")+
  labs (x = "", y = "", title = {{plot_title}})+
  theme(axis.text.x= element_blank(),
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank(),
        plot.title = element_text(face = "bold", size=13, vjust = 0, hjust = 0.02, margin = margin(0,0,-15,0)),
 plot.margin = margin(0, 0, -0.3, -0.5, "cm"), #trbl
        axis.ticks  = element_blank()) +
  facet_grid(.~ bioassay, scales = "free_x") + 
  theme(strip.background =element_rect(fill="white"), 
        legend.position = "none",
        strip.text = element_text(size = 13,  face = "bold"),
        legend.box.margin=margin(l=-10))
}

clusters_heatmap_bottom <- function (taxonomic_groups, plot_title) {
clusters <- filter (norm_proteomicsdata, grepl(taxonomic_groups, F)) 
clusters <- clusters [c(12, 3, 24:53)]
clusters2 <- clusters %>% mutate_at (3:32, funs (((. / sum(.)))))
clusters3 <- filter(clusters2, grepl("^clust_1820$|^clust_5562$|^clust_2934$|^clust_2501$|^clust_6573|^clust_1906$|^clust_812$|^clust_655$|^clust_225$|^clust_1217$|^clust_1372$|^clust_320$", cluster)) 


clusters3 <- melt(clusters3, id.vars=c("cluster", "F")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


clusters3_aggregated <- aggregate(clusters3$norm_abundance, by=list(Category=clusters3$cluster, clusters3$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(cluster = Category) %>% 
                    separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

clusters3_aggregated$fulltreatment <- paste (clusters3_aggregated$temperature, clusters3_aggregated$iron , sep ='_')
clusters3_aggregated$fulltreatment <- ifelse (clusters3_aggregated$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))

clusters4_aggregated <- filter (clusters3_aggregated, !grepl("T0", iron_treatment)) 
clusters4_aggregated$zscore_BA <- ave(clusters4_aggregated$norm_abundance, list(clusters4_aggregated$cluster, clusters4_aggregated$bioassay), FUN=scale) #z-score for each bioassay separately

clusters4_aggregated_means <- aggregate(list(zscore_BA =clusters4_aggregated$zscore_BA, norm_abundance = clusters4_aggregated$norm_abundance), by=list( clusters4_aggregated$fulltreatment, clusters4_aggregated$bioassay, clusters4_aggregated$cluster), FUN=mean) %>%
  rename (fulltreatment = Group.1, 
          bioassay = Group.2, 
          cluster = Group.3)

clusters4_aggregated_means$clust_annotation <-   ifelse ( clusters4_aggregated_means$cluster ==  "clust_1820", "Plastocyanin", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_2934", "Multi-Cu Oxidase", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_2501", "Fe-Mn SOD", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_5562", "Cu-Zn SOD - centric", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_6573", "CA_clust_6573-centric", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_655", "CA_clust_655-pennate",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1906", "Mn-cluster",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_812", "ZIP",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_225", "Cyt Oxidase",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1217", "RNA polymerase",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1372", "Urea transporter",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_320", "Urea Carboxylase",
                            clusters4_aggregated_means$cluster))))))))))))
    
clusters4_aggregated_means$zscore_BAnew <- ifelse (clusters4_aggregated_means$norm_abundance ==  "0", -20, 
                                             clusters4_aggregated_means$zscore_BA) #this is for z-score for each bioassay separately

proteinclusterorder <- c("Plastocyanin", 
                         "Multi-Cu Oxidase",  
                         "Cyt Oxidase",
                         "Cu-Zn SOD - centric",  
                         "ZIP", 
                         "RNA polymerase",
                         "Fe-Mn SOD", 
                         "Mn-cluster",
                         "CDCA - centric", 
                         "CDCA - pennate", 
                         "Urea transporter",
                         "Urea Carboxylase")

ggplot(clusters4_aggregated_means, aes(x = factor(fulltreatment, level = treatment_orders),
  y = factor(clust_annotation, level = proteinclusterorder), 
  fill = zscore_BAnew)) +  
  geom_tile(color = "black", lwd = 0.7)+
  scale_fill_gradientn(colors = c("deepskyblue2","white","darkred"),
         values = rescale(c(-1.2,0,1.7)),
         guide = "colorbar", limits=c(-1.2,1.7),
         name = "Row Z-score")+
  scale_y_discrete(position = "right")+
  labs (x = "", y = "", title = {{plot_title}})+
  theme(axis.text.x= element_text(face = "bold", size = 13, color = "black"),
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank(),
        plot.title = element_text(face = "bold", size=13, vjust = 0, hjust = 0.02, margin = margin(0.5,0,1.5,0)),
 plot.margin = margin(0, 0, -0.3, -0.5, "cm"), #trbl
        axis.ticks  = element_blank()) +
  facet_grid(.~ bioassay, scales = "free_x") + 
  theme(strip.background =element_rect(fill="white"), 
        legend.position = "none",
        legend.text = element_text(face = "bold", size=13),
        legend.title = element_text (face = "bold", size = 13),
        strip.text = element_blank())+
        #legend.box.margin=margin(l=-10))+ 
   guides(fill = guide_colourbar(barwidth = 1.5, barheight = 1.5))
}


##legend
x<- c(1,2,3)
y <- c("a", "b", "c")
colr <- c (-1.2,0, 1.7)
z <- data.frame(x,y, colr)

legend3b <- ggplot(z, aes (x,y, fill = colr))+
   geom_point()+
    scale_fill_gradientn(colors = c("deepskyblue2","white","darkred"),
         values = rescale(c(-1.2,0,1.7)),
         guide = "colorbar", limits=c(-1.2,1.7),
         name = "Row Z-score")+
   theme(legend.text = element_text(face = "bold", size=13),
        legend.title = element_text (face = "bold", size = 13, angle = 90), 
        legend.title.align=0.5)+
   guides(fill = guide_colourbar(title.position = "left", barwidth = 1.5, barheight = 15))
  

legend3b <- get_legend(legend3b)
legend3b <- as_ggplot(legend3b)
```

Figure3 - metals, heatmap
```{r}
metals <- plot_grid(NULL, metals_Mn, NULL, NULL, metals_Cu,
          NULL, metals_Zn, NULL, NULL, metals_Ni, 
          NULL, metals_Cd, NULL, NULL, metals_Co, 
          ncol =5,
          rel_widths = c(0.05, 1, 0.1, 0.06, 1),
          labels = c("A", "", "", "D", "",
                     "B", "", "", "E", "",
                     "C", "", "", "F", ""), 
          align = 'v')

heatmaps <- plot_grid(NULL, heatmap_diatom, 
          NULL, heatmap_phaeo,
          nrow =2,
          rel_heights = c(1,0.7),
          rel_widths = c(0.05,1),
          labels = c("G", "",
                     "H", ""), 
          align = 'v')

legends <- plot_grid(NULL, legend3b,legend3a,
          nrow =3, 
          rel_heights = c(0.2,2,1))

plot_grid(metals, NULL, legends,  heatmaps,
          ncol=4,
          rel_widths = c(1,0.05,0.06, 1.2), 
          align = 'hv') 

```

```{r}

clusters_heatmap_top <- function (taxonomic_groups, plot_title, protein) {
clusters <- filter (norm_proteomicsdata, grepl(taxonomic_groups, F)) 
clusters <- clusters [c(12, 3, 24:53)]
clusters2 <- clusters %>% mutate_at (3:32, funs (((. / sum(.)))))
clusters3 <- filter(clusters2, grepl("clust_2934|^clust_891$|^clust_332$|^clust_808$|^clust_55$|^clust_4660|^clust_1820$|^clust_1154$|^clust_343$|^clust_1814$|^clust_534$|^clust_5562$|^clust_417$|^clust_411$|clust_2501|clust_6573|^clust_1525$|clust_1552|^clust_231$|^clust_1906$|^clust_812$|^clust_16168$|^clust_10646$|^clust_1372$|^clust_320$|^clust_6019$|^clust_35$|^clust_1830$|^clust_655$|^clust_2490$|^clust_1156$|^clust_2244$", cluster)) 


clusters3 <- melt(clusters3, id.vars=c("cluster", "F")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


clusters3_aggregated <- aggregate(clusters3$norm_abundance, by=list(Category=clusters3$cluster, clusters3$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(cluster = Category) %>% 
                    separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

clusters3_aggregated$fulltreatment <- paste (clusters3_aggregated$temperature, clusters3_aggregated$iron , sep ='_')
clusters3_aggregated$fulltreatment <- ifelse (clusters3_aggregated$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))

clusters4_aggregated <- filter (clusters3_aggregated, !grepl("T0", iron_treatment)) 
clusters4_aggregated$zscore_BA <- ave(clusters4_aggregated$norm_abundance, list(clusters4_aggregated$cluster, clusters4_aggregated$bioassay), FUN=scale) #z-score for each bioassay separately

clusters4_aggregated_means <- aggregate(list(zscore_BA =clusters4_aggregated$zscore_BA, norm_abundance = clusters4_aggregated$norm_abundance), by=list( clusters4_aggregated$fulltreatment, clusters4_aggregated$bioassay, clusters4_aggregated$cluster), FUN=mean) %>%
  rename (fulltreatment = Group.1, 
          bioassay = Group.2, 
          cluster = Group.3)


clusters4_aggregated_means$clust_annotation <-   ifelse (clusters4_aggregated_means$cluster ==  "clust_343", "ISIP-2A", 
                   ifelse (clusters4_aggregated_means$cluster ==  "clust_411", "Nitrate reductase", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_534", "Flavodoxin", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1154", "ISIP-3", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1814", "ISIP-1", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1820", "Plastocyanin", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_4660", "Flavodoxin-1", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_5562", "Cu-Zn SOD", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_55", "PSI", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_808", "Ferredoxin", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_2934", "Multicopper Oxidase", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_891", "Glutamate Synthase", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_332", "Chl A-B binding protein", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_417", "AMT", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_2501", "Fe-Mn SOD", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1525", "SLC-4",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_231", "Rhodopsin",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1906", "Mn-cluster",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_812", "ZIP",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_16168", "TonB",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_10646", "TonB_2",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1372", "Urea transporter",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_320", "Urea carboxylase",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_6019", "Ornithine cyclodeaminase",    
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_655", "CA_clust_655",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_6573", "CA_clust_6573", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_35", "Ribosome recycling",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_2490", "Rubisco",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_2244", "PEPC",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1830", "Acireductone dioxygenase Fe containing",
                                    clusters4_aggregated_means$cluster 
                    ))))))))))))))))))))))))))))))




    
clusters4_aggregated_means$zscore_BAnew <- ifelse (clusters4_aggregated_means$norm_abundance ==  "0", -20, 
                                             clusters4_aggregated_means$zscore_BA) #this is for z-score for each bioassay separately

proteinclusterorder <- c("ISIP-2A", 
                         "Nitrate reductase", 
                         "Flavodoxin",
                         "ISIP-3",
                         "ISIP-1",
                         "Plastocyanin", 
                         "Flavodoxin-1",
                         "Cu-Zn SOD", 
                         "PSI",
                        "Ferredoxin", 
                        "Multicopper Oxidase",
                        "Glutamate Synthase", 
                        "Chl A-B binding protein", 
                        "AMT", 
                        "Fe-Mn SOD", 
                        "Cd Carbonic Anhydrase", 
                        "SLC-4",
                        "Rhodopsin",
                        "Mn-cluster",
                        "ZIP",
                        "TonB",
                        "TonB_2",
                        "Urea transporter",
                        "Urea carboxylase",
                        "Ornithine cyclodeaminase",
                        "CA",
                        "Ribosome recycling",
                        "Rubisco",
                        "PEPC",
                        "Acireductone dioxygenase Fe containing")

ggplot((filter(clusters4_aggregated_means, grepl (protein, clust_annotation))), aes(x = factor(fulltreatment, level = tempFe_order),
  y = factor(clust_annotation, level = proteinclusterorder), 
  fill = zscore_BAnew)) +  
  geom_tile(color = "black", lwd = 0.7)+
  scale_fill_gradientn(colors = c("deepskyblue2","white","darkred"),
         values = rescale(c(-1.2,0,1.7)),
         guide = "colorbar", limits=c(-1.2,1.7),
         name = "Row Z-score")+
  scale_y_discrete(position = "right")+
  labs (x = "", y = "", title = {{plot_title}})+
  theme(axis.text.x= element_blank(),
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank(),
        plot.title = element_text(face = "bold", size=13, vjust = 0, hjust = 0.02, margin = margin(0,0,-15,0)),
 plot.margin = margin(0, 0, -0.3, -0.5, "cm"), #trbl
        axis.ticks  = element_blank()) +
  facet_grid(.~ bioassay, scales = "free_x") + 
  theme(strip.background =element_rect(fill="white"), 
        legend.position = "none",
        strip.text = element_text(size = 13,  face = "bold"),
        legend.box.margin=margin(l=-10))
}

clusters_heatmap_bottom <- function (taxonomic_groups, plot_title) {
clusters <- filter (norm_proteomicsdata, grepl(taxonomic_groups, F)) 
clusters <- clusters [c(12, 3, 24:53)]
clusters2 <- clusters %>% mutate_at (3:32, funs (((. / sum(.)))))
clusters3 <- filter(clusters2, grepl("^clust_1820$|^clust_5562$|^clust_2934$|^clust_2501$|^clust_6573|^clust_1906$|^clust_812$|^clust_655$|^clust_225$|^clust_1217$|^clust_1372$|^clust_320$", cluster)) 


clusters3 <- melt(clusters3, id.vars=c("cluster", "F")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


clusters3_aggregated <- aggregate(clusters3$norm_abundance, by=list(Category=clusters3$cluster, clusters3$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(cluster = Category) %>% 
                    separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")

clusters3_aggregated$fulltreatment <- paste (clusters3_aggregated$temperature, clusters3_aggregated$iron , sep ='_')
clusters3_aggregated$fulltreatment <- ifelse (clusters3_aggregated$fulltreatment ==  "LT_noFe", "LT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "LT_Fe", "LT_Fe",
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_noFe", "HT_noFe", 
                            ifelse (clusters3_aggregated$fulltreatment ==  "HT_Fe", "HT_Fe",
                            "T0"))))

clusters4_aggregated <- filter (clusters3_aggregated, !grepl("T0", iron_treatment)) 
clusters4_aggregated$zscore_BA <- ave(clusters4_aggregated$norm_abundance, list(clusters4_aggregated$cluster, clusters4_aggregated$bioassay), FUN=scale) #z-score for each bioassay separately

clusters4_aggregated_means <- aggregate(list(zscore_BA =clusters4_aggregated$zscore_BA, norm_abundance = clusters4_aggregated$norm_abundance), by=list( clusters4_aggregated$fulltreatment, clusters4_aggregated$bioassay, clusters4_aggregated$cluster), FUN=mean) %>%
  rename (fulltreatment = Group.1, 
          bioassay = Group.2, 
          cluster = Group.3)

clusters4_aggregated_means$clust_annotation <-   ifelse ( clusters4_aggregated_means$cluster ==  "clust_1820", "Plastocyanin", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_2934", "Multi-Cu Oxidase", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_2501", "Fe-Mn SOD", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_5562", "Cu-Zn SOD - centric", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_6573", "CA_clust_6573-centric", 
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_655", "CA_clust_655-pennate",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1906", "Mn-cluster",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_812", "ZIP",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_225", "Cyt Oxidase",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1217", "RNA polymerase",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_1372", "Urea transporter",
                            ifelse (clusters4_aggregated_means$cluster ==  "clust_320", "Urea Carboxylase",
                            clusters4_aggregated_means$cluster))))))))))))
    
clusters4_aggregated_means$zscore_BAnew <- ifelse (clusters4_aggregated_means$norm_abundance ==  "0", -20, 
                                             clusters4_aggregated_means$zscore_BA) #this is for z-score for each bioassay separately

proteinclusterorder <- c("Plastocyanin", 
                         "Multi-Cu Oxidase",  
                         "Cyt Oxidase",
                         "Cu-Zn SOD - centric",  
                         "ZIP", 
                         "RNA polymerase",
                         "Fe-Mn SOD", 
                         "Mn-cluster",
                         "CDCA - centric", 
                         "CDCA - pennate", 
                         "Urea transporter",
                         "Urea Carboxylase")

ggplot(clusters4_aggregated_means, aes(x = factor(fulltreatment, level = tempFe_order),
  y = factor(clust_annotation, level = proteinclusterorder), 
  fill = zscore_BAnew)) +  
  geom_tile(color = "black", lwd = 0.7)+
  scale_fill_gradientn(colors = c("deepskyblue2","white","darkred"),
         values = rescale(c(-1.2,0,1.7)),
         guide = "colorbar", limits=c(-1.2,1.7),
         name = "Row Z-score")+
  scale_y_discrete(position = "right")+
  labs (x = "", y = "", title = {{plot_title}})+
  theme(axis.text.x= element_text(face = "bold", size = 13, color = "black"),
        axis.text.y=element_text(face = "bold", size = 11, color = "black"),
        panel.border = element_blank(),
        panel.background = element_blank(),
        plot.title = element_text(face = "bold", size=13, vjust = 0, hjust = 0.02, margin = margin(0.5,0,1.5,0)),
 plot.margin = margin(0, 0, -0.3, -0.5, "cm"), #trbl
        axis.ticks  = element_blank()) +
  facet_grid(.~ bioassay, scales = "free_x") + 
  theme(strip.background =element_rect(fill="white"), 
        legend.position = "none",
        legend.text = element_text(face = "bold", size=13),
        legend.title = element_text (face = "bold", size = 13),
        strip.text = element_blank())+
        #legend.box.margin=margin(l=-10))+ 
   guides(fill = guide_colourbar(barwidth = 1.5, barheight = 1.5))
}




```

NMDS plots
Protein taxonomy NMDS 2022-07-27
```{r}
proteomicsdata_euk <- filter(norm_proteomicsdata, grepl("Euk", domain))

proteomics_taxonomy <- proteomicsdata_euk [c(10, 24:53)]

proteomics_taxonomy <- filter (proteomics_taxonomy, grepl("Bacillario|Dinophy|Haptoph|Prymnesio|Pavlov|Cryptoph|Pelagophy|Chlorophy|Pyramim|Prasinophy", class))

proteomics_taxonomy <- proteomics_taxonomy %>% mutate_at (2:31, funs ((100*(. / sum(.)))))

proteomics_taxonomy <- melt(proteomics_taxonomy, id.vars=c("class")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )

proteomics_taxonomy_aggregated <- aggregate(proteomics_taxonomy$norm_abundance, by=list(Category=proteomics_taxonomy$class, proteomics_taxonomy$treatment), FUN=sum)%>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(genus = Category)

proteomics_taxonomy_aggregated_2 <- dcast(proteomics_taxonomy_aggregated, treatment ~ genus)

proteomics_taxonomy_aggregated_2 <- proteomics_taxonomy_aggregated_2 %>% separate(treatment, into=c("bioassay", "timepoint" ,"temperature_treatment", "iron_treatment", "replicate"), sep = "_")

proteomics_taxonomy_aggregated_2_nmds = proteomics_taxonomy_aggregated_2 [c(6:18)] #select only the class/genus..etc

proteomics_taxonomy_aggregated_2_nmds = as.matrix(proteomics_taxonomy_aggregated_2_nmds)

set.seed(123)
nmds = metaMDS(proteomics_taxonomy_aggregated_2_nmds, try = 1000, distance = "bray") # try = 500, trymax = 500
nmds
scores(nmds, display = "sites")

data.scores <- as.data.frame(scores(nmds, "sites")) #here we can select sites or species
stressplot(nmds) #for the suplement

data.scores$temperature_treatment <- proteomics_taxonomy_aggregated_2$temperature_treatment
data.scores$iron_treatment <- proteomics_taxonomy_aggregated_2$iron_treatment
data.scores$bioassay <- proteomics_taxonomy_aggregated_2$bioassay

data.scores$tempfe <- paste(data.scores$temperature_treatment,data.scores$iron_treatment)

fig2_proteomicsnmds <- ggplot(data.scores, aes(x = NMDS1, 
                        y = NMDS2))+
  geom_point(size = 3.5, 
                aes(fill = temperature_treatment,
                    alpha = iron_treatment, 
                    shape = bioassay,
                    color = temperature_treatment))+
  geom_point(size=3.5, 
             aes(shape = bioassay, 
                 color = temperature_treatment) ) +
  scale_shape_manual(values=c(21, 24),
                     name = "Bioassay")+
  scale_color_manual(values = c("HT" = "darkorange", "LT" = "black", "T0" = "blue"), 
                     name = "Temperature")+
  scale_alpha_manual(values = c("Fe" = 1, "noFe" = 0.2, "T0" = 0.2) , 
                    name = "Iron") +
  scale_fill_manual(values = c("HT" = "darkorange", "LT" = "black", "T0" = "blue"),    guide = FALSE) +
  ggtitle("Proteomics")+
    theme(axis.text.y = element_text(colour = "black", size = 11, face = "bold"), 
    axis.text.x = element_text(colour = "black", face = "bold", size = 11), 
    legend.text = element_text(size = 12, face ="bold", colour ="black"), 
    legend.position = 'none', axis.title.y = element_text(face = "bold", size = 12), 
    axis.title.x = element_text(face = "bold", size = 12, colour = "black"), 
    legend.title = element_text(size = 14, colour = "black", face = "bold"), 
    panel.background = element_blank(), panel.border = element_rect(colour = "black",     fill = NA, size = 1.2),
    legend.key=element_blank(), 
   plot.title = element_text(vjust = -8, hjust = 0.02))
```

PCA
```{r}
com0 = alldata [c(4,11,12,13,64:88)]
#com0 = alldata [c(4,11,12,13,33:46)]
#com0 = alldata [c(14,47:50)]
com = na.omit(com0)


com1 <- com[c(5:29)]

com_pca <- t(com1)

colnames(com_pca) <- com_pca[1,]
com_pca <- com_pca[-1,]
com_pca <- data.frame(com_pca)

com_pca <- sapply(com_pca, as.numeric)
com_pca <- data.frame(com_pca)



pca <- prcomp ((m_com2), scale = TRUE)
plot(pca$x[,1], pca$x[,2])

pca.var <- pca$sdev^2
pca.var.per <- round(pca.var/sum(pca.var)*100,1)

barplot(pca.var.per, main = "Scree Plot", xlab = "Principal Component", ylab = "% variation")


pca.data <- data.frame(Sample = rownames(pca$x), 
                        X = pca$x[,1], 
                        Y = pca$x[,2])

pca.data <- pca.data %>% separate(Sample, c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"))

ggplot(pca.data, aes(x = X, y = Y,
                        shape = bioassay, 
                        color = temperature_treatment)) + 
    geom_point(size = 5)+
    scale_colour_manual(values = c("darkorange", "black", "blue")) +
  scale_shape_manual(values =c (19, 1,1))+
    theme(axis.text.y = element_text(colour = "black", size = 12, face = "bold"), 
    axis.text.x = element_text(colour = "black", face = "bold", size = 12), 
    legend.text = element_text(size = 12, face ="bold", colour ="black"), 
    legend.position = "right", axis.title.y = element_text(face = "bold", size = 14), 
    axis.title.x = element_text(face = "bold", size = 14, colour = "black"), 
    legend.title = element_text(size = 14, colour = "black", face = "bold"), 
    panel.background = element_blank(), panel.border = element_rect(colour = "black", fill = NA, size = 1.2),
    legend.key=element_blank()) + 
    labs(x = paste("PC1 - ", pca.var.per[1], "%", sep = ""), 
         y = paste("PC2 - ", pca.var.per[2], "%", sep = ""),
         colour = "Temperature",
         fill = "Iron", 
         shape = "Iron")  

```

treemap
```{r}
ggplotly(x)
#treemap
#https://r-charts.com/part-whole/treemapify/
#https://r-charts.com/part-whole/ggvoronoi/
class_treemap <- aggregate(class_aggregated$norm_abundance, by=list(Category=class_aggregated$treatment, class_aggregated$class, class_aggregated$bioassay), FUN=mean) %>%
                  rename(norm_abundance = x) %>%
                  rename(class = Group.2) %>%
                  rename(treatment = Category) %>%
                  rename(bioassay = Group.3)

class_treemap_t0 <- filter (class_treemap, grepl('HT_noFe|HT_Fe',treatment)) 

class_treemap_ba1 <- filter (class_treemap_t0, grepl('BA1',bioassay)) 

ggplot (class_treemap_ba1, aes (area = norm_abundance, fill = class, label = class)) + 
  geom_treemap() +
  theme (legend.position = "none")+
  geom_treemap_text (color = "white", 
                     place = "centre",
                     #size = 15,
                     grow = TRUE) + 
      facet_grid(.~ treatment)


#####
##genus (frag, pseudo-nitz etc. )

all_genus_euks <- filter (all_taxonomy, grepl('Eukar',domain)) 
all_genus <- all_genus_euks [c(69, 2:61)]

all_genus <- melt(all_genus, id.vars=c("genus")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )


all_genus$genus_a <-        #diatoms 
                            ifelse (all_genus$genus ==  "Amphiprora", "Other diatom",
                            ifelse (all_genus$genus ==  "Araphid-pennate", "Pennate diatom",
                            ifelse (all_genus$genus ==  "Bacillariophyta_X", "Other diatom",
                            ifelse (all_genus$genus ==  "Chaetoceros", "Chaetoceros",
                            ifelse (all_genus$genus ==  "Corethron", "Other diatom",
                            ifelse (all_genus$genus ==  "Ditylum", "Other diatom",
                            ifelse (all_genus$genus ==  "Fragilariopsis", "Fragilariopsis", 
                            ifelse (all_genus$genus ==  "Nitzschia", "Pennate diatom",  
                            ifelse (all_genus$genus ==  "Odontella", "Odontella",
                            ifelse (all_genus$genus ==  "Polar-centric-Mediophyceae", "Polar centrics",
                            ifelse (all_genus$genus ==  "Proboscia", "Other diatom",
                            ifelse (all_genus$genus ==  "Pseudo-nitzschia", "Pseudo-nitzschia",
                            ifelse (all_genus$genus ==  "Raphid-pennate", "Pennate diatom",
                            ifelse (all_genus$genus ==  "Skeletonema", "Other diatom",
                            ifelse (all_genus$genus ==  "Thalassiosira", "Thalassiosira",
                            #dinos  
                            ifelse (all_genus$genus ==  "Alexandrium", "Dinophytes",
                            ifelse (all_genus$genus ==  "Dinophyceae_XX", "Dinophytes",
                            ifelse (all_genus$genus ==  "Dinophyceae_X", "Dinophytes",
                            ifelse (all_genus$genus ==  "Dinophyta", "Dinophytes",
                            ifelse (all_genus$genus ==  "Dinophysis", "Dinophytes",
                            ifelse (all_genus$genus ==  "Durinskia", "Dinophytes",
                            ifelse (all_genus$genus ==  "Karlodinium", "Dinophytes",
                            #haptophytes
                            ifelse (all_genus$genus ==  "Ciliophora", "Ciliates",
                            ifelse (all_genus$genus ==  "Emiliania", "Prymnesiophyceae",
                            ifelse (all_genus$genus ==  "Phaeocystis", "Phaeocystis",        
                            ifelse (all_genus$genus ==  "Prymnesiophyceae", "Prymnesiophyceae",
                            "All Other"))))))))))))))))))))))))))

genus_aggregated <- aggregate(all_genus$norm_abundance, by=list(Category=all_genus$genus_a, all_genus$treatment), FUN=sum)%>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Group.2) %>%
                  rename(genus = Category)

genus_aggregated <- genus_aggregated %>% separate(treatment, into=c("bioassay", "temperature", "iron", "timepoint", "replicate", "injection"), sep = "_")

genus_aggregated$replicate <- str_sub(genus_aggregated$replicate, -2,-2)
genus_aggregated$injection <- str_sub(genus_aggregated$injection, 2)

genus_aggregated$treatment <- paste (genus_aggregated$temperature, genus_aggregated$iron, sep ='_')


treatment_order <- c('T0_T0', 'LT_noFe', 'LT_Fe', 'HT_noFe', 'HT_Fe')

genus_aggregated1 <- filter (genus_aggregated, grepl('Fragil|Pseudo',genus)) 

ggplot (genus_aggregated1, aes (x = (factor(treatment, level = treatment_order)), y = norm_abundance, color = genus)) +  
  stat_summary(fun = mean, geom = "point", size = 3, stroke = 1, alpha = 0.6)+
  stat_summary(fun.data =  mean_se, geom = "errorbar", size = 0.7, width = 0.1)+
  #geom_jitter(size = 3, width = 0.05, height = 0.5)+
  #geom_point(size = 4, alpha = 0.4)+
  theme_bw()+
  #scale_y_log10()+
  #theme (legend.position = "none", 
        theme (axis.text.x = element_text(angle = 90)) +
  xlab("")
  facet_grid(bioassay ~.)

ggplotly(y)

#treemap
#https://r-charts.com/part-whole/treemapify/
#https://r-charts.com/part-whole/ggvoronoi/
genus_treemap <- aggregate(genus_aggregated$norm_abundance, by=list(Category=genus_aggregated$treatment, genus_aggregated$genus, genus_aggregated$bioassay), FUN=mean) %>%
                  rename(norm_abundance = x) %>%
                  rename(genus = Group.2) %>%
                  rename(treatment = Category) %>%
                  rename(bioassay = Group.3)

genus_treemap_t0 <- filter (genus_treemap, grepl('HT_noFe|HT_Fe',treatment)) 

genus_treemap_ba1 <- filter (genus_treemap_t0, grepl('BA2',bioassay)) 

ggplot (genus_treemap_ba1, aes (area = norm_abundance, fill = genus, label = genus)) + 
  geom_treemap() +
  theme (legend.position = "none")+
  geom_treemap_text (color = "white", 
                     place = "centre",
                     #size = 15,
                     grow = TRUE) + 
      facet_grid(.~ treatment)

```


PTMS 
```{r}
ptms <- filter (norm_proteomicsdata, grepl ( "Gln", peptide))

ptms2 <- ptms [c(23:53)]


ptms3 <- melt(ptms2, id.vars=c("grpnorm_compartment")) %>%
            rename(treatment = variable) %>%
            rename(norm_abundance = value )
   

ptms3_aggregated <- aggregate(ptms3$norm_abundance, by=list(Category=ptms3$treatment), FUN=sum) %>%
                  rename(norm_abundance = x) %>%
                  rename(treatment = Category) %>%
                  separate(treatment, into=c("bioassay", "timepoint", "temperature_treatment", "iron_treatment", "replicate"), sep = "_")



ggplot(ptms3_aggregated, aes(x = iron_treatment,
 y = norm_abundance, 
  color  = temperature_treatment))+
 #shape = replicate))+ 
 #shape = grpnorm_compartment)) +
  #geom_point(size = 4) +
   stat_summary(fun = mean, geom = "point", size = 6, stroke = 1, alpha = 0.5) + 
   stat_summary(fun.data = mean_se, geom = "errorbar", size = 2, alpha = 0.5) + 
    facet_wrap(~bioassay) + 
  scale_color_manual(name="Temperature",
                     breaks = c("LT","HT"),
                     values = c( 'black','darkorange2', "black")) +
  xlab (expression ("")) +
  theme_bw()+
  theme(axis.text.x=element_text(face = "bold", size = 8, color = "black"),
        axis.title.y=element_text(size=20, color = "black"), 
        axis.text.y=element_text(face = "bold", size = 15, color = "black"),
        strip.background =element_rect(fill = "white"),
        strip.text.x = element_text(size = 15, face = "bold"),
        legend.title = element_text(size = 14, face = "bold"),
        legend.text = element_text(size = 14))
```

